{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MODELLER examples\n",
    "- Here is an exmaple/tutorial on doing homology modeling with modeller.\n",
    "- We follow in part this tutorial https://salilab.org/modeller/tutorial/basic.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Example 1: build a model from one structure with one chain\n",
    " - We have chicken histone H1 structure, PDB id 1HST\n",
    " - We want to build a human model of UniProtID P07305"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's get template structure\n",
    "import pypdb\n",
    "import nglview as nv\n",
    "\n",
    "with open('1HST.pdb','w') as f:\n",
    "    f.write(pypdb.get_pdb_file('1HST', filetype='pdb', compression=False))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SHPTYSEMIAAAIRAEKSRGGSSRQSIQKYIKSHYKVGHNADLQIKLSIRRLLAAGVLKQTKGVGASGSFRLAK\n",
      "SHPTYSEMIAAAIRAEKSRGGSSRQSIQKYIKSHYKVGHNADLQIKLSIRRLLAAGVLKQTKGVGASGSFRLAK\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "9e47d5548ed64b74b13555b4707bc7e5",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "NGLWidget()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Let's get template sequence\n",
    "from Bio.PDB import *\n",
    "#See docs here https://biopython.org/wiki/The_Biopython_Structural_Bioinformatics_FAQ\n",
    "p = PDBParser()\n",
    "structure = p.get_structure('H1', '1HST.pdb')\n",
    "ppb = PPBuilder()\n",
    "for pp in ppb.build_peptides(structure):\n",
    "    print(pp.get_sequence())\n",
    "# Let's view structure\n",
    "nv.show_biopython(structure)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "H10_HUMAN\n",
      "MTENSTSAPAAKPKRAKASKKSTDHPKYSDMIVAAIQAEKNRAGSSRQSIQKYIKSHYKVGENADSQIKLSIKRLVTTGVLKQTKGVGASGSFRLAKSDEPKKSVAFKKTKKEIKKVATPKKASKPKKAASKAPTKKPKATPVKKAKKKLAATPKKAKKPKTVKAKPVKASKPKKAKPVKPKAKSSAKRAGKKK\n"
     ]
    }
   ],
   "source": [
    "#Let's get target seqeunce\n",
    "from Bio import SeqIO\n",
    "import urllib.request\n",
    "handle = urllib.request.urlopen(\"http://www.uniprot.org/uniprot/P07305.xml\")\n",
    "record = SeqIO.read(handle, \"uniprot-xml\")\n",
    "#print(record)\n",
    "print(record.name)\n",
    "print(record.seq)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "A tricky thing about MODELLER is the preparation of sequences and/or alignment files in PIR-like-format with additional information.\n",
    "Here is an example\n",
    "```\n",
    ">P1;TvLDH\n",
    "sequence:TvLDH:::::::0.00: 0.00\n",
    "MSEAAHVLITGAAGQIGYILSHWIASGELYGDRQVYLHLLDIPPAMNRLTALTMELEDCAFPHLAGFVATTDPKA\n",
    "AFKDIDCAFLVASMPLKPGQVRADLISSNSVIFKNTGEYLSKWAKPSVKVLVIGNPDNTNCEIAMLHAKNLKPEN\n",
    "FSSLSMLDQNRAYYEVASKLGVDVKDVHDIIVWGNHGESMVADLTQATFTKEGKTQKVVDVLDHDYVFDTFFKKI\n",
    "GHRAWDILEHRGFTSAASPTKAAIQHMKAWLFGTAPGEVLSMGIPVPEGNPYGIKPGVVFSFPCNVDKEGKIHVV\n",
    "EGFKVNDWLREKLDFTEKDLFHEKEIALNHLAQGG*\n",
    "```\n",
    "The first line contains the sequence code, in the format \">P1;code\". The second line with ten fields separated by colons generally contains information about the structure file, if applicable. Only two of these fields are used for sequences, \"sequence\" (indicating that the file contains a sequence without known structure) and \"TvLDH\" (the model file name). The rest of the file contains the sequence of TvLDH, with \"*\" marking its end. The standard one-letter amino acid codes are used. (Note that they must be upper case; some lower case letters are used for non-standard residues. See the file modlib/restyp.lib in the Modeller distribution for more information.)\n",
    "\n",
    "- Here is decription https://salilab.org/modeller/manual/node496.html#alignmentformat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "record.name='sequence:model_file_name::::::::'\n",
    "record.description=''\n",
    "record.id='humanH1'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">P1;humanH1\r\n",
      "sequence:model_file_name::::::::\r\n",
      "MTENSTSAPAAKPKRAKASKKSTDHPKYSDMIVAAIQAEKNRAGSSRQSIQKYIKSHYKV\r\n",
      "GENADSQIKLSIKRLVTTGVLKQTKGVGASGSFRLAKSDEPKKSVAFKKTKKEIKKVATP\r\n",
      "KKASKPKKAASKAPTKKPKATPVKKAKKKLAATPKKAKKPKTVKAKPVKASKPKKAKPVK\r\n",
      "PKAKSSAKRAGKKK*\r\n"
     ]
    }
   ],
   "source": [
    "handle = open('humanH1.ali', \"w\")\n",
    "writer = SeqIO.PirIO.PirWriter(handle)\n",
    "writer.write_file([record])\n",
    "handle.close()\n",
    "!cat humanH1.ali"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkapsa__637W> No residue topology library is in memory.\n",
      "              Better radii would be used if topology.read() is called first.\n",
      "iup2crm_280W> No topology library in memory or assigning a BLK residue.\n",
      "              Default CHARMM atom type assigned:  N -->  N\n",
      "              This message is written only for the first such atom.\n",
      "\n",
      "Pairwise dynamic programming alignment (ALIGN2D):\n",
      "  Residue-residue metric   : $(LIB)/as1.sim.mat\n",
      "  Diagonal                 :          100\n",
      "  Overhang                 :            0\n",
      "  Maximal gap length       :       999999\n",
      "  Local alignment          :            F\n",
      "  MATRIX_OFFSET (local aln):       0.0000\n",
      "  FIX_OFFSETS              :      0.0    -1.0    -2.0    -3.0    -4.0\n",
      "  N_SUBOPT                 :            0\n",
      "  SUBOPT_OFFSET            :       0.0000\n",
      "  Alignment block          :            1\n",
      "  Gap introduction penalty :    -100.0000\n",
      "  Gap extension penalty    :       0.0000\n",
      "  Gap diagonal penalty     :       0.0000\n",
      "  Structure gap penalties  :    3.500   3.500   3.500   0.200   4.000   6.500   2.000   0.000\n",
      "  Break-break bonus        :   10000.0000\n",
      "  Length of alignment      :          194\n",
      "  Score                    :   60734.6406\n"
     ]
    }
   ],
   "source": [
    "#Now we'll do alignment using Modeller\n",
    "from modeller import *\n",
    "\n",
    "env = environ()\n",
    "aln = alignment(env)\n",
    "mdl = model(env, file='1HST', model_segment=('FIRST:A','LAST:A'))\n",
    "aln.append_model(mdl, align_codes='1HSTA', atom_files='1HST.pdb')\n",
    "aln.append(file='humanH1.ali', align_codes='humanH1')\n",
    "aln.align2d()\n",
    "aln.write(file='humanH1-1HSTA.ali', alignment_format='PIR')\n",
    "aln.write(file='humanH1-1HSTA.pap', alignment_format='PAP')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "END OF TABLE\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "0 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "0 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:    10799     9324\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LYS:OXT  LYS\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      194\n",
      "Number of all, selected real atoms                :     1466    1466\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :     9324    9324\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   125549\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :       -9054.6621\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -9054.662109\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /opt/miniconda3/envs/moldyn/lib/modeller-9.21/modlib/surf5.de\n",
      "Pair library             : /opt/miniconda3/envs/moldyn/lib/modeller-9.21/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    83.783997\n",
      "Sequence length          :          194\n",
      "Compactness              :     0.003771\n",
      "Native energy (pair)     :  -124.838480\n",
      "Native energy (surface)  :   -14.022242\n",
      "Native energy (combined) :    -8.591460\n",
      "Z score (pair)           :    -3.909420\n",
      "Z score (surface)        :    -5.846238\n",
      "Z score (combined)       :    -6.844015\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      194\n",
      "Number of all, selected real atoms                :     1466    1466\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :     9324    9324\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2169\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1033.1530\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    1487       0      0   0.004   0.004      7.1599       1.000\n",
      " 2 Bond angle potential               :    1972       0      2   1.795   1.795      124.68       1.000\n",
      " 3 Stereochemical cosine torsion poten:     809       0     38  55.111  55.111      385.21       1.000\n",
      " 4 Stereochemical improper torsion pot:     495       0      0   1.016   1.016      11.100       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2169       0      0   0.002   0.002     0.58377       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    1114       0      1   0.281   0.281      34.361       1.000\n",
      "10 Distance restraints 2 (N-O)        :    1176       0      5   0.307   0.307      62.692       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     193       0      0   3.001   3.001      20.495       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     158       0      1  79.353  79.353      49.710       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     115       0      1  70.041  70.041      34.290       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      89       0      0  75.155  75.155      57.357       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      62       0      0  82.731  82.731      45.511       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :     951       0      0   0.414   0.414      25.524       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     192     103     11  32.969 129.994      134.84       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     511       0      0   0.751   0.751      39.642       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: humanH1.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   33485.4414\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   4765   2T   3E C   N      14   16 -121.35 -117.80   10.99    0.60  -63.60  176.42   20.94\n",
      "    1          3E   3E N   CA     16   17  126.40  136.80                  -40.30\n",
      "    2   4766   3E   4N C   N      23   25 -137.41 -119.90   22.21    1.19  -63.20 -179.59   19.56\n",
      "    2          4N   4N N   CA     25   26  123.34  137.00                  -41.10\n",
      "    3   4767   4N   5S C   N      31   33   55.08  -64.10  150.68   13.56  -64.10  150.68   13.56\n",
      "    3          5S   5S N   CA     33   34 -127.19  -35.00                  -35.00\n",
      "    4   4768   5S   6T C   N      37   39 -115.00 -124.80   10.02    0.47  -63.20 -176.06   27.04\n",
      "    4          6T   6T N   CA     39   40  141.41  143.50                  -42.10\n",
      "    5   4769   6T   7S C   N      44   46  -65.18  -72.40    8.64    0.70  -64.10  167.84   12.22\n",
      "    5          7S   7S N   CA     46   47  157.16  152.40                  -35.00\n",
      "    6   4770   7S   8A C   N      50   52 -165.09 -134.00   33.84    0.77  -62.50 -170.99   36.42\n",
      "    6          8A   8A N   CA     52   53  160.36  147.00                  -40.90\n",
      "    7   4772   9P  10A C   N      62   64  -68.80  -68.20    0.64    0.05  -62.50  173.68   28.77\n",
      "    7         10A  10A N   CA     64   65  145.54  145.30                  -40.90\n",
      "    8   4773  10A  11A C   N      67   69  -81.61  -68.20   31.96    2.13  -62.50  146.04   24.92\n",
      "    8         11A  11A N   CA     69   70  174.32  145.30                  -40.90\n",
      "    9   4774  11A  12K C   N      72   74  -64.76  -70.20   17.38    1.38  -62.90  162.31   21.18\n",
      "    9         12K  12K N   CA     74   75  156.91  140.40                  -40.80\n",
      "   10   4776  13P  14K C   N      88   90   91.64   56.60  148.13   13.31  -62.90 -153.69   25.64\n",
      "   10         14K  14K N   CA     90   91 -177.47   38.60                  -40.80\n",
      "   11   4777  14K  15R C   N      97   99  -95.45  -72.10   23.39    1.82  -63.00 -178.72   26.73\n",
      "   11         15R  15R N   CA     99  100  140.55  141.90                  -41.10\n",
      "   12   4778  15R  16A C   N     108  110 -179.36 -134.00   53.04    1.25  -62.50 -174.06   36.56\n",
      "   12         16A  16A N   CA    110  111  174.48  147.00                  -40.90\n",
      "   13   4779  16A  17K C   N     113  115  -67.90  -70.20   15.20    1.15  -62.90  163.85   21.55\n",
      "   13         17K  17K N   CA    115  116  155.43  140.40                  -40.80\n",
      "   14   4780  17K  18A C   N     122  124 -122.16 -134.00   14.93    0.83  -62.50  173.58   31.64\n",
      "   14         18A  18A N   CA    124  125  156.09  147.00                  -40.90\n",
      "   15   4781  18A  19S C   N     127  129  -68.51  -72.40   14.76    1.00  -64.10  158.42   11.74\n",
      "   15         19S  19S N   CA    129  130  166.64  152.40                  -35.00\n",
      "   16   4782  19S  20K C   N     133  135  -62.79  -70.20    7.45    0.56  -62.90  178.08   23.12\n",
      "   16         20K  20K N   CA    135  136  141.12  140.40                  -40.80\n",
      "   17   4783  20K  21K C   N     142  144  -75.31  -70.20    6.74    0.40  -62.90  174.85   23.40\n",
      "   17         21K  21K N   CA    144  145  144.79  140.40                  -40.80\n",
      "   18   4784  21K  22S C   N     151  153  -83.83  -72.40   14.22    0.72  -64.10  165.33   13.25\n",
      "   18         22S  22S N   CA    153  154  160.85  152.40                  -35.00\n",
      "   19   4785  22S  23T C   N     157  159 -147.15 -124.80   29.88    1.00  -63.20  175.88   27.97\n",
      "   19         23T  23T N   CA    159  160  163.34  143.50                  -42.10\n",
      "   20   4786  23T  24D C   N     164  166   94.99   54.50   96.88   10.32  -63.30 -128.51   38.26\n",
      "   20         24D  24D N   CA    166  167  128.91   40.90                  -40.00\n",
      "   21   4825  62E  63N C   N     475  477 -101.18 -119.90   92.03    3.72  -63.20   95.84   10.42\n",
      "   21         63N  63N N   CA    477  478   46.89  137.00                  -41.10\n",
      "   22   4859  96A  97K C   N     723  725   67.31  -70.20  140.19    9.58  -62.90 -158.39   33.75\n",
      "   22         97K  97K N   CA    725  726  113.12  140.40                  -40.80\n",
      "   23   4860  97K  98S C   N     732  734 -174.77 -136.60   38.78    1.37  -64.10 -159.69   22.50\n",
      "   23         98S  98S N   CA    734  735  158.04  151.20                  -35.00\n",
      "   24   4861  98S  99D C   N     738  740 -100.80  -96.50   31.28    1.31  -63.30  128.79   14.07\n",
      "   24         99D  99D N   CA    740  741   83.22  114.20                  -40.00\n",
      "   25   4862  99D 100E C   N     746  748 -121.54 -117.80   12.56    0.55  -63.60 -179.54   27.34\n",
      "   25        100E 100E N   CA    748  749  148.80  136.80                  -40.30\n",
      "   26   4864 101P 102K C   N     762  764 -122.93 -118.00   64.92    3.21  -62.90  129.87   14.92\n",
      "   26        102K 102K N   CA    764  765   74.37  139.10                  -40.80\n",
      "   27   4865 102K 103K C   N     771  773 -156.06 -118.00   52.66    1.84  -62.90  171.25   27.77\n",
      "   27        103K 103K N   CA    773  774  175.50  139.10                  -40.80\n",
      "   28   4866 103K 104S C   N     780  782 -111.29 -136.60   38.50    1.29  -64.10  164.12   10.13\n",
      "   28        104S 104S N   CA    782  783  122.19  151.20                  -35.00\n",
      "   29   4868 105V 106A C   N     793  795  -70.64  -68.20    9.35    0.67  -62.50  164.97   27.44\n",
      "   29        106A 106A N   CA    795  796  154.33  145.30                  -40.90\n",
      "   30   4869 106A 107F C   N     798  800  -95.38  -71.40   24.97    1.91  -63.20 -179.08   23.05\n",
      "   30        107F 107F N   CA    800  801  133.73  140.70                  -44.30\n",
      "   31   4870 107F 108K C   N     809  811  -92.00  -70.20   24.12    1.97  -62.90  173.34   21.09\n",
      "   31        108K 108K N   CA    811  812  130.08  140.40                  -40.80\n",
      "   32   4871 108K 109K C   N     818  820 -131.66 -118.00   13.72    0.50  -62.90 -168.34   29.00\n",
      "   32        109K 109K N   CA    820  821  140.30  139.10                  -40.80\n",
      "   33   4872 109K 110T C   N     827  829  -69.33  -78.10   48.14    1.79  -63.20  144.70   18.16\n",
      "   33        110T 110T N   CA    829  830  102.47  149.80                  -42.10\n",
      "   34   4873 110T 111K C   N     834  836 -124.91 -118.00    7.75    0.26  -62.90 -172.85   28.00\n",
      "   34        111K 111K N   CA    836  837  142.62  139.10                  -40.80\n",
      "   35   4874 111K 112K C   N     843  845 -108.26 -118.00   17.93    0.69  -62.90  170.98   20.19\n",
      "   35        112K 112K N   CA    845  846  124.05  139.10                  -40.80\n",
      "   36   4875 112K 113E C   N     852  854 -124.18 -117.80    9.00    0.48  -63.60 -178.82   21.47\n",
      "   36        113E 113E N   CA    854  855  130.45  136.80                  -40.30\n",
      "   37   4877 114I 115K C   N     869  871 -130.89 -118.00   19.67    1.00  -62.90  178.51   20.61\n",
      "   37        115K 115K N   CA    871  872  124.25  139.10                  -40.80\n",
      "   38   4878 115K 116K C   N     878  880  -82.09  -70.20   12.11    0.83  -62.90  177.54   24.15\n",
      "   38        116K 116K N   CA    880  881  142.70  140.40                  -40.80\n",
      "   39   4880 117V 118A C   N     894  896  172.98 -134.00   55.07    1.35  -62.50 -159.45   39.37\n",
      "   39        118A 118A N   CA    896  897  161.89  147.00                  -40.90\n",
      "   40   4881 118A 119T C   N     899  901 -132.60 -124.80   24.19    1.26  -63.20  176.88   19.07\n",
      "   40        119T 119T N   CA    901  902  120.60  143.50                  -42.10\n",
      "   41   4882 119T 120P C   N     906  908  -66.77  -58.70    8.23    1.12  -64.50  179.30   13.50\n",
      "   41        120P 120P N   CA    908  909  -32.09  -30.50                  147.20\n",
      "   42   4884 121K 122K C   N     922  924 -107.80 -118.00   42.09    1.88  -62.90  146.14   17.08\n",
      "   42        122K 122K N   CA    924  925   98.27  139.10                  -40.80\n",
      "   43   4885 122K 123A C   N     931  933  175.35 -134.00   52.35    1.30  -62.50 -159.59   39.24\n",
      "   43        123A 123A N   CA    933  934  160.22  147.00                  -40.90\n",
      "   44   4886 123A 124S C   N     936  938 -177.41 -136.60   41.89    1.43  -64.10 -160.36   22.65\n",
      "   44        124S 124S N   CA    938  939  160.63  151.20                  -35.00\n",
      "   45   4887 124S 125K C   N     942  944 -118.94 -118.00    7.30    0.37  -62.90 -178.47   21.21\n",
      "   45        125K 125K N   CA    944  945  131.87  139.10                  -40.80\n",
      "   46   4888 125K 126P C   N     951  953  -67.85  -58.70   13.26    0.91  -64.50  168.13   12.72\n",
      "   46        126P 126P N   CA    953  954  -20.90  -30.50                  147.20\n",
      "   47   4889 126P 127K C   N     958  960   51.32   56.60    7.04    0.48  -62.90  141.81   24.61\n",
      "   47        127K 127K N   CA    960  961   43.25   38.60                  -40.80\n",
      "   48   4890 127K 128K C   N     967  969 -129.13 -118.00   17.24    0.63  -62.90  179.59   27.28\n",
      "   48        128K 128K N   CA    969  970  152.27  139.10                  -40.80\n",
      "   49   4891 128K 129A C   N     976  978 -116.32 -134.00   18.07    0.75  -62.50  176.73   31.83\n",
      "   49        129A 129A N   CA    978  979  150.77  147.00                  -40.90\n",
      "   50   4893 130A 131S C   N     986  988 -110.00 -136.60   27.21    1.23  -64.10  174.19   15.78\n",
      "   50        131S 131S N   CA    988  989  156.96  151.20                  -35.00\n",
      "   51   4894 131S 132K C   N     992  994  -81.16  -70.20   19.80    1.20  -62.90  163.34   22.25\n",
      "   51        132K 132K N   CA    994  995  156.89  140.40                  -40.80\n",
      "   52   4895 132K 133A C   N    1001 1003  -62.46  -68.20    9.66    0.94  -62.50  166.03   27.21\n",
      "   52        133A 133A N   CA   1003 1004  153.07  145.30                  -40.90\n",
      "   53   4897 134P 135T C   N    1013 1015  -80.24  -78.10    6.47    0.22  -63.20  162.89   22.02\n",
      "   53        135T 135T N   CA   1015 1016  155.91  149.80                  -42.10\n",
      "   54   4898 135T 136K C   N    1020 1022  -75.69  -70.20    8.51    0.71  -62.90  175.16   22.08\n",
      "   54        136K 136K N   CA   1022 1023  133.89  140.40                  -40.80\n",
      "   55   4899 136K 137K C   N    1029 1031  -79.69  -70.20   15.21    0.91  -62.90  167.76   22.74\n",
      "   55        137K 137K N   CA   1031 1032  152.28  140.40                  -40.80\n",
      "   56   4901 138P 139K C   N    1045 1047  -61.75  -70.20   23.83    1.91  -62.90  156.52   20.26\n",
      "   56        139K 139K N   CA   1047 1048  162.68  140.40                  -40.80\n",
      "   57   4903 140A 141T C   N    1059 1061 -113.88 -124.80   21.14    0.66  -63.20  175.00   19.54\n",
      "   57        141T 141T N   CA   1061 1062  125.40  143.50                  -42.10\n",
      "   58   4906 143V 144K C   N    1080 1082 -161.03 -118.00   54.38    1.84  -62.90  176.62   28.75\n",
      "   58        144K 144K N   CA   1082 1083  172.35  139.10                  -40.80\n",
      "   59   4907 144K 145K C   N    1089 1091  -77.12  -70.20   12.11    0.73  -62.90  169.46   22.81\n",
      "   59        145K 145K N   CA   1091 1092  150.33  140.40                  -40.80\n",
      "   60   4908 145K 146A C   N    1098 1100  -63.86  -68.20    8.51    0.55  -62.50  178.89   29.25\n",
      "   60        146A 146A N   CA   1100 1101  137.99  145.30                  -40.90\n",
      "   61   4909 146A 147K C   N    1103 1105  -81.05  -70.20   11.05    0.85  -62.90 -179.99   22.44\n",
      "   61        147K 147K N   CA   1105 1106  138.29  140.40                  -40.80\n",
      "   62   4910 147K 148K C   N    1112 1114  -96.69 -118.00   23.50    0.78  -62.90  173.31   20.89\n",
      "   62        148K 148K N   CA   1114 1115  129.18  139.10                  -40.80\n",
      "   63   4911 148K 149K C   N    1121 1123 -105.99 -118.00   24.43    1.27  -62.90  164.56   23.92\n",
      "   63        149K 149K N   CA   1123 1124  160.38  139.10                  -40.80\n",
      "   64   4912 149K 150L C   N    1130 1132  -62.74  -70.70   15.05    0.93  -63.50  170.03   23.65\n",
      "   64        150L 150L N   CA   1132 1133  128.82  141.60                  -41.20\n",
      "   65   4913 150L 151A C   N    1138 1140 -126.22 -134.00    9.14    0.22  -62.50 -171.96   34.22\n",
      "   65        151A 151A N   CA   1140 1141  142.19  147.00                  -40.90\n",
      "   66   4914 151A 152A C   N    1143 1145  -80.38  -68.20   20.42    1.29  -62.50  158.43   26.88\n",
      "   66        152A 152A N   CA   1145 1146  161.68  145.30                  -40.90\n",
      "   67   4915 152A 153T C   N    1148 1150  -91.05  -78.10   22.56    1.38  -63.20  175.65   20.80\n",
      "   67        153T 153T N   CA   1150 1151  131.33  149.80                  -42.10\n",
      "   68   4916 153T 154P C   N    1155 1157  -67.92  -58.70   13.22    0.92  -64.50  168.26   12.74\n",
      "   68        154P 154P N   CA   1157 1158  -21.02  -30.50                  147.20\n",
      "   69   4917 154P 155K C   N    1162 1164 -100.22 -118.00   19.44    0.88  -62.90  176.22   25.07\n",
      "   69        155K 155K N   CA   1164 1165  146.98  139.10                  -40.80\n",
      "   70   4918 155K 156K C   N    1171 1173 -107.32 -118.00   15.81    0.80  -62.90  174.20   25.25\n",
      "   70        156K 156K N   CA   1173 1174  150.76  139.10                  -40.80\n",
      "   71   4919 156K 157A C   N    1180 1182  -61.07  -68.20    8.52    0.84  -62.50  169.14   27.65\n",
      "   71        157A 157A N   CA   1182 1183  149.97  145.30                  -40.90\n",
      "   72   4920 157A 158K C   N    1185 1187 -119.45 -118.00   17.96    0.85  -62.90  171.77   25.68\n",
      "   72        158K 158K N   CA   1187 1188  157.01  139.10                  -40.80\n",
      "   73   4921 158K 159K C   N    1194 1196 -127.36 -118.00   30.00    1.30  -62.90  164.73   25.25\n",
      "   73        159K 159K N   CA   1196 1197  167.61  139.10                  -40.80\n",
      "   74   4922 159K 160P C   N    1203 1205  -65.50  -58.70    9.42    0.68  -64.50  171.18   12.83\n",
      "   74        160P 160P N   CA   1205 1206  -23.98  -30.50                  147.20\n",
      "   75   4923 160P 161K C   N    1210 1212   55.93   56.60    5.51    0.45  -62.90  139.95   24.41\n",
      "   75        161K 161K N   CA   1212 1213   33.14   38.60                  -40.80\n",
      "   76   4924 161K 162T C   N    1219 1221  -84.69  -78.10   24.76    1.28  -63.20  169.40   20.37\n",
      "   76        162T 162T N   CA   1221 1222  125.93  149.80                  -42.10\n",
      "   77   4926 163V 164K C   N    1233 1235  -99.36 -118.00   36.02    1.41  -62.90  153.47   18.26\n",
      "   77        164K 164K N   CA   1235 1236  108.28  139.10                  -40.80\n",
      "   78   4927 164K 165A C   N    1242 1244 -148.08 -134.00   14.34    0.58  -62.50 -165.35   36.48\n",
      "   78        165A 165A N   CA   1244 1245  144.27  147.00                  -40.90\n",
      "   79   4928 165A 166K C   N    1247 1249 -126.61 -118.00   15.01    0.57  -62.90  179.49   27.12\n",
      "   79        166K 166K N   CA   1249 1250  151.40  139.10                  -40.80\n",
      "   80   4929 166K 167P C   N    1256 1258  -66.90  -58.70   11.14    0.82  -64.50  170.18   12.83\n",
      "   80        167P 167P N   CA   1258 1259  -22.97  -30.50                  147.20\n",
      "   81   4931 168V 169K C   N    1270 1272 -116.01 -118.00    6.09    0.26  -62.90 -177.93   21.36\n",
      "   81        169K 169K N   CA   1272 1273  133.35  139.10                  -40.80\n",
      "   82   4932 169K 170A C   N    1279 1281 -170.20 -134.00   38.43    0.90  -62.50 -167.80   37.20\n",
      "   82        170A 170A N   CA   1281 1282  159.91  147.00                  -40.90\n",
      "   83   4933 170A 171S C   N    1284 1286 -142.64 -136.60   14.46    0.82  -64.10 -169.95   11.92\n",
      "   83        171S 171S N   CA   1286 1287  138.06  151.20                  -35.00\n",
      "   84   4934 171S 172K C   N    1290 1292 -138.41 -118.00   27.68    1.38  -62.90  178.01   20.47\n",
      "   84        172K 172K N   CA   1292 1293  120.40  139.10                  -40.80\n",
      "   85   4935 172K 173P C   N    1299 1301  -65.86  -58.70   11.21    1.48  -64.50  173.69   12.89\n",
      "   85        173P 173P N   CA   1301 1302  -39.12  -30.50                  147.20\n",
      "   86   4936 173P 174K C   N    1306 1308 -141.71 -118.00   25.14    1.10  -62.90 -171.20   21.72\n",
      "   86        174K 174K N   CA   1308 1309  130.77  139.10                  -40.80\n",
      "   87   4937 174K 175K C   N    1315 1317 -106.53 -118.00   18.22    0.67  -62.90  171.39   20.30\n",
      "   87        175K 175K N   CA   1317 1318  124.95  139.10                  -40.80\n",
      "   88   4938 175K 176A C   N    1324 1326 -169.14 -134.00   39.69    0.91  -62.50 -172.96   36.29\n",
      "   88        176A 176A N   CA   1326 1327  165.44  147.00                  -40.90\n",
      "   89   4939 176A 177K C   N    1329 1331  -69.01  -70.20    1.20    0.08  -62.90  179.05   23.59\n",
      "   89        177K 177K N   CA   1331 1332  140.26  140.40                  -40.80\n",
      "   90   4940 177K 178P C   N    1338 1340  -64.57  -58.70    7.41    1.03  -64.50  177.79   13.26\n",
      "   90        178P 178P N   CA   1340 1341  -35.02  -30.50                  147.20\n",
      "   91   4941 178P 179V C   N    1345 1347  -66.11  -62.40    6.84    1.09 -125.40  178.67   10.00\n",
      "   91        179V 179V N   CA   1347 1348  -48.15  -42.40                  143.30\n",
      "   92   4942 179V 180K C   N    1352 1354   49.83   56.60   10.58    0.61  -62.90  142.72   24.70\n",
      "   92        180K 180K N   CA   1354 1355   46.73   38.60                  -40.80\n",
      "   93   4944 181P 182K C   N    1368 1370 -145.47 -118.00   28.45    1.21  -62.90 -168.76   21.98\n",
      "   93        182K 182K N   CA   1370 1371  131.70  139.10                  -40.80\n",
      "   94   4945 182K 183A C   N    1377 1379 -152.06 -134.00   18.07    0.58  -62.50 -166.34   36.53\n",
      "   94        183A 183A N   CA   1379 1380  147.40  147.00                  -40.90\n",
      "   95   4946 183A 184K C   N    1382 1384 -156.14 -118.00   40.46    1.36  -62.90 -169.09   30.36\n",
      "   95        184K 184K N   CA   1384 1385  152.61  139.10                  -40.80\n",
      "   96   4947 184K 185S C   N    1391 1393  -66.17  -72.40    8.72    0.69  -64.10  166.51   12.19\n",
      "   96        185S 185S N   CA   1393 1394  158.50  152.40                  -35.00\n",
      "   97   4948 185S 186S C   N    1397 1399  -66.56  -72.40   15.30    1.10  -64.10  158.47   11.63\n",
      "   97        186S 186S N   CA   1399 1400  166.55  152.40                  -35.00\n",
      "   98   4949 186S 187A C   N    1403 1405  -62.61  -68.20   14.93    1.01  -62.50  172.35   28.24\n",
      "   98        187A 187A N   CA   1405 1406  131.45  145.30                  -40.90\n",
      "   99   4950 187A 188K C   N    1408 1410 -162.47 -118.00   53.61    1.79  -62.90 -179.84   29.30\n",
      "   99        188K 188K N   CA   1410 1411  169.05  139.10                  -40.80\n",
      "  100   4951 188K 189R C   N    1417 1419  -64.51  -72.10   15.13    1.01  -63.00  169.92   22.85\n",
      "  100        189R 189R N   CA   1419 1420  128.81  141.90                  -41.10\n",
      "  101   4952 189R 190A C   N    1428 1430 -160.26 -134.00   38.06    1.18  -62.50  174.51   33.77\n",
      "  101        190A 190A N   CA   1430 1431  174.55  147.00                  -40.90\n",
      "  102   4954 191G 192K C   N    1437 1439 -100.74 -118.00   20.13    0.95  -62.90  173.92   24.81\n",
      "  102        192K 192K N   CA   1439 1440  149.45  139.10                  -40.80\n",
      "  103   4955 192K 193K C   N    1446 1448  -89.30  -70.20   20.49    1.32  -62.90  173.40   24.04\n",
      "  103        193K 193K N   CA   1448 1449  147.82  140.40                  -40.80\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0    3   14   53   62  112   95  132  153  139  162\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LYS:OXT  LYS\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      194\n",
      "Number of all, selected real atoms                :     1466    1466\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :     9324    9324\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   124513\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :       -9417.9326\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -9417.932617\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /opt/miniconda3/envs/moldyn/lib/modeller-9.21/modlib/surf5.de\n",
      "Pair library             : /opt/miniconda3/envs/moldyn/lib/modeller-9.21/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    83.783997\n",
      "Sequence length          :          194\n",
      "Compactness              :     0.003320\n",
      "Native energy (pair)     :  -128.101136\n",
      "Native energy (surface)  :   -12.950911\n",
      "Native energy (combined) :    -8.682799\n",
      "Z score (pair)           :    -4.242465\n",
      "Z score (surface)        :    -5.488369\n",
      "Z score (combined)       :    -6.254914\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      194\n",
      "Number of all, selected real atoms                :     1466    1466\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :     9324    9324\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2118\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :         876.3749\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    1487       0      0   0.004   0.004      5.8252       1.000\n",
      " 2 Bond angle potential               :    1972       0      2   1.577   1.577      98.364       1.000\n",
      " 3 Stereochemical cosine torsion poten:     809       0     25  54.174  54.174      372.08       1.000\n",
      " 4 Stereochemical improper torsion pot:     495       0      0   0.848   0.848      8.3024       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2118       0      0   0.001   0.001     0.47013       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    1114       0      0   0.164   0.164      19.970       1.000\n",
      "10 Distance restraints 2 (N-O)        :    1176       0      1   0.219   0.219      45.978       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     193       0      0   2.372   2.372      12.801       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     158       0      1  74.340  74.340      43.078       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     115       0      0  68.477  68.477      28.716       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      89       0      0  83.212  83.212      54.559       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      62       0      0  87.770  87.770      36.546       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :     951       0      0   0.413   0.413      23.123       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     192      98      9  25.974 125.935      94.466       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     511       0      0   0.670   0.670      32.094       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: humanH1.V99990002\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   31788.5977\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   4765   2T   3E C   N      14   16  -62.37  -69.30    6.96    0.55  -63.60  176.61   23.48\n",
      "    1          3E   3E N   CA     16   17  143.09  142.50                  -40.30\n",
      "    2   4766   3E   4N C   N      23   25  -85.16  -71.20   20.74    1.11  -63.20  162.27   21.52\n",
      "    2          4N   4N N   CA     25   26  158.13  142.80                  -41.10\n",
      "    3   4767   4N   5S C   N      31   33  -74.51  -72.40   25.48    1.43  -64.10  147.57   11.34\n",
      "    3          5S   5S N   CA     33   34  177.79  152.40                  -35.00\n",
      "    4   4768   5S   6T C   N      37   39 -139.53 -124.80   59.57    2.14  -63.20  139.43   22.70\n",
      "    4          6T   6T N   CA     39   40 -158.78  143.50                  -42.10\n",
      "    5   4769   6T   7S C   N      44   46  -63.31  -72.40    9.86    0.78  -64.10  168.79   12.18\n",
      "    5          7S   7S N   CA     46   47  156.21  152.40                  -35.00\n",
      "    6   4770   7S   8A C   N      50   52 -147.84 -134.00   31.31    1.33  -62.50  167.40   31.98\n",
      "    6          8A   8A N   CA     52   53  175.08  147.00                  -40.90\n",
      "    7   4772   9P  10A C   N      62   64 -107.60 -134.00   26.40    0.89  -62.50  177.63   31.50\n",
      "    7         10A  10A N   CA     64   65  147.30  147.00                  -40.90\n",
      "    8   4773  10A  11A C   N      67   69  -69.79  -68.20    4.44    0.30  -62.50  169.81   28.19\n",
      "    8         11A  11A N   CA     69   70  149.45  145.30                  -40.90\n",
      "    9   4774  11A  12K C   N      72   74 -106.69 -118.00   26.49    1.37  -62.90  162.17   23.65\n",
      "    9         12K  12K N   CA     74   75  163.06  139.10                  -40.80\n",
      "   10   4776  13P  14K C   N      88   90 -157.88 -118.00   44.67    1.48  -62.90 -173.97   29.82\n",
      "   10         14K  14K N   CA     90   91  159.24  139.10                  -40.80\n",
      "   11   4778  15R  16A C   N     108  110 -116.80 -134.00   17.25    0.52  -62.50 -178.30   32.67\n",
      "   11         16A  16A N   CA    110  111  145.71  147.00                  -40.90\n",
      "   12   4779  16A  17K C   N     113  115 -117.76 -118.00    6.16    0.30  -62.90 -177.61   26.95\n",
      "   12         17K  17K N   CA    115  116  145.25  139.10                  -40.80\n",
      "   13   4780  17K  18A C   N     122  124  -82.31  -68.20   27.98    1.81  -62.50  150.94   25.75\n",
      "   13         18A  18A N   CA    124  125  169.47  145.30                  -40.90\n",
      "   14   4781  18A  19S C   N     127  129  -68.20  -72.40   15.14    1.03  -64.10  158.11   11.70\n",
      "   14         19S  19S N   CA    129  130  166.94  152.40                  -35.00\n",
      "   15   4782  19S  20K C   N     133  135  -94.23 -118.00   29.70    1.00  -62.90  165.09   19.94\n",
      "   15         20K  20K N   CA    135  136  121.29  139.10                  -40.80\n",
      "   16   4783  20K  21K C   N     142  144 -116.89 -118.00    5.35    0.27  -62.90 -176.99   26.98\n",
      "   16         21K  21K N   CA    144  145  144.34  139.10                  -40.80\n",
      "   17   4784  21K  22S C   N     151  153 -133.15 -136.60    7.88    0.45  -64.10 -179.55   17.98\n",
      "   17         22S  22S N   CA    153  154  158.28  151.20                  -35.00\n",
      "   18   4785  22S  23T C   N     157  159 -108.36 -124.80   17.80    0.75  -63.20 -175.61   20.98\n",
      "   18         23T  23T N   CA    159  160  136.68  143.50                  -42.10\n",
      "   19   4786  23T  24D C   N     164  166  -83.67  -70.90   23.50    0.80  -63.30  151.35   19.97\n",
      "   19         24D  24D N   CA    166  167  170.03  150.30                  -40.00\n",
      "   20   4825  62E  63N C   N     475  477  -97.26  -63.20   86.14    9.37  -63.20   86.14    9.37\n",
      "   20         63N  63N N   CA    477  478   38.03  -41.10                  -41.10\n",
      "   21   4859  96A  97K C   N     723  725   69.40  -70.20  139.77   10.04  -62.90 -141.17   36.18\n",
      "   21         97K  97K N   CA    725  726  133.51  140.40                  -40.80\n",
      "   22   4860  97K  98S C   N     732  734 -136.05 -136.60   17.44    0.90  -64.10  172.13   17.61\n",
      "   22         98S  98S N   CA    734  735  168.62  151.20                  -35.00\n",
      "   23   4861  98S  99D C   N     738  740  -86.84  -96.50   42.52    1.79  -63.30  115.22   12.97\n",
      "   23         99D  99D N   CA    740  741   72.79  114.20                  -40.00\n",
      "   24   4862  99D 100E C   N     746  748 -125.03 -117.80    7.73    0.35  -63.60 -175.12   21.91\n",
      "   24        100E 100E N   CA    748  749  134.07  136.80                  -40.30\n",
      "   25   4864 101P 102K C   N     762  764 -113.87 -118.00   28.56    1.33  -62.90  159.98   18.66\n",
      "   25        102K 102K N   CA    764  765  110.84  139.10                  -40.80\n",
      "   26   4865 102K 103K C   N     771  773 -149.27 -118.00   31.66    1.12  -62.90 -164.77   30.52\n",
      "   26        103K 103K N   CA    773  774  144.12  139.10                  -40.80\n",
      "   27   4866 103K 104S C   N     780  782 -120.29 -136.60   25.16    0.85  -64.10  176.24   10.86\n",
      "   27        104S 104S N   CA    782  783  132.05  151.20                  -35.00\n",
      "   28   4868 105V 106A C   N     793  795  -66.72  -68.20    2.15    0.14  -62.50  175.40   28.95\n",
      "   28        106A 106A N   CA    795  796  143.75  145.30                  -40.90\n",
      "   29   4869 106A 107F C   N     798  800 -122.78 -124.20    8.94    0.45  -63.20  174.09   28.11\n",
      "   29        107F 107F N   CA    800  801  152.12  143.30                  -44.30\n",
      "   30   4870 107F 108K C   N     809  811  -81.12  -70.20   26.73    2.16  -62.90  157.86   19.57\n",
      "   30        108K 108K N   CA    811  812  116.00  140.40                  -40.80\n",
      "   31   4871 108K 109K C   N     818  820 -138.96 -118.00   24.60    1.17  -62.90 -176.47   21.12\n",
      "   31        109K 109K N   CA    820  821  126.23  139.10                  -40.80\n",
      "   32   4872 109K 110T C   N     827  829 -124.85 -124.80   17.04    0.73  -63.20  179.48   19.64\n",
      "   32        110T 110T N   CA    829  830  126.45  143.50                  -42.10\n",
      "   33   4873 110T 111K C   N     834  836 -125.74 -118.00   29.36    1.50  -62.90  164.09   18.94\n",
      "   33        111K 111K N   CA    836  837  110.78  139.10                  -40.80\n",
      "   34   4874 111K 112K C   N     843  845 -169.04 -118.00   51.74    1.82  -62.90 -158.26   32.51\n",
      "   34        112K 112K N   CA    845  846  147.64  139.10                  -40.80\n",
      "   35   4875 112K 113E C   N     852  854 -105.09 -117.80   13.37    0.60  -63.60 -176.51   26.81\n",
      "   35        113E 113E N   CA    854  855  140.96  136.80                  -40.30\n",
      "   36   4877 114I 115K C   N     869  871 -133.36 -118.00   27.95    1.44  -62.90  171.67   19.76\n",
      "   36        115K 115K N   CA    871  872  115.75  139.10                  -40.80\n",
      "   37   4878 115K 116K C   N     878  880 -142.85 -118.00   28.93    1.37  -62.90 -176.58   21.08\n",
      "   37        116K 116K N   CA    880  881  124.28  139.10                  -40.80\n",
      "   38   4880 117V 118A C   N     894  896 -177.23 -134.00   45.01    1.09  -62.50 -163.45   38.25\n",
      "   38        118A 118A N   CA    896  897  159.52  147.00                  -40.90\n",
      "   39   4881 118A 119T C   N     899  901  -86.71  -78.10   11.96    0.77  -63.20  177.96   24.39\n",
      "   39        119T 119T N   CA    901  902  141.50  149.80                  -42.10\n",
      "   40   4882 119T 120P C   N     906  908  -53.65  -58.70    9.84    0.58  -64.50  174.19   13.58\n",
      "   40        120P 120P N   CA    908  909  -38.95  -30.50                  147.20\n",
      "   41   4883 120P 121K C   N     913  915 -142.86 -118.00   24.96    0.98  -62.90 -165.16   22.42\n",
      "   41        121K 121K N   CA    915  916  136.87  139.10                  -40.80\n",
      "   42   4884 121K 122K C   N     922  924 -117.55 -118.00   10.60    0.51  -62.90  177.91   20.80\n",
      "   42        122K 122K N   CA    924  925  128.51  139.10                  -40.80\n",
      "   43   4885 122K 123A C   N     931  933 -157.46 -134.00   25.20    0.58  -62.50 -171.44   35.97\n",
      "   43        123A 123A N   CA    933  934  156.20  147.00                  -40.90\n",
      "   44   4886 123A 124S C   N     936  938 -140.07 -136.60    3.71    0.18  -64.10 -169.13   19.24\n",
      "   44        124S 124S N   CA    938  939  149.90  151.20                  -35.00\n",
      "   45   4887 124S 125K C   N     942  944 -123.84 -118.00   12.83    0.66  -62.90  179.16   20.81\n",
      "   45        125K 125K N   CA    944  945  127.68  139.10                  -40.80\n",
      "   46   4889 126P 127K C   N     958  960 -116.78 -118.00    3.21    0.13  -62.90 -175.04   21.70\n",
      "   46        127K 127K N   CA    960  961  136.14  139.10                  -40.80\n",
      "   47   4890 127K 128K C   N     967  969 -159.68 -118.00   43.15    1.47  -62.90 -165.32   31.05\n",
      "   47        128K 128K N   CA    969  970  150.28  139.10                  -40.80\n",
      "   48   4891 128K 129A C   N     976  978 -111.43 -134.00   28.90    0.77  -62.50  176.75   27.02\n",
      "   48        129A 129A N   CA    978  979  128.94  147.00                  -40.90\n",
      "   49   4892 129A 130A C   N     981  983 -148.92 -134.00   15.19    0.40  -62.50 -169.96   35.77\n",
      "   49        130A 130A N   CA    983  984  149.85  147.00                  -40.90\n",
      "   50   4893 130A 131S C   N     986  988 -160.98 -136.60   30.73    0.95  -64.10 -177.13   20.24\n",
      "   50        131S 131S N   CA    988  989  169.90  151.20                  -35.00\n",
      "   51   4894 131S 132K C   N     992  994  -98.33 -118.00   23.70    0.79  -62.90  170.41   20.46\n",
      "   51        132K 132K N   CA    994  995  125.88  139.10                  -40.80\n",
      "   52   4895 132K 133A C   N    1001 1003  -74.35  -68.20    6.19    0.55  -62.50  174.89   29.26\n",
      "   52        133A 133A N   CA   1003 1004  144.61  145.30                  -40.90\n",
      "   53   4897 134P 135T C   N    1013 1015 -120.71 -124.80    7.35    0.23  -63.20 -171.52   20.92\n",
      "   53        135T 135T N   CA   1015 1016  137.39  143.50                  -42.10\n",
      "   54   4898 135T 136K C   N    1020 1022 -118.82 -118.00    1.39    0.05  -62.90 -172.49   27.68\n",
      "   54        136K 136K N   CA   1022 1023  140.22  139.10                  -40.80\n",
      "   55   4899 136K 137K C   N    1029 1031 -101.51 -118.00   23.32    0.82  -62.90  167.92   20.03\n",
      "   55        137K 137K N   CA   1031 1032  122.62  139.10                  -40.80\n",
      "   56   4901 138P 139K C   N    1045 1047 -118.20 -118.00   25.38    1.23  -62.90  164.12   19.07\n",
      "   56        139K 139K N   CA   1047 1048  113.72  139.10                  -40.80\n",
      "   57   4902 139K 140A C   N    1054 1056 -147.67 -134.00   13.80    0.39  -62.50 -169.67   35.75\n",
      "   57        140A 140A N   CA   1056 1057  148.89  147.00                  -40.90\n",
      "   58   4903 140A 141T C   N    1059 1061 -116.93 -124.80   20.88    0.69  -63.20  174.72   19.37\n",
      "   58        141T 141T N   CA   1061 1062  124.15  143.50                  -42.10\n",
      "   59   4906 143V 144K C   N    1080 1082 -147.55 -118.00   36.36    1.22  -62.90 -179.94   28.44\n",
      "   59        144K 144K N   CA   1082 1083  160.28  139.10                  -40.80\n",
      "   60   4907 144K 145K C   N    1089 1091  -77.57  -70.20   11.47    0.69  -62.90  170.65   22.99\n",
      "   60        145K 145K N   CA   1091 1092  149.18  140.40                  -40.80\n",
      "   61   4908 145K 146A C   N    1098 1100  -63.37  -68.20    7.67    0.48  -62.50  179.76   29.50\n",
      "   61        146A 146A N   CA   1100 1101  139.35  145.30                  -40.90\n",
      "   62   4909 146A 147K C   N    1103 1105 -104.67 -118.00   20.35    0.74  -62.90  169.74   20.15\n",
      "   62        147K 147K N   CA   1105 1106  123.72  139.10                  -40.80\n",
      "   63   4910 147K 148K C   N    1112 1114 -139.71 -118.00   21.73    0.83  -62.90 -165.15   22.46\n",
      "   63        148K 148K N   CA   1114 1115  138.27  139.10                  -40.80\n",
      "   64   4911 148K 149K C   N    1121 1123  -92.75  -70.20   45.89    3.77  -62.90  144.36   17.34\n",
      "   64        149K 149K N   CA   1123 1124  100.44  140.40                  -40.80\n",
      "   65   4912 149K 150L C   N    1130 1132 -123.10 -108.50   14.73    0.68  -63.50 -174.49   23.12\n",
      "   65        150L 150L N   CA   1132 1133  134.47  132.50                  -41.20\n",
      "   66   4913 150L 151A C   N    1138 1140 -115.48 -134.00   19.26    0.47  -62.50 -174.85   33.16\n",
      "   66        151A 151A N   CA   1140 1141  141.70  147.00                  -40.90\n",
      "   67   4914 151A 152A C   N    1143 1145  -70.27  -68.20    3.12    0.31  -62.50  176.31   29.28\n",
      "   67        152A 152A N   CA   1145 1146  142.96  145.30                  -40.90\n",
      "   68   4915 152A 153T C   N    1148 1150 -124.04 -124.80    5.09    0.24  -63.20  179.97   27.11\n",
      "   68        153T 153T N   CA   1150 1151  148.53  143.50                  -42.10\n",
      "   69   4917 154P 155K C   N    1162 1164  -93.03  -70.20   25.49    1.60  -62.90  170.17   23.85\n",
      "   69        155K 155K N   CA   1164 1165  151.72  140.40                  -40.80\n",
      "   70   4918 155K 156K C   N    1171 1173  -66.96  -70.20   13.34    0.89  -62.90  168.30   21.64\n",
      "   70        156K 156K N   CA   1173 1174  127.45  140.40                  -40.80\n",
      "   71   4919 156K 157A C   N    1180 1182 -166.39 -134.00   35.69    0.81  -62.50 -171.63   36.38\n",
      "   71        157A 157A N   CA   1182 1183  161.98  147.00                  -40.90\n",
      "   72   4920 157A 158K C   N    1185 1187 -139.44 -118.00   41.98    1.65  -62.90  163.08   25.75\n",
      "   72        158K 158K N   CA   1187 1188  175.20  139.10                  -40.80\n",
      "   73   4921 158K 159K C   N    1194 1196  -58.86  -70.20   13.04    0.81  -62.90  174.81   22.92\n",
      "   73        159K 159K N   CA   1196 1197  133.96  140.40                  -40.80\n",
      "   74   4922 159K 160P C   N    1203 1205  -68.76  -58.70   10.85    1.11  -64.50  173.67   13.19\n",
      "   74        160P 160P N   CA   1205 1206  -26.42  -30.50                  147.20\n",
      "   75   4924 161K 162T C   N    1219 1221 -153.13 -124.80   31.25    1.27  -63.20 -175.41   29.46\n",
      "   75        162T 162T N   CA   1221 1222  156.70  143.50                  -42.10\n",
      "   76   4926 163V 164K C   N    1233 1235  -85.71  -70.20   54.02    4.25  -62.90  131.45   15.96\n",
      "   76        164K 164K N   CA   1235 1236   88.65  140.40                  -40.80\n",
      "   77   4927 164K 165A C   N    1242 1244  163.42 -134.00   65.43    1.57  -62.50 -156.55   40.27\n",
      "   77        165A 165A N   CA   1244 1245  166.09  147.00                  -40.90\n",
      "   78   4928 165A 166K C   N    1247 1249 -109.26 -118.00    9.89    0.46  -62.90 -178.50   26.31\n",
      "   78        166K 166K N   CA   1249 1250  143.73  139.10                  -40.80\n",
      "   79   4931 168V 169K C   N    1270 1272 -131.66 -118.00   15.82    0.52  -62.90 -174.65   28.18\n",
      "   79        169K 169K N   CA   1272 1273  147.08  139.10                  -40.80\n",
      "   80   4932 169K 170A C   N    1279 1281 -104.58 -134.00   36.28    0.91  -62.50  171.90   26.44\n",
      "   80        170A 170A N   CA   1281 1282  125.77  147.00                  -40.90\n",
      "   81   4933 170A 171S C   N    1284 1286 -137.41 -136.60    2.66    0.12  -64.10 -173.71   18.71\n",
      "   81        171S 171S N   CA   1286 1287  153.73  151.20                  -35.00\n",
      "   82   4934 171S 172K C   N    1290 1292   52.14   56.60   12.84    0.66  -62.90  146.95   25.41\n",
      "   82        172K 172K N   CA   1292 1293   50.64   38.60                  -40.80\n",
      "   83   4935 172K 173P C   N    1299 1301  -54.65  -58.70    4.52    0.64  -64.50  175.97   12.62\n",
      "   83        173P 173P N   CA   1301 1302  -28.49  -30.50                  147.20\n",
      "   84   4936 173P 174K C   N    1306 1308  -72.97  -70.20   12.00    0.80  -62.90  167.43   22.31\n",
      "   84        174K 174K N   CA   1308 1309  152.07  140.40                  -40.80\n",
      "   85   4937 174K 175K C   N    1315 1317  -90.92  -70.20   26.97    1.62  -62.90  163.95   22.92\n",
      "   85        175K 175K N   CA   1317 1318  157.66  140.40                  -40.80\n",
      "   86   4938 175K 176A C   N    1324 1326  -69.43  -68.20    3.89    0.27  -62.50  170.25   28.25\n",
      "   86        176A 176A N   CA   1326 1327  148.99  145.30                  -40.90\n",
      "   87   4939 176A 177K C   N    1329 1331  -82.07  -70.20   28.41    1.79  -62.90  154.18   21.12\n",
      "   87        177K 177K N   CA   1331 1332  166.21  140.40                  -40.80\n",
      "   88   4940 177K 178P C   N    1338 1340  -64.60  -58.70   10.87    0.65  -64.50  168.57   12.58\n",
      "   88        178P 178P N   CA   1340 1341  -21.37  -30.50                  147.20\n",
      "   89   4942 179V 180K C   N    1352 1354 -101.71 -118.00   18.24    0.60  -62.90  176.03   21.05\n",
      "   89        180K 180K N   CA   1354 1355  130.90  139.10                  -40.80\n",
      "   90   4945 182K 183A C   N    1377 1379  -74.61  -68.20    8.52    0.54  -62.50  168.63   28.25\n",
      "   90        183A 183A N   CA   1379 1380  150.91  145.30                  -40.90\n",
      "   91   4946 183A 184K C   N    1382 1384 -133.90 -118.00   22.59    0.80  -62.90  178.76   27.46\n",
      "   91        184K 184K N   CA   1384 1385  155.14  139.10                  -40.80\n",
      "   92   4947 184K 185S C   N    1391 1393 -125.57 -136.60   16.48    0.55  -64.10 -175.51   11.37\n",
      "   92        185S 185S N   CA   1393 1394  138.95  151.20                  -35.00\n",
      "   93   4948 185S 186S C   N    1397 1399  -97.05  -72.40   32.61    1.59  -64.10  154.81   13.44\n",
      "   93        186S 186S N   CA   1399 1400  173.74  152.40                  -35.00\n",
      "   94   4949 186S 187A C   N    1403 1405  -79.69  -68.20   21.94    1.41  -62.50  156.06   26.45\n",
      "   94        187A 187A N   CA   1405 1406  163.99  145.30                  -40.90\n",
      "   95   4950 187A 188K C   N    1408 1410  -66.48  -70.20   18.32    1.41  -62.90  160.90   21.09\n",
      "   95        188K 188K N   CA   1410 1411  158.34  140.40                  -40.80\n",
      "   96   4951 188K 189R C   N    1417 1419 -108.21 -125.20   20.25    1.07  -63.00  173.30   26.54\n",
      "   96        189R 189R N   CA   1419 1420  151.60  140.60                  -41.10\n",
      "   97   4952 189R 190A C   N    1428 1430  -66.91  -68.20    1.30    0.10  -62.50  174.01   28.74\n",
      "   97        190A 190A N   CA   1430 1431  145.14  145.30                  -40.90\n",
      "   98   4955 192K 193K C   N    1446 1448  -68.98  -70.20    4.40    0.29  -62.90  177.07   22.67\n",
      "   98        193K 193K N   CA   1448 1449  136.17  140.40                  -40.80\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0    5   16   44   62  104   91  132  133  131  173\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LYS:OXT  LYS\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      194\n",
      "Number of all, selected real atoms                :     1466    1466\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :     9324    9324\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   120592\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :       -9577.0410\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -9577.041016\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /opt/miniconda3/envs/moldyn/lib/modeller-9.21/modlib/surf5.de\n",
      "Pair library             : /opt/miniconda3/envs/moldyn/lib/modeller-9.21/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    83.783997\n",
      "Sequence length          :          194\n",
      "Compactness              :     0.003507\n",
      "Native energy (pair)     :  -116.767127\n",
      "Native energy (surface)  :   -15.430811\n",
      "Native energy (combined) :    -9.325725\n",
      "Z score (pair)           :    -4.231141\n",
      "Z score (surface)        :    -6.621595\n",
      "Z score (combined)       :    -7.170076\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      194\n",
      "Number of all, selected real atoms                :     1466    1466\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :     9324    9324\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2094\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :         894.5311\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    1487       0      0   0.004   0.004      5.3207       1.000\n",
      " 2 Bond angle potential               :    1972       0      2   1.642   1.642      105.78       1.000\n",
      " 3 Stereochemical cosine torsion poten:     809       0     34  56.082  56.082      391.71       1.000\n",
      " 4 Stereochemical improper torsion pot:     495       0      0   0.854   0.854      8.4892       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2094       0      0   0.001   0.001     0.28048       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    1114       0      0   0.170   0.170      19.435       1.000\n",
      "10 Distance restraints 2 (N-O)        :    1176       0      0   0.194   0.194      41.309       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     193       0      0   2.707   2.707      16.683       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     158       0      1  75.762  75.762      35.205       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     115       0      0  63.936  63.936      19.619       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      89       0      0  78.382  78.382      53.629       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      62       0      0  69.181  69.181      36.552       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :     951       0      0   0.402   0.402      25.166       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     192      98     16  28.999 125.122      112.94       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     511       0      0   0.563   0.563      22.413       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: humanH1.V99990003\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   30993.9238\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   4764   1M   2T C   N       7    9 -112.67 -124.80   28.14    0.90  -63.20  167.67   18.68\n",
      "    1          2T   2T N   CA      9   10  118.11  143.50                  -42.10\n",
      "    2   4766   3E   4N C   N      23   25 -129.51 -119.90   38.05    1.49  -63.20  159.53   23.69\n",
      "    2          4N   4N N   CA     25   26  173.81  137.00                  -41.10\n",
      "    3   4767   4N   5S C   N      31   33  177.94 -136.60   49.74    1.55  -64.10 -166.31   22.54\n",
      "    3          5S   5S N   CA     33   34  171.37  151.20                  -35.00\n",
      "    4   4768   5S   6T C   N      37   39 -104.31  -78.10   47.10    1.56  -63.20  135.37   20.09\n",
      "    4          6T   6T N   CA     39   40 -171.07  149.80                  -42.10\n",
      "    5   4769   6T   7S C   N      44   46  -61.04  -72.40   45.00    2.28  -64.10  143.89   10.61\n",
      "    5          7S   7S N   CA     46   47  108.85  152.40                  -35.00\n",
      "    6   4772   9P  10A C   N      62   64 -161.45 -134.00   29.82    0.68  -62.50 -171.47   36.16\n",
      "    6         10A  10A N   CA     64   65  158.63  147.00                  -40.90\n",
      "    7   4773  10A  11A C   N      67   69 -138.55 -134.00    7.19    0.24  -62.50 -176.91   34.08\n",
      "    7         11A  11A N   CA     69   70  152.56  147.00                  -40.90\n",
      "    8   4774  11A  12K C   N      72   74 -104.52 -118.00   42.86    2.21  -62.90  145.49   21.37\n",
      "    8         12K  12K N   CA     74   75  179.78  139.10                  -40.80\n",
      "    9   4776  13P  14K C   N      88   90   55.85   56.60    8.00    0.52  -62.90  147.43   25.58\n",
      "    9         14K  14K N   CA     90   91   46.57   38.60                  -40.80\n",
      "   10   4777  14K  15R C   N      97   99   61.61   57.30   22.37    1.20  -63.00  137.09   24.90\n",
      "   10         15R  15R N   CA     99  100   16.05   38.00                  -41.10\n",
      "   11   4778  15R  16A C   N     108  110  -74.56  -68.20   10.66    0.67  -62.50  165.69   27.76\n",
      "   11         16A  16A N   CA    110  111  153.85  145.30                  -40.90\n",
      "   12   4779  16A  17K C   N     113  115 -145.88 -118.00   29.42    1.29  -62.90 -170.38   21.79\n",
      "   12         17K  17K N   CA    115  116  129.70  139.10                  -40.80\n",
      "   13   4780  17K  18A C   N     122  124 -125.22 -134.00    9.63    0.22  -62.50 -173.09   33.98\n",
      "   13         18A  18A N   CA    124  125  143.03  147.00                  -40.90\n",
      "   14   4781  18A  19S C   N     127  129 -129.56 -136.60   25.73    1.14  -64.10  174.21   10.80\n",
      "   14         19S  19S N   CA    129  130  126.45  151.20                  -35.00\n",
      "   15   4782  19S  20K C   N     133  135 -136.63 -118.00   52.19    2.70  -62.90  150.45   17.29\n",
      "   15         20K  20K N   CA    135  136   90.35  139.10                  -40.80\n",
      "   16   4783  20K  21K C   N     142  144 -142.09 -118.00   25.55    0.86  -62.90 -171.02   29.28\n",
      "   16         21K  21K N   CA    144  145  147.62  139.10                  -40.80\n",
      "   17   4784  21K  22S C   N     151  153  -73.82  -72.40    4.22    0.20  -64.10  168.91   12.84\n",
      "   17         22S  22S N   CA    153  154  156.37  152.40                  -35.00\n",
      "   18   4785  22S  23T C   N     157  159 -115.71 -124.80   26.53    0.89  -63.20  169.04   18.72\n",
      "   18         23T  23T N   CA    159  160  118.58  143.50                  -42.10\n",
      "   19   4786  23T  24D C   N     164  166 -176.69  -96.50   83.59    3.40  -63.30  172.96   19.05\n",
      "   19         24D  24D N   CA    166  167   90.62  114.20                  -40.00\n",
      "   20   4825  62E  63N C   N     475  477  -99.47  -63.20   91.30    9.92  -63.20   91.30    9.92\n",
      "   20         63N  63N N   CA    477  478   42.68  -41.10                  -41.10\n",
      "   21   4859  96A  97K C   N     723  725   69.07  -70.20  139.41   10.32  -62.90 -142.81   25.44\n",
      "   21         97K  97K N   CA    725  726  146.70  140.40                  -40.80\n",
      "   22   4860  97K  98S C   N     732  734 -169.43 -136.60   67.75    2.59  -64.10  155.60   18.84\n",
      "   22         98S  98S N   CA    734  735 -149.54  151.20                  -35.00\n",
      "   23   4861  98S  99D C   N     738  740  -62.23  -70.90   14.82    1.14  -63.30  157.68   19.40\n",
      "   23         99D  99D N   CA    740  741  162.32  150.30                  -40.00\n",
      "   24   4862  99D 100E C   N     746  748  -56.56  -69.30   20.28    1.14  -63.60  167.16   22.68\n",
      "   24        100E 100E N   CA    748  749  126.71  142.50                  -40.30\n",
      "   25   4863 100E 101P C   N     755  757  -67.72  -58.70    9.50    1.02  -64.50  174.76   13.21\n",
      "   25        101P 101P N   CA    757  758  -27.53  -30.50                  147.20\n",
      "   26   4864 101P 102K C   N     762  764 -111.98 -118.00   62.26    2.94  -62.90  127.74   14.74\n",
      "   26        102K 102K N   CA    764  765   77.14  139.10                  -40.80\n",
      "   27   4865 102K 103K C   N     771  773 -133.78 -118.00   15.84    0.57  -62.90 -167.77   29.20\n",
      "   27        103K 103K N   CA    773  774  140.52  139.10                  -40.80\n",
      "   28   4866 103K 104S C   N     780  782  -75.55  -72.40    4.65    0.22  -64.10  169.57   13.00\n",
      "   28        104S 104S N   CA    782  783  155.82  152.40                  -35.00\n",
      "   29   4868 105V 106A C   N     793  795 -127.33 -134.00   15.78    0.69  -62.50 -174.70   28.01\n",
      "   29        106A 106A N   CA    795  796  132.69  147.00                  -40.90\n",
      "   30   4869 106A 107F C   N     798  800 -155.02 -124.20   35.34    0.94  -63.20 -179.77   30.80\n",
      "   30        107F 107F N   CA    800  801  160.61  143.30                  -44.30\n",
      "   31   4870 107F 108K C   N     809  811 -100.59 -118.00   26.71    0.97  -62.90  164.03   19.56\n",
      "   31        108K 108K N   CA    811  812  118.84  139.10                  -40.80\n",
      "   32   4871 108K 109K C   N     818  820 -116.44 -118.00    6.43    0.29  -62.90 -178.27   21.30\n",
      "   32        109K 109K N   CA    820  821  132.87  139.10                  -40.80\n",
      "   33   4872 109K 110T C   N     827  829 -139.18 -124.80   22.32    0.70  -63.20  174.72   27.36\n",
      "   33        110T 110T N   CA    829  830  160.57  143.50                  -42.10\n",
      "   34   4873 110T 111K C   N     834  836 -111.45 -118.00   24.56    1.26  -62.90  163.79   24.16\n",
      "   34        111K 111K N   CA    836  837  162.77  139.10                  -40.80\n",
      "   35   4874 111K 112K C   N     843  845  -77.02  -70.20   58.92    4.43  -62.90  123.48   15.32\n",
      "   35        112K 112K N   CA    845  846   81.87  140.40                  -40.80\n",
      "   36   4875 112K 113E C   N     852  854 -157.87 -117.80   42.10    1.87  -63.60 -170.66   21.88\n",
      "   36        113E 113E N   CA    854  855  123.90  136.80                  -40.30\n",
      "   37   4877 114I 115K C   N     869  871 -160.54 -118.00   42.89    1.72  -62.90 -160.09   22.98\n",
      "   37        115K 115K N   CA    871  872  133.64  139.10                  -40.80\n",
      "   38   4878 115K 116K C   N     878  880 -126.09 -118.00   13.02    0.67  -62.90 -178.91   21.00\n",
      "   38        116K 116K N   CA    880  881  128.90  139.10                  -40.80\n",
      "   39   4880 117V 118A C   N     894  896 -118.89 -134.00   18.44    0.46  -62.50 -173.93   28.32\n",
      "   39        118A 118A N   CA    896  897  136.42  147.00                  -40.90\n",
      "   40   4881 118A 119T C   N     899  901 -117.82 -124.80   13.63    0.42  -63.20 -177.73   20.27\n",
      "   40        119T 119T N   CA    901  902  131.79  143.50                  -42.10\n",
      "   41   4883 120P 121K C   N     913  915 -118.68 -118.00   57.51    2.80  -62.90  134.51   15.48\n",
      "   41        121K 121K N   CA    915  916   81.60  139.10                  -40.80\n",
      "   42   4884 121K 122K C   N     922  924 -152.98 -118.00   49.40    1.74  -62.90  170.88   27.55\n",
      "   42        122K 122K N   CA    924  925  173.99  139.10                  -40.80\n",
      "   43   4885 122K 123A C   N     931  933 -103.17 -134.00   34.70    0.79  -62.50  176.72   27.27\n",
      "   43        123A 123A N   CA    933  934  131.08  147.00                  -40.90\n",
      "   44   4886 123A 124S C   N     936  938  -72.34  -72.40    0.86    0.05  -64.10  173.66   13.09\n",
      "   44        124S 124S N   CA    938  939  151.54  152.40                  -35.00\n",
      "   45   4887 124S 125K C   N     942  944 -142.91 -118.00   31.83    1.08  -62.90  179.13   28.05\n",
      "   45        125K 125K N   CA    944  945  158.92  139.10                  -40.80\n",
      "   46   4889 126P 127K C   N     958  960  -74.55  -70.20   31.63    2.39  -62.90  150.33   18.91\n",
      "   46        127K 127K N   CA    960  961  109.07  140.40                  -40.80\n",
      "   47   4890 127K 128K C   N     967  969 -140.33 -118.00   23.35    0.79  -62.90 -170.22   29.28\n",
      "   47        128K 128K N   CA    969  970  145.94  139.10                  -40.80\n",
      "   48   4891 128K 129A C   N     976  978  -76.34  -68.20   19.32    1.29  -62.50  156.89   26.41\n",
      "   48        129A 129A N   CA    978  979  162.82  145.30                  -40.90\n",
      "   49   4893 130A 131S C   N     986  988   80.79 -136.60  143.29    5.33  -64.10 -144.29   16.75\n",
      "   49        131S 131S N   CA    988  989  165.20  151.20                  -35.00\n",
      "   50   4894 131S 132K C   N     992  994  -84.13  -70.20   14.51    0.96  -62.90  176.03   24.07\n",
      "   50        132K 132K N   CA    994  995  144.45  140.40                  -40.80\n",
      "   51   4895 132K 133A C   N    1001 1003  -75.80  -68.20   10.41    0.66  -62.50  167.22   28.08\n",
      "   51        133A 133A N   CA   1003 1004  152.42  145.30                  -40.90\n",
      "   52   4897 134P 135T C   N    1013 1015 -128.35 -124.80   13.39    0.47  -63.20  174.14   26.62\n",
      "   52        135T 135T N   CA   1015 1016  156.41  143.50                  -42.10\n",
      "   53   4898 135T 136K C   N    1020 1022 -101.33 -118.00   16.68    0.62  -62.90 -176.46   22.03\n",
      "   53        136K 136K N   CA   1022 1023  138.67  139.10                  -40.80\n",
      "   54   4899 136K 137K C   N    1029 1031  -95.45 -118.00   25.57    1.19  -62.90  171.17   24.13\n",
      "   54        137K 137K N   CA   1031 1032  151.15  139.10                  -40.80\n",
      "   55   4901 138P 139K C   N    1045 1047  -79.39  -70.20   10.06    0.82  -62.90  177.86   22.24\n",
      "   55        139K 139K N   CA   1047 1048  136.29  140.40                  -40.80\n",
      "   56   4902 139K 140A C   N    1054 1056 -130.05 -134.00    8.56    0.36  -62.50 -168.02   35.07\n",
      "   56        140A 140A N   CA   1056 1057  139.40  147.00                  -40.90\n",
      "   57   4903 140A 141T C   N    1059 1061 -134.81 -124.80   11.16    0.67  -63.20 -166.89   29.48\n",
      "   57        141T 141T N   CA   1061 1062  138.56  143.50                  -42.10\n",
      "   58   4906 143V 144K C   N    1080 1082 -114.34 -118.00   15.96    0.72  -62.90  172.23   20.17\n",
      "   58        144K 144K N   CA   1082 1083  123.57  139.10                  -40.80\n",
      "   59   4907 144K 145K C   N    1089 1091 -139.59 -118.00   24.72    0.82  -62.90 -175.26   28.58\n",
      "   59        145K 145K N   CA   1091 1092  151.13  139.10                  -40.80\n",
      "   60   4908 145K 146A C   N    1098 1100 -127.21 -134.00    7.18    0.17  -62.50 -173.97   33.95\n",
      "   60        146A 146A N   CA   1100 1101  144.68  147.00                  -40.90\n",
      "   61   4909 146A 147K C   N    1103 1105  -75.57  -70.20    8.97    0.75  -62.90  174.48   21.99\n",
      "   61        147K 147K N   CA   1105 1106  133.22  140.40                  -40.80\n",
      "   62   4910 147K 148K C   N    1112 1114 -121.64 -118.00   34.16    1.70  -62.90  157.32   18.18\n",
      "   62        148K 148K N   CA   1114 1115  105.14  139.10                  -40.80\n",
      "   63   4911 148K 149K C   N    1121 1123 -142.74 -118.00   27.46    1.26  -62.90 -174.00   21.38\n",
      "   63        149K 149K N   CA   1123 1124  127.19  139.10                  -40.80\n",
      "   64   4912 149K 150L C   N    1130 1132 -129.32 -108.50   32.34    1.83  -63.50  162.85   20.03\n",
      "   64        150L 150L N   CA   1132 1133  107.76  132.50                  -41.20\n",
      "   65   4913 150L 151A C   N    1138 1140 -137.17 -134.00    9.51    0.58  -62.50 -166.12   29.19\n",
      "   65        151A 151A N   CA   1140 1141  138.03  147.00                  -40.90\n",
      "   66   4914 151A 152A C   N    1143 1145  178.65 -134.00   52.43    1.19  -62.50 -168.93   37.52\n",
      "   66        152A 152A N   CA   1145 1146  169.50  147.00                  -40.90\n",
      "   67   4915 152A 153T C   N    1148 1150 -119.46 -124.80    9.19    0.53  -63.20  176.15   26.32\n",
      "   67        153T 153T N   CA   1150 1151  150.97  143.50                  -42.10\n",
      "   68   4916 153T 154P C   N    1155 1157  -65.31  -58.70    6.65    0.89  -64.50  178.44   13.36\n",
      "   68        154P 154P N   CA   1157 1158  -31.24  -30.50                  147.20\n",
      "   69   4917 154P 155K C   N    1162 1164   50.17   56.60    8.30    0.59  -62.90  141.24   24.49\n",
      "   69        155K 155K N   CA   1164 1165   43.85   38.60                  -40.80\n",
      "   70   4918 155K 156K C   N    1171 1173  -65.49  -70.20    7.06    0.42  -62.90  175.96   22.71\n",
      "   70        156K 156K N   CA   1173 1174  135.14  140.40                  -40.80\n",
      "   71   4919 156K 157A C   N    1180 1182 -138.09 -134.00    6.64    0.23  -62.50 -176.81   34.07\n",
      "   71        157A 157A N   CA   1182 1183  152.23  147.00                  -40.90\n",
      "   72   4920 157A 158K C   N    1185 1187 -110.43 -118.00   17.45    0.90  -62.90  171.11   25.04\n",
      "   72        158K 158K N   CA   1187 1188  154.83  139.10                  -40.80\n",
      "   73   4921 158K 159K C   N    1194 1196  -73.89  -70.20   19.20    1.30  -62.90  160.34   21.44\n",
      "   73        159K 159K N   CA   1196 1197  159.24  140.40                  -40.80\n",
      "   74   4923 160P 161K C   N    1210 1212  -99.70 -118.00   18.87    0.80  -62.90  179.34   25.44\n",
      "   74        161K 161K N   CA   1212 1213  143.68  139.10                  -40.80\n",
      "   75   4924 161K 162T C   N    1219 1221 -124.91 -124.80   10.00    0.43  -63.20  175.61   26.60\n",
      "   75        162T 162T N   CA   1221 1222  153.49  143.50                  -42.10\n",
      "   76   4926 163V 164K C   N    1233 1235 -107.07 -118.00   11.83    0.39  -62.90 -179.15   21.48\n",
      "   76        164K 164K N   CA   1235 1236  134.57  139.10                  -40.80\n",
      "   77   4927 164K 165A C   N    1242 1244 -141.49 -134.00   11.67    0.39  -62.50 -178.72   33.94\n",
      "   77        165A 165A N   CA   1244 1245  155.94  147.00                  -40.90\n",
      "   78   4928 165A 166K C   N    1247 1249  -84.21  -70.20   20.42    1.71  -62.90  167.71   20.70\n",
      "   78        166K 166K N   CA   1249 1250  125.55  140.40                  -40.80\n",
      "   79   4931 168V 169K C   N    1270 1272  -82.13  -70.20   45.03    3.53  -62.90  139.11   17.10\n",
      "   79        169K 169K N   CA   1272 1273   96.98  140.40                  -40.80\n",
      "   80   4932 169K 170A C   N    1279 1281 -158.33 -134.00   24.73    0.99  -62.50 -159.15   38.04\n",
      "   80        170A 170A N   CA   1281 1282  142.59  147.00                  -40.90\n",
      "   81   4933 170A 171S C   N    1284 1286 -143.89 -136.60    9.20    0.50  -64.10 -163.65   19.91\n",
      "   81        171S 171S N   CA   1286 1287  145.59  151.20                  -35.00\n",
      "   82   4934 171S 172K C   N    1290 1292 -133.16 -118.00   15.45    0.54  -62.90 -169.45   28.95\n",
      "   82        172K 172K N   CA   1292 1293  142.08  139.10                  -40.80\n",
      "   83   4936 173P 174K C   N    1306 1308 -109.32 -118.00    9.03    0.39  -62.90 -176.43   26.58\n",
      "   83        174K 174K N   CA   1308 1309  141.60  139.10                  -40.80\n",
      "   84   4937 174K 175K C   N    1315 1317 -122.77 -118.00   21.73    0.98  -62.90  169.81   25.63\n",
      "   84        175K 175K N   CA   1317 1318  160.30  139.10                  -40.80\n",
      "   85   4938 175K 176A C   N    1324 1326  -72.39  -68.20   17.18    1.24  -62.50  157.45   26.30\n",
      "   85        176A 176A N   CA   1326 1327  161.96  145.30                  -40.90\n",
      "   86   4939 176A 177K C   N    1329 1331  -58.50  -70.20   16.59    1.39  -62.90  167.09   21.46\n",
      "   86        177K 177K N   CA   1331 1332  152.17  140.40                  -40.80\n",
      "   87   4942 179V 180K C   N    1352 1354  -72.59  -70.20    8.14    0.53  -62.90  171.29   22.79\n",
      "   87        180K 180K N   CA   1354 1355  148.18  140.40                  -40.80\n",
      "   88   4944 181P 182K C   N    1368 1370 -144.54 -118.00   59.85    2.44  -62.90  150.52   24.39\n",
      "   88        182K 182K N   CA   1370 1371 -167.25  139.10                  -40.80\n",
      "   89   4946 183A 184K C   N    1382 1384   52.20   56.60   19.27    1.08  -62.90  151.27   26.04\n",
      "   89        184K 184K N   CA   1384 1385   57.36   38.60                  -40.80\n",
      "   90   4947 184K 185S C   N    1391 1393  -61.50  -72.40   14.62    0.71  -64.10  177.67   13.03\n",
      "   90        185S 185S N   CA   1393 1394  142.65  152.40                  -35.00\n",
      "   91   4948 185S 186S C   N    1397 1399 -137.36 -136.60    5.48    0.26  -64.10 -176.38   18.52\n",
      "   91        186S 186S N   CA   1399 1400  156.62  151.20                  -35.00\n",
      "   92   4949 186S 187A C   N    1403 1405 -139.07 -134.00   12.59    0.56  -62.50  177.90   33.26\n",
      "   92        187A 187A N   CA   1405 1406  158.52  147.00                  -40.90\n",
      "   93   4950 187A 188K C   N    1408 1410  -64.05  -70.20    9.20    0.55  -62.90  174.36   22.58\n",
      "   93        188K 188K N   CA   1410 1411  133.56  140.40                  -40.80\n",
      "   94   4951 188K 189R C   N    1417 1419  -81.61  -72.10    9.51    0.73  -63.00  177.97   25.31\n",
      "   94        189R 189R N   CA   1419 1420  141.91  141.90                  -41.10\n",
      "   95   4952 189R 190A C   N    1428 1430  -78.58  -68.20   12.04    0.81  -62.50  168.47   28.43\n",
      "   95        190A 190A N   CA   1430 1431  151.40  145.30                  -40.90\n",
      "   96   4953 190A 191G C   N    1433 1435  -83.06  -80.20    3.05    0.22   82.20 -126.82    7.31\n",
      "   96        191G 191G N   CA   1435 1436  173.01  174.10                    8.50\n",
      "   97   4954 191G 192K C   N    1437 1439  -74.76  -70.20    9.97    0.62  -62.90  170.34   22.79\n",
      "   97        192K 192K N   CA   1439 1440  149.27  140.40                  -40.80\n",
      "   98   4955 192K 193K C   N    1446 1448  -67.07  -70.20    4.34    0.26  -62.90  178.25   22.92\n",
      "   98        193K 193K N   CA   1448 1449  137.40  140.40                  -40.80\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0    6   15   41   66   83  107  106  137  139  176\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LYS:OXT  LYS\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      194\n",
      "Number of all, selected real atoms                :     1466    1466\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :     9324    9324\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   119819\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :       -9488.5547\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -9488.554688\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /opt/miniconda3/envs/moldyn/lib/modeller-9.21/modlib/surf5.de\n",
      "Pair library             : /opt/miniconda3/envs/moldyn/lib/modeller-9.21/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    83.783997\n",
      "Sequence length          :          194\n",
      "Compactness              :     0.002982\n",
      "Native energy (pair)     :  -126.684586\n",
      "Native energy (surface)  :   -14.011874\n",
      "Native energy (combined) :    -9.036022\n",
      "Z score (pair)           :    -3.846072\n",
      "Z score (surface)        :    -5.924415\n",
      "Z score (combined)       :    -6.393249\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      194\n",
      "Number of all, selected real atoms                :     1466    1466\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :     9324    9324\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2082\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :         917.3951\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    1487       0      0   0.004   0.004      5.8857       1.000\n",
      " 2 Bond angle potential               :    1972       0      2   1.654   1.654      108.34       1.000\n",
      " 3 Stereochemical cosine torsion poten:     809       0     30  54.877  54.877      377.84       1.000\n",
      " 4 Stereochemical improper torsion pot:     495       0      0   0.981   0.981      10.311       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2082       0      0   0.002   0.002     0.77390       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    1114       0      0   0.187   0.187      21.844       1.000\n",
      "10 Distance restraints 2 (N-O)        :    1176       0      0   0.228   0.228      47.856       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     193       0      0   2.892   2.892      19.039       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     158       0      1  82.331  82.331      48.888       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     115       0      0  63.638  63.638      23.152       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      89       0      0  79.240  79.240      51.494       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      62       0      0  72.640  72.640      33.839       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :     951       0      0   0.431   0.431      28.893       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     192      99     10  26.137 124.525      100.67       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     511       0      0   0.728   0.728      38.567       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: humanH1.V99990004\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   32001.6934\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   4764   1M   2T C   N       7    9  -80.25  -78.10   21.92    0.87  -63.20  147.28   20.02\n",
      "    1          2T   2T N   CA      9   10  171.61  149.80                  -42.10\n",
      "    2   4765   2T   3E C   N      14   16 -115.18 -117.80   14.79    0.69  -63.60  170.53   20.37\n",
      "    2          3E   3E N   CA     16   17  122.24  136.80                  -40.30\n",
      "    3   4766   3E   4N C   N      23   25 -135.14 -119.90   18.98    1.01  -63.20 -178.35   19.75\n",
      "    3          4N   4N N   CA     25   26  125.70  137.00                  -41.10\n",
      "    4   4767   4N   5S C   N      31   33 -146.78 -136.60   44.93    2.04  -64.10  154.10   17.13\n",
      "    4          5S   5S N   CA     33   34 -165.04  151.20                  -35.00\n",
      "    5   4768   5S   6T C   N      37   39 -139.93 -124.80   47.20    1.61  -63.20  150.70   24.23\n",
      "    5          6T   6T N   CA     39   40 -171.80  143.50                  -42.10\n",
      "    6   4769   6T   7S C   N      44   46  -62.46  -72.40   15.54    0.72  -64.10  175.47   12.81\n",
      "    6          7S   7S N   CA     46   47  140.46  152.40                  -35.00\n",
      "    7   4770   7S   8A C   N      50   52  -80.30  -68.20   16.82    1.06  -62.50  163.10   27.64\n",
      "    7          8A   8A N   CA     52   53  156.98  145.30                  -40.90\n",
      "    8   4772   9P  10A C   N      62   64  -75.34  -68.20   14.08    0.91  -62.50  162.17   27.23\n",
      "    8         10A  10A N   CA     64   65  157.44  145.30                  -40.90\n",
      "    9   4773  10A  11A C   N      67   69 -117.63 -134.00   20.82    1.16  -62.50  168.52   30.56\n",
      "    9         11A  11A N   CA     69   70  159.86  147.00                  -40.90\n",
      "   10   4774  11A  12K C   N      72   74  -67.00  -70.20    9.26    0.74  -62.90  170.16   22.32\n",
      "   10         12K  12K N   CA     74   75  149.09  140.40                  -40.80\n",
      "   11   4776  13P  14K C   N      88   90   52.18   56.60   11.68    0.59  -62.90  146.22   25.30\n",
      "   11         14K  14K N   CA     90   91   49.41   38.60                  -40.80\n",
      "   12   4777  14K  15R C   N      97   99 -144.44 -125.20   49.30    1.89  -63.00  155.87   26.50\n",
      "   12         15R  15R N   CA     99  100 -174.01  140.60                  -41.10\n",
      "   13   4779  16A  17K C   N     113  115   59.60   56.60   10.69    0.57  -62.90  140.66   24.57\n",
      "   13         17K  17K N   CA    115  116   28.34   38.60                  -40.80\n",
      "   14   4780  17K  18A C   N     122  124  -67.72  -68.20    3.73    0.28  -62.50  177.58   29.36\n",
      "   14         18A  18A N   CA    124  125  141.60  145.30                  -40.90\n",
      "   15   4781  18A  19S C   N     127  129 -174.56 -136.60   40.62    1.29  -64.10 -166.11   22.02\n",
      "   15         19S  19S N   CA    129  130  165.65  151.20                  -35.00\n",
      "   16   4782  19S  20K C   N     133  135 -117.81 -118.00   20.66    1.01  -62.90  168.63   25.17\n",
      "   16         20K  20K N   CA    135  136  159.76  139.10                  -40.80\n",
      "   17   4783  20K  21K C   N     142  144  -63.42  -70.20   26.59    1.77  -62.90  155.49   20.16\n",
      "   17         21K  21K N   CA    144  145  114.69  140.40                  -40.80\n",
      "   18   4784  21K  22S C   N     151  153  -68.77  -72.40   10.66    0.52  -64.10  177.44   12.59\n",
      "   18         22S  22S N   CA    153  154  142.38  152.40                  -35.00\n",
      "   19   4785  22S  23T C   N     157  159 -112.15 -124.80   21.85    0.68  -63.20  174.78   19.59\n",
      "   19         23T  23T N   CA    159  160  125.68  143.50                  -42.10\n",
      "   20   4786  23T  24D C   N     164  166 -172.94  -96.50   79.24    3.22  -63.30  172.61   18.86\n",
      "   20         24D  24D N   CA    166  167   93.32  114.20                  -40.00\n",
      "   21   4825  62E  63N C   N     475  477 -101.73 -119.90   91.68    3.72  -63.20   96.28   10.46\n",
      "   21         63N  63N N   CA    477  478   47.14  137.00                  -41.10\n",
      "   22   4859  96A  97K C   N     723  725   71.48  -70.20  144.81    9.86  -62.90 -157.69   34.04\n",
      "   22         97K  97K N   CA    725  726  110.44  140.40                  -40.80\n",
      "   23   4860  97K  98S C   N     732  734 -143.28 -136.60   23.39    1.03  -64.10  170.84   18.06\n",
      "   23         98S  98S N   CA    734  735  173.61  151.20                  -35.00\n",
      "   24   4861  98S  99D C   N     738  740  -64.73  -96.50   32.67    1.34  -63.30  146.57   18.01\n",
      "   24         99D  99D N   CA    740  741  106.56  114.20                  -40.00\n",
      "   25   4862  99D 100E C   N     746  748 -100.61 -117.80   18.27    0.55  -63.60  174.86   21.49\n",
      "   25        100E 100E N   CA    748  749  130.60  136.80                  -40.30\n",
      "   26   4864 101P 102K C   N     762  764  -92.38  -70.20   74.51    4.88  -62.90  111.63   16.24\n",
      "   26        102K 102K N   CA    764  765 -148.47  140.40                  -40.80\n",
      "   27   4865 102K 103K C   N     771  773  -61.76  -70.20   22.88    1.46  -62.90  159.94   20.83\n",
      "   27        103K 103K N   CA    773  774  119.14  140.40                  -40.80\n",
      "   28   4866 103K 104S C   N     780  782  -72.20  -72.40   12.36    0.72  -64.10  175.23   12.24\n",
      "   28        104S 104S N   CA    782  783  140.04  152.40                  -35.00\n",
      "   29   4868 105V 106A C   N     793  795 -114.53 -134.00   20.54    0.48  -62.50 -173.92   33.26\n",
      "   29        106A 106A N   CA    795  796  140.44  147.00                  -40.90\n",
      "   30   4869 106A 107F C   N     798  800 -107.33 -124.20   19.09    0.51  -63.20 -175.96   22.78\n",
      "   30        107F 107F N   CA    800  801  134.37  143.30                  -44.30\n",
      "   31   4870 107F 108K C   N     809  811  -95.00 -118.00   41.37    1.59  -62.90  149.02   17.85\n",
      "   31        108K 108K N   CA    811  812  104.72  139.10                  -40.80\n",
      "   32   4871 108K 109K C   N     818  820 -145.47 -118.00   29.26    1.29  -62.90 -171.17   21.70\n",
      "   32        109K 109K N   CA    820  821  129.03  139.10                  -40.80\n",
      "   33   4872 109K 110T C   N     827  829 -124.09 -124.80    4.75    0.22  -63.20 -179.70   27.15\n",
      "   33        110T 110T N   CA    829  830  148.19  143.50                  -42.10\n",
      "   34   4873 110T 111K C   N     834  836  -95.79 -118.00   90.85    4.06  -62.90   97.52   11.33\n",
      "   34        111K 111K N   CA    836  837   51.01  139.10                  -40.80\n",
      "   35   4874 111K 112K C   N     843  845 -145.51 -118.00   28.69    0.97  -62.90 -169.22   29.72\n",
      "   35        112K 112K N   CA    845  846  147.23  139.10                  -40.80\n",
      "   36   4875 112K 113E C   N     852  854 -111.98 -117.80   15.09    0.62  -63.60  170.20   20.44\n",
      "   36        113E 113E N   CA    854  855  122.87  136.80                  -40.30\n",
      "   37   4877 114I 115K C   N     869  871 -154.09 -118.00   38.65    1.72  -62.90 -170.55   21.77\n",
      "   37        115K 115K N   CA    871  872  125.27  139.10                  -40.80\n",
      "   38   4878 115K 116K C   N     878  880  -85.35  -70.20   15.70    1.05  -62.90  176.12   24.16\n",
      "   38        116K 116K N   CA    880  881  144.51  140.40                  -40.80\n",
      "   39   4880 117V 118A C   N     894  896 -140.63 -134.00    7.71    0.18  -62.50 -174.57   34.57\n",
      "   39        118A 118A N   CA    896  897  150.94  147.00                  -40.90\n",
      "   40   4881 118A 119T C   N     899  901 -117.18 -124.80   10.23    0.34  -63.20 -173.25   20.86\n",
      "   40        119T 119T N   CA    901  902  136.67  143.50                  -42.10\n",
      "   41   4883 120P 121K C   N     913  915 -116.90 -118.00   27.53    1.32  -62.90  161.68   18.80\n",
      "   41        121K 121K N   CA    915  916  111.60  139.10                  -40.80\n",
      "   42   4884 121K 122K C   N     922  924 -126.19 -118.00   16.47    0.85  -62.90  177.29   20.53\n",
      "   42        122K 122K N   CA    924  925  124.81  139.10                  -40.80\n",
      "   43   4885 122K 123A C   N     931  933 -129.31 -134.00   12.35    0.76  -62.50  174.02   32.10\n",
      "   43        123A 123A N   CA    933  934  158.43  147.00                  -40.90\n",
      "   44   4886 123A 124S C   N     936  938 -123.42 -136.60   13.83    0.45  -64.10 -172.39   17.74\n",
      "   44        124S 124S N   CA    938  939  147.01  151.20                  -35.00\n",
      "   45   4887 124S 125K C   N     942  944  -78.64  -70.20   14.68    0.89  -62.90  167.53   22.65\n",
      "   45        125K 125K N   CA    944  945  152.41  140.40                  -40.80\n",
      "   46   4889 126P 127K C   N     958  960 -108.75 -118.00   10.47    0.35  -62.90 -179.11   21.43\n",
      "   46        127K 127K N   CA    960  961  134.18  139.10                  -40.80\n",
      "   47   4890 127K 128K C   N     967  969   57.35   56.60    7.93    0.51  -62.90  139.90   24.42\n",
      "   47        128K 128K N   CA    969  970   30.71   38.60                  -40.80\n",
      "   48   4891 128K 129A C   N     976  978  -71.90  -68.20    4.20    0.41  -62.50  176.03   29.32\n",
      "   48        129A 129A N   CA    978  979  143.32  145.30                  -40.90\n",
      "   49   4892 129A 130A C   N     981  983 -136.82 -134.00    4.29    0.26  -62.50 -169.56   35.19\n",
      "   49        130A 130A N   CA    983  984  143.77  147.00                  -40.90\n",
      "   50   4893 130A 131S C   N     986  988  -89.55  -72.40   26.63    1.23  -64.10  154.33   12.87\n",
      "   50        131S 131S N   CA    988  989  172.78  152.40                  -35.00\n",
      "   51   4894 131S 132K C   N     992  994  -98.47 -118.00   21.92    0.73  -62.90  173.64   20.87\n",
      "   51        132K 132K N   CA    994  995  129.16  139.10                  -40.80\n",
      "   52   4895 132K 133A C   N    1001 1003 -136.14 -134.00    2.68    0.07  -62.50 -174.28   34.38\n",
      "   52        133A 133A N   CA   1003 1004  148.61  147.00                  -40.90\n",
      "   53   4897 134P 135T C   N    1013 1015 -105.48 -124.80   22.91    0.85  -63.20  178.37   20.36\n",
      "   53        135T 135T N   CA   1015 1016  131.19  143.50                  -42.10\n",
      "   54   4898 135T 136K C   N    1020 1022 -129.24 -118.00   13.02    0.43  -62.90 -174.23   28.09\n",
      "   54        136K 136K N   CA   1022 1023  145.68  139.10                  -40.80\n",
      "   55   4899 136K 137K C   N    1029 1031  -91.35  -70.20   24.48    1.51  -62.90  168.88   23.58\n",
      "   55        137K 137K N   CA   1031 1032  152.73  140.40                  -40.80\n",
      "   56   4901 138P 139K C   N    1045 1047 -128.39 -118.00   15.72    0.57  -62.90 -179.41   27.37\n",
      "   56        139K 139K N   CA   1047 1048  150.90  139.10                  -40.80\n",
      "   57   4902 139K 140A C   N    1054 1056  -81.05  -68.20   24.61    1.58  -62.50  153.94   26.18\n",
      "   57        140A 140A N   CA   1056 1057  166.29  145.30                  -40.90\n",
      "   58   4903 140A 141T C   N    1059 1061  -76.55  -78.10    9.39    0.45  -63.20  159.40   21.33\n",
      "   58        141T 141T N   CA   1061 1062  159.06  149.80                  -42.10\n",
      "   59   4904 141T 142P C   N    1066 1068  -67.93  -58.70   12.56    0.92  -64.50  169.21   12.81\n",
      "   59        142P 142P N   CA   1068 1069  -21.98  -30.50                  147.20\n",
      "   60   4906 143V 144K C   N    1080 1082 -120.27 -118.00    4.24    0.16  -62.90 -174.40   27.52\n",
      "   60        144K 144K N   CA   1082 1083  142.69  139.10                  -40.80\n",
      "   61   4907 144K 145K C   N    1089 1091 -130.41 -118.00   16.33    0.56  -62.90 -177.56   27.73\n",
      "   61        145K 145K N   CA   1091 1092  149.71  139.10                  -40.80\n",
      "   62   4908 145K 146A C   N    1098 1100 -129.38 -134.00    8.19    0.30  -62.50 -169.03   34.87\n",
      "   62        146A 146A N   CA   1100 1101  140.23  147.00                  -40.90\n",
      "   63   4909 146A 147K C   N    1103 1105 -124.00 -118.00   33.65    1.70  -62.90  159.00   18.35\n",
      "   63        147K 147K N   CA   1105 1106  105.99  139.10                  -40.80\n",
      "   64   4910 147K 148K C   N    1112 1114 -135.11 -118.00   20.19    0.67  -62.90 -175.87   28.23\n",
      "   64        148K 148K N   CA   1114 1115  149.81  139.10                  -40.80\n",
      "   65   4911 148K 149K C   N    1121 1123 -105.06 -118.00   19.70    0.72  -62.90  170.34   20.21\n",
      "   65        149K 149K N   CA   1123 1124  124.24  139.10                  -40.80\n",
      "   66   4912 149K 150L C   N    1130 1132 -114.68 -108.50    8.27    0.37  -63.50 -173.63   23.48\n",
      "   66        150L 150L N   CA   1132 1133  138.00  132.50                  -41.20\n",
      "   67   4913 150L 151A C   N    1138 1140  -64.90  -68.20    3.46    0.32  -62.50  172.78   28.43\n",
      "   67        151A 151A N   CA   1140 1141  146.34  145.30                  -40.90\n",
      "   68   4914 151A 152A C   N    1143 1145  -99.31  -68.20   33.75    3.23  -62.50  176.99   27.44\n",
      "   68        152A 152A N   CA   1145 1146  132.23  145.30                  -40.90\n",
      "   69   4915 152A 153T C   N    1148 1150 -112.77 -124.80   19.78    0.61  -63.20  176.99   19.84\n",
      "   69        153T 153T N   CA   1150 1151  127.80  143.50                  -42.10\n",
      "   70   4916 153T 154P C   N    1155 1157  -72.21  -58.70   19.14    1.35  -64.50  164.32   12.67\n",
      "   70        154P 154P N   CA   1157 1158  -16.94  -30.50                  147.20\n",
      "   71   4917 154P 155K C   N    1162 1164   56.46   56.60    2.52    0.19  -62.90  141.98   24.74\n",
      "   71        155K 155K N   CA   1164 1165   36.09   38.60                  -40.80\n",
      "   72   4918 155K 156K C   N    1171 1173  -77.44  -70.20    8.64    0.53  -62.90  174.68   23.51\n",
      "   72        156K 156K N   CA   1173 1174  145.12  140.40                  -40.80\n",
      "   73   4919 156K 157A C   N    1180 1182  -74.34  -68.20   11.39    0.73  -62.50  164.64   27.58\n",
      "   73        157A 157A N   CA   1182 1183  154.89  145.30                  -40.90\n",
      "   74   4920 157A 158K C   N    1185 1187  -65.82  -70.20    4.51    0.30  -62.90  179.91   23.52\n",
      "   74        158K 158K N   CA   1187 1188  139.32  140.40                  -40.80\n",
      "   75   4921 158K 159K C   N    1194 1196  -74.75  -70.20    4.59    0.35  -62.90  179.75   24.01\n",
      "   75        159K 159K N   CA   1196 1197  139.84  140.40                  -40.80\n",
      "   76   4923 160P 161K C   N    1210 1212 -103.02 -118.00   24.56    0.91  -62.90  165.38   19.65\n",
      "   76        161K 161K N   CA   1212 1213  119.64  139.10                  -40.80\n",
      "   77   4926 163V 164K C   N    1233 1235 -140.85 -118.00   51.83    2.12  -62.90  154.66   24.73\n",
      "   77        164K 164K N   CA   1235 1236 -174.38  139.10                  -40.80\n",
      "   78   4927 164K 165A C   N    1242 1244  -59.83  -68.20    9.60    0.65  -62.50  178.52   29.13\n",
      "   78        165A 165A N   CA   1244 1245  140.60  145.30                  -40.90\n",
      "   79   4928 165A 166K C   N    1247 1249  -74.91  -70.20    9.52    0.78  -62.90  173.35   21.88\n",
      "   79        166K 166K N   CA   1249 1250  132.13  140.40                  -40.80\n",
      "   80   4932 169K 170A C   N    1279 1281 -120.19 -134.00   14.27    0.35  -62.50 -175.05   33.39\n",
      "   80        170A 170A N   CA   1281 1282  143.38  147.00                  -40.90\n",
      "   81   4933 170A 171S C   N    1284 1286 -131.39 -136.60    6.78    0.37  -64.10 -177.67   17.97\n",
      "   81        171S 171S N   CA   1286 1287  155.54  151.20                  -35.00\n",
      "   82   4934 171S 172K C   N    1290 1292 -126.35 -118.00   22.45    0.95  -62.90  171.43   26.06\n",
      "   82        172K 172K N   CA   1292 1293  159.94  139.10                  -40.80\n",
      "   83   4936 173P 174K C   N    1306 1308  -64.26  -70.20   23.87    1.59  -62.90  158.08   20.46\n",
      "   83        174K 174K N   CA   1308 1309  117.28  140.40                  -40.80\n",
      "   84   4937 174K 175K C   N    1315 1317 -139.22 -118.00   22.63    1.00  -62.90 -171.79   21.67\n",
      "   84        175K 175K N   CA   1317 1318  131.24  139.10                  -40.80\n",
      "   85   4938 175K 176A C   N    1324 1326 -159.13 -134.00   25.28    0.74  -62.50 -165.06   37.10\n",
      "   85        176A 176A N   CA   1326 1327  149.79  147.00                  -40.90\n",
      "   86   4939 176A 177K C   N    1329 1331 -117.08 -118.00    2.26    0.09  -62.90 -174.09   21.81\n",
      "   86        177K 177K N   CA   1331 1332  137.04  139.10                  -40.80\n",
      "   87   4942 179V 180K C   N    1352 1354 -124.32 -118.00   19.38    0.84  -62.90  173.05   26.14\n",
      "   87        180K 180K N   CA   1354 1355  157.42  139.10                  -40.80\n",
      "   88   4944 181P 182K C   N    1368 1370 -117.54 -118.00    0.79    0.04  -62.90 -172.41   27.61\n",
      "   88        182K 182K N   CA   1370 1371  139.74  139.10                  -40.80\n",
      "   89   4945 182K 183A C   N    1377 1379 -156.00 -134.00   22.14    0.64  -62.50 -166.30   36.74\n",
      "   89        183A 183A N   CA   1379 1380  149.46  147.00                  -40.90\n",
      "   90   4946 183A 184K C   N    1382 1384 -113.87 -118.00    4.43    0.20  -62.90 -174.36   27.13\n",
      "   90        184K 184K N   CA   1384 1385  140.70  139.10                  -40.80\n",
      "   91   4947 184K 185S C   N    1391 1393 -131.79 -136.60   14.72    0.82  -64.10  173.62   17.39\n",
      "   91        185S 185S N   CA   1393 1394  165.11  151.20                  -35.00\n",
      "   92   4948 185S 186S C   N    1397 1399 -111.49 -136.60   30.49    0.93  -64.10  175.41   10.86\n",
      "   92        186S 186S N   CA   1399 1400  133.89  151.20                  -35.00\n",
      "   93   4949 186S 187A C   N    1403 1405 -134.17 -134.00    0.88    0.05  -62.50 -174.36   34.26\n",
      "   93        187A 187A N   CA   1405 1406  147.86  147.00                  -40.90\n",
      "   94   4950 187A 188K C   N    1408 1410 -134.14 -118.00   17.07    0.57  -62.90 -171.47   28.74\n",
      "   94        188K 188K N   CA   1410 1411  144.65  139.10                  -40.80\n",
      "   95   4951 188K 189R C   N    1417 1419 -114.37 -125.20   12.99    0.69  -63.00  178.68   27.69\n",
      "   95        189R 189R N   CA   1419 1420  147.76  140.60                  -41.10\n",
      "   96   4952 189R 190A C   N    1428 1430 -105.21 -134.00   29.30    1.18  -62.50  172.04   30.45\n",
      "   96        190A 190A N   CA   1430 1431  152.45  147.00                  -40.90\n",
      "   97   4953 190A 191G C   N    1433 1435  106.95   78.70   51.41    0.83   82.20  144.58    8.24\n",
      "   97        191G 191G N   CA   1435 1436  150.95 -166.10                    8.50\n",
      "   98   4954 191G 192K C   N    1437 1439  -78.27  -70.20   11.89    0.71  -62.90  170.77   23.05\n",
      "   98        192K 192K N   CA   1439 1440  149.13  140.40                  -40.80\n",
      "   99   4955 192K 193K C   N    1446 1448  -71.57  -70.20    1.96    0.16  -62.90 -179.99   22.92\n",
      "   99        193K 193K N   CA   1448 1449  139.01  140.40                  -40.80\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0    7   13   59   65   83   89  127  129  147  145\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LYS:OXT  LYS\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      194\n",
      "Number of all, selected real atoms                :     1466    1466\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :     9324    9324\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   139785\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :       -9147.1748\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -9147.174805\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /opt/miniconda3/envs/moldyn/lib/modeller-9.21/modlib/surf5.de\n",
      "Pair library             : /opt/miniconda3/envs/moldyn/lib/modeller-9.21/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    83.783997\n",
      "Sequence length          :          194\n",
      "Compactness              :     0.003789\n",
      "Native energy (pair)     :   -64.540437\n",
      "Native energy (surface)  :    -9.168195\n",
      "Native energy (combined) :    -4.544536\n",
      "Z score (pair)           :    -4.435039\n",
      "Z score (surface)        :    -4.984606\n",
      "Z score (combined)       :    -5.831621\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      194\n",
      "Number of all, selected real atoms                :     1466    1466\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :     9324    9324\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2394\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1068.4377\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    1487       0      0   0.004   0.004      8.2117       1.000\n",
      " 2 Bond angle potential               :    1972       0      3   1.812   1.812      126.17       1.000\n",
      " 3 Stereochemical cosine torsion poten:     809       0     33  55.031  55.031      392.51       1.000\n",
      " 4 Stereochemical improper torsion pot:     495       0      0   1.069   1.069      12.146       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2394       0      0   0.004   0.004      3.9423       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    1114       0      3   0.197   0.197      39.910       1.000\n",
      "10 Distance restraints 2 (N-O)        :    1176       0      1   0.257   0.257      58.689       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     193       0      0   3.000   3.000      20.485       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     158       0      2  84.225  84.225      56.329       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     115       0      1  61.879  61.879      27.564       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      89       0      0  86.165  86.165      62.173       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      62       0      0  71.782  71.782      33.910       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :     951       0      0   0.541   0.541      29.531       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     192      98     16  30.051 127.222      120.03       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     511       0      0   1.189   1.189      76.840       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: humanH1.V99990005\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   33724.5703\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   4764   1M   2T C   N       7    9 -124.27 -124.80    2.23    0.11  -63.20 -177.26   27.48\n",
      "    1          2T   2T N   CA      9   10  145.66  143.50                  -42.10\n",
      "    2   4765   2T   3E C   N      14   16 -128.80 -117.80   15.40    0.82  -63.60  178.65   21.02\n",
      "    2          3E   3E N   CA     16   17  126.02  136.80                  -40.30\n",
      "    3   4766   3E   4N C   N      23   25 -174.60 -119.90   61.80    1.89  -63.20 -170.61   29.74\n",
      "    3          4N   4N N   CA     25   26  165.75  137.00                  -41.10\n",
      "    4   4768   5S   6T C   N      37   39  170.76 -124.80   71.51    2.88  -63.20 -169.08   32.08\n",
      "    4          6T   6T N   CA     39   40  174.50  143.50                  -42.10\n",
      "    5   4769   6T   7S C   N      44   46 -151.39 -136.60   42.11    1.77  -64.10  160.24   17.91\n",
      "    5          7S   7S N   CA     46   47 -169.38  151.20                  -35.00\n",
      "    6   4770   7S   8A C   N      50   52  -74.26  -68.20   16.39    1.52  -62.50  171.38   27.53\n",
      "    6          8A   8A N   CA     52   53  130.08  145.30                  -40.90\n",
      "    7   4772   9P  10A C   N      62   64 -163.76 -134.00   35.19    0.84  -62.50 -176.24   35.48\n",
      "    7         10A  10A N   CA     64   65  165.76  147.00                  -40.90\n",
      "    8   4773  10A  11A C   N      67   69 -111.07 -134.00   22.95    0.72  -62.50  179.85   32.05\n",
      "    8         11A  11A N   CA     69   70  145.93  147.00                  -40.90\n",
      "    9   4774  11A  12K C   N      72   74  -82.57  -70.20   38.67    2.51  -62.90  143.52   19.77\n",
      "    9         12K  12K N   CA     74   75  177.03  140.40                  -40.80\n",
      "   10   4776  13P  14K C   N      88   90 -101.07 -118.00   17.47    0.60  -62.90  179.71   21.55\n",
      "   10         14K  14K N   CA     90   91  134.81  139.10                  -40.80\n",
      "   11   4777  14K  15R C   N      97   99 -139.18 -125.20   30.32    1.11  -63.00  169.48   28.07\n",
      "   11         15R  15R N   CA     99  100  167.51  140.60                  -41.10\n",
      "   12   4778  15R  16A C   N     108  110  -66.07  -68.20   11.41    0.84  -62.50  175.03   28.51\n",
      "   12         16A  16A N   CA    110  111  134.09  145.30                  -40.90\n",
      "   13   4779  16A  17K C   N     113  115 -174.59 -118.00   57.14    2.03  -62.90 -154.76   33.28\n",
      "   13         17K  17K N   CA    115  116  147.01  139.10                  -40.80\n",
      "   14   4780  17K  18A C   N     122  124 -122.09 -134.00   12.22    0.31  -62.50 -175.28   33.45\n",
      "   14         18A  18A N   CA    124  125  144.26  147.00                  -40.90\n",
      "   15   4781  18A  19S C   N     127  129 -139.98 -136.60    6.15    0.35  -64.10 -165.64   19.48\n",
      "   15         19S  19S N   CA    129  130  146.06  151.20                  -35.00\n",
      "   16   4782  19S  20K C   N     133  135  -75.79  -70.20    5.59    0.41  -62.90  179.44   24.03\n",
      "   16         20K  20K N   CA    135  136  140.22  140.40                  -40.80\n",
      "   17   4783  20K  21K C   N     142  144  -71.06  -70.20    3.32    0.22  -62.90  175.78   23.28\n",
      "   17         21K  21K N   CA    144  145  143.61  140.40                  -40.80\n",
      "   18   4784  21K  22S C   N     151  153 -137.09 -136.60   23.33    1.17  -64.10  167.24   17.34\n",
      "   18         22S  22S N   CA    153  154  174.53  151.20                  -35.00\n",
      "   19   4785  22S  23T C   N     157  159  -98.29 -124.80   87.17    3.01  -63.20  108.40   11.95\n",
      "   19         23T  23T N   CA    159  160   60.46  143.50                  -42.10\n",
      "   20   4786  23T  24D C   N     164  166 -131.23  -96.50   50.86    2.13  -63.30 -178.18   26.83\n",
      "   20         24D  24D N   CA    166  167  151.35  114.20                  -40.00\n",
      "   21   4825  62E  63N C   N     475  477  -98.78 -119.90   91.10    3.62  -63.20   96.30   10.53\n",
      "   21         63N  63N N   CA    477  478   48.39  137.00                  -41.10\n",
      "   22   4859  96A  97K C   N     723  725   75.25  -70.20  149.46   10.09  -62.90 -158.40   34.11\n",
      "   22         97K  97K N   CA    725  726  106.02  140.40                  -40.80\n",
      "   23   4860  97K  98S C   N     732  734  178.80 -136.60   69.56    2.36  -64.10  167.97   20.56\n",
      "   23         98S  98S N   CA    734  735 -155.42  151.20                  -35.00\n",
      "   24   4861  98S  99D C   N     738  740  -60.79  -70.90   26.07    0.89  -63.30  166.28   20.68\n",
      "   24         99D  99D N   CA    740  741  126.27  150.30                  -40.00\n",
      "   25   4862  99D 100E C   N     746  748  -66.30  -69.30   15.76    1.09  -63.60  167.35   22.17\n",
      "   25        100E 100E N   CA    748  749  127.03  142.50                  -40.30\n",
      "   26   4863 100E 101P C   N     755  757  -57.06  -58.70    1.81    0.26  -64.50  177.07   12.83\n",
      "   26        101P 101P N   CA    757  758  -29.72  -30.50                  147.20\n",
      "   27   4864 101P 102K C   N     762  764   73.44   56.60  107.70    9.00  -62.90 -138.77   25.97\n",
      "   27        102K 102K N   CA    764  765  144.98   38.60                  -40.80\n",
      "   28   4865 102K 103K C   N     771  773  -76.02  -70.20   10.52    0.64  -62.90  170.55   22.89\n",
      "   28        103K 103K N   CA    773  774  149.16  140.40                  -40.80\n",
      "   29   4866 103K 104S C   N     780  782 -104.23 -136.60   50.54    1.71  -64.10  152.75    9.47\n",
      "   29        104S 104S N   CA    782  783  112.38  151.20                  -35.00\n",
      "   30   4868 105V 106A C   N     793  795  -62.43  -68.20   24.25    1.75  -62.50  162.64   26.66\n",
      "   30        106A 106A N   CA    795  796  121.74  145.30                  -40.90\n",
      "   31   4869 106A 107F C   N     798  800 -141.70 -124.20   19.48    0.99  -63.20 -164.50   22.72\n",
      "   31        107F 107F N   CA    800  801  134.74  143.30                  -44.30\n",
      "   32   4870 107F 108K C   N     809  811 -141.24 -118.00   30.29    1.50  -62.90  178.58   20.52\n",
      "   32        108K 108K N   CA    811  812  119.68  139.10                  -40.80\n",
      "   33   4871 108K 109K C   N     818  820 -157.22 -118.00   52.26    1.80  -62.90  173.45   28.12\n",
      "   33        109K 109K N   CA    820  821  173.64  139.10                  -40.80\n",
      "   34   4872 109K 110T C   N     827  829 -121.17 -124.80    9.72    0.52  -63.20  175.25   26.32\n",
      "   34        110T 110T N   CA    829  830  152.52  143.50                  -42.10\n",
      "   35   4873 110T 111K C   N     834  836  -68.92  -70.20   11.45    0.86  -62.90  167.53   22.09\n",
      "   35        111K 111K N   CA    836  837  151.77  140.40                  -40.80\n",
      "   36   4874 111K 112K C   N     843  845  -95.76 -118.00   45.09    1.79  -62.90  144.47   17.24\n",
      "   36        112K 112K N   CA    845  846   99.88  139.10                  -40.80\n",
      "   37   4875 112K 113E C   N     852  854 -131.52 -117.80   22.10    0.76  -63.60  178.96   27.69\n",
      "   37        113E 113E N   CA    854  855  154.12  136.80                  -40.30\n",
      "   38   4877 114I 115K C   N     869  871 -177.13 -118.00   61.38    2.62  -62.90 -160.60   23.16\n",
      "   38        115K 115K N   CA    871  872  122.64  139.10                  -40.80\n",
      "   39   4878 115K 116K C   N     878  880 -144.32 -118.00   28.71    0.95  -62.90 -172.74   29.19\n",
      "   39        116K 116K N   CA    880  881  150.57  139.10                  -40.80\n",
      "   40   4880 117V 118A C   N     894  896 -150.16 -134.00   18.91    0.45  -62.50 -175.55   34.91\n",
      "   40        118A 118A N   CA    896  897  156.82  147.00                  -40.90\n",
      "   41   4881 118A 119T C   N     899  901 -125.39 -124.80   31.35    1.33  -63.20  155.99   24.08\n",
      "   41        119T 119T N   CA    901  902  174.85  143.50                  -42.10\n",
      "   42   4883 120P 121K C   N     913  915 -104.01 -118.00   40.26    1.72  -62.90  147.98   17.42\n",
      "   42        121K 121K N   CA    915  916  101.35  139.10                  -40.80\n",
      "   43   4884 121K 122K C   N     922  924 -149.70 -118.00   41.43    2.05  -62.90  176.11   20.25\n",
      "   43        122K 122K N   CA    924  925  112.43  139.10                  -40.80\n",
      "   44   4885 122K 123A C   N     931  933 -147.82 -134.00   13.86    0.49  -62.50 -167.09   36.18\n",
      "   44        123A 123A N   CA    933  934  146.09  147.00                  -40.90\n",
      "   45   4886 123A 124S C   N     936  938 -163.85 -136.60   30.72    0.94  -64.10 -171.77   20.83\n",
      "   45        124S 124S N   CA    938  939  165.37  151.20                  -35.00\n",
      "   46   4887 124S 125K C   N     942  944  -78.77  -70.20    8.82    0.59  -62.90  177.43   23.94\n",
      "   46        125K 125K N   CA    944  945  142.48  140.40                  -40.80\n",
      "   47   4889 126P 127K C   N     958  960 -135.26 -118.00   17.79    0.75  -62.90 -170.07   21.92\n",
      "   47        127K 127K N   CA    960  961  134.81  139.10                  -40.80\n",
      "   48   4890 127K 128K C   N     967  969 -128.13 -118.00   10.14    0.38  -62.90 -168.70   28.74\n",
      "   48        128K 128K N   CA    969  970  139.36  139.10                  -40.80\n",
      "   49   4891 128K 129A C   N     976  978  -68.27  -68.20    7.04    0.57  -62.50  179.25   29.10\n",
      "   49        129A 129A N   CA    978  979  138.26  145.30                  -40.90\n",
      "   50   4892 129A 130A C   N     981  983 -168.38 -134.00   38.21    0.87  -62.50 -171.91   36.42\n",
      "   50        130A 130A N   CA    983  984  163.65  147.00                  -40.90\n",
      "   51   4893 130A 131S C   N     986  988 -135.64 -136.60    4.60    0.25  -64.10 -176.20   18.40\n",
      "   51        131S 131S N   CA    988  989  155.70  151.20                  -35.00\n",
      "   52   4894 131S 132K C   N     992  994 -109.35 -118.00    9.71    0.45  -62.90 -178.29   26.34\n",
      "   52        132K 132K N   CA    994  995  143.52  139.10                  -40.80\n",
      "   53   4895 132K 133A C   N    1001 1003 -142.04 -134.00    9.48    0.23  -62.50 -174.96   34.59\n",
      "   53        133A 133A N   CA   1003 1004  152.03  147.00                  -40.90\n",
      "   54   4897 134P 135T C   N    1013 1015  -71.56  -78.10   14.59    0.48  -63.20  179.05   22.42\n",
      "   54        135T 135T N   CA   1015 1016  136.75  149.80                  -42.10\n",
      "   55   4898 135T 136K C   N    1020 1022 -129.03 -118.00   30.65    1.30  -62.90  165.31   25.43\n",
      "   55        136K 136K N   CA   1022 1023  167.70  139.10                  -40.80\n",
      "   56   4899 136K 137K C   N    1029 1031  -91.51  -70.20   23.13    1.87  -62.90  174.57   21.27\n",
      "   56        137K 137K N   CA   1031 1032  131.41  140.40                  -40.80\n",
      "   57   4900 137K 138P C   N    1038 1040  -67.21  -58.70    8.52    1.11  -64.50  178.05   13.43\n",
      "   57        138P 138P N   CA   1040 1041  -30.83  -30.50                  147.20\n",
      "   58   4901 138P 139K C   N    1045 1047 -152.84 -118.00   35.23    1.25  -62.90 -163.32   30.92\n",
      "   58        139K 139K N   CA   1047 1048  144.29  139.10                  -40.80\n",
      "   59   4902 139K 140A C   N    1054 1056 -135.29 -134.00    2.01    0.12  -62.50 -171.71   34.76\n",
      "   59        140A 140A N   CA   1056 1057  145.46  147.00                  -40.90\n",
      "   60   4903 140A 141T C   N    1059 1061 -134.77 -124.80   17.73    0.55  -63.20  175.04   27.13\n",
      "   60        141T 141T N   CA   1061 1062  158.16  143.50                  -42.10\n",
      "   61   4906 143V 144K C   N    1080 1082 -139.55 -118.00   22.28    0.95  -62.90 -169.64   21.92\n",
      "   61        144K 144K N   CA   1082 1083  133.44  139.10                  -40.80\n",
      "   62   4907 144K 145K C   N    1089 1091 -142.85 -118.00   26.90    1.21  -62.90 -172.49   21.56\n",
      "   62        145K 145K N   CA   1091 1092  128.81  139.10                  -40.80\n",
      "   63   4908 145K 146A C   N    1098 1100 -133.41 -134.00    0.79    0.02  -62.50 -173.37   34.38\n",
      "   63        146A 146A N   CA   1100 1101  146.47  147.00                  -40.90\n",
      "   64   4909 146A 147K C   N    1103 1105 -115.28 -118.00   29.30    1.38  -62.90  159.57   18.57\n",
      "   64        147K 147K N   CA   1105 1106  109.93  139.10                  -40.80\n",
      "   65   4910 147K 148K C   N    1112 1114 -171.71 -118.00   53.71    2.02  -62.90 -149.83   33.77\n",
      "   65        148K 148K N   CA   1114 1115  139.39  139.10                  -40.80\n",
      "   66   4911 148K 149K C   N    1121 1123 -103.38 -118.00   23.73    0.88  -62.90  166.22   19.74\n",
      "   66        149K 149K N   CA   1123 1124  120.41  139.10                  -40.80\n",
      "   67   4912 149K 150L C   N    1130 1132 -100.09 -108.50   13.49    0.62  -63.50  167.21   21.41\n",
      "   67        150L 150L N   CA   1132 1133  121.96  132.50                  -41.20\n",
      "   68   4913 150L 151A C   N    1138 1140 -132.78 -134.00   10.60    0.57  -62.50 -169.22   28.77\n",
      "   68        151A 151A N   CA   1140 1141  136.46  147.00                  -40.90\n",
      "   69   4914 151A 152A C   N    1143 1145 -165.68 -134.00   33.58    0.78  -62.50 -168.79   36.81\n",
      "   69        152A 152A N   CA   1145 1146  158.12  147.00                  -40.90\n",
      "   70   4915 152A 153T C   N    1148 1150 -127.18 -124.80    2.55    0.11  -63.20 -175.11   27.94\n",
      "   70        153T 153T N   CA   1150 1151  144.43  143.50                  -42.10\n",
      "   71   4917 154P 155K C   N    1162 1164  -92.37  -70.20   35.73    2.98  -62.90  155.99   18.84\n",
      "   71        155K 155K N   CA   1164 1165  112.38  140.40                  -40.80\n",
      "   72   4918 155K 156K C   N    1171 1173 -118.31 -118.00    8.14    0.39  -62.90 -179.33   26.76\n",
      "   72        156K 156K N   CA   1173 1174  147.24  139.10                  -40.80\n",
      "   73   4919 156K 157A C   N    1180 1182 -101.18  -68.20   34.37    3.20  -62.50 -179.30   27.98\n",
      "   73        157A 157A N   CA   1182 1183  135.61  145.30                  -40.90\n",
      "   74   4920 157A 158K C   N    1185 1187  -96.18 -118.00   25.48    0.85  -62.90  170.04   20.50\n",
      "   74        158K 158K N   CA   1187 1188  125.95  139.10                  -40.80\n",
      "   75   4921 158K 159K C   N    1194 1196  -97.32 -118.00   42.66    1.70  -62.90  146.68   17.47\n",
      "   75        159K 159K N   CA   1196 1197  101.79  139.10                  -40.80\n",
      "   76   4924 161K 162T C   N    1219 1221  -75.41  -78.10   22.76    1.06  -63.20  146.01   19.53\n",
      "   76        162T 162T N   CA   1221 1222  172.40  149.80                  -42.10\n",
      "   77   4926 163V 164K C   N    1233 1235  -71.53  -70.20    1.36    0.09  -62.90  178.75   23.69\n",
      "   77        164K 164K N   CA   1235 1236  140.66  140.40                  -40.80\n",
      "   78   4927 164K 165A C   N    1242 1244  -77.66  -68.20    9.50    0.78  -62.50  173.58   29.22\n",
      "   78        165A 165A N   CA   1244 1245  146.18  145.30                  -40.90\n",
      "   79   4928 165A 166K C   N    1247 1249 -146.19 -118.00   34.34    1.15  -62.90 -179.19   28.46\n",
      "   79        166K 166K N   CA   1249 1250  158.72  139.10                  -40.80\n",
      "   80   4931 168V 169K C   N    1270 1272 -112.05 -118.00   16.76    0.87  -62.90  171.62   25.20\n",
      "   80        169K 169K N   CA   1272 1273  154.77  139.10                  -40.80\n",
      "   81   4933 170A 171S C   N    1284 1286  -77.40  -72.40    6.92    0.33  -64.10  168.34   13.03\n",
      "   81        171S 171S N   CA   1286 1287  157.18  152.40                  -35.00\n",
      "   82   4934 171S 172K C   N    1290 1292 -121.05 -118.00   11.15    0.57  -62.90  178.89   20.84\n",
      "   82        172K 172K N   CA   1292 1293  128.38  139.10                  -40.80\n",
      "   83   4936 173P 174K C   N    1306 1308 -118.71 -118.00    2.66    0.14  -62.90 -174.08   21.77\n",
      "   83        174K 174K N   CA   1308 1309  136.54  139.10                  -40.80\n",
      "   84   4937 174K 175K C   N    1315 1317 -133.15 -118.00   18.21    0.61  -62.90 -176.07   28.09\n",
      "   84        175K 175K N   CA   1317 1318  149.21  139.10                  -40.80\n",
      "   85   4938 175K 176A C   N    1324 1326 -110.27 -134.00   27.17    0.63  -62.50 -178.92   27.75\n",
      "   85        176A 176A N   CA   1326 1327  133.77  147.00                  -40.90\n",
      "   86   4939 176A 177K C   N    1329 1331  -88.76  -70.20   29.17    1.75  -62.90  158.42   22.07\n",
      "   86        177K 177K N   CA   1331 1332  162.90  140.40                  -40.80\n",
      "   87   4942 179V 180K C   N    1352 1354  -76.25  -70.20   10.83    0.66  -62.90  170.34   22.87\n",
      "   87        180K 180K N   CA   1354 1355  149.38  140.40                  -40.80\n",
      "   88   4944 181P 182K C   N    1368 1370 -154.87 -118.00   43.17    1.43  -62.90 -177.49   29.18\n",
      "   88        182K 182K N   CA   1370 1371  161.56  139.10                  -40.80\n",
      "   89   4945 182K 183A C   N    1377 1379  -69.45  -68.20    1.26    0.10  -62.50  173.76   28.82\n",
      "   89        183A 183A N   CA   1379 1380  145.48  145.30                  -40.90\n",
      "   90   4947 184K 185S C   N    1391 1393 -151.78 -136.60   32.22    1.25  -64.10  169.77   18.62\n",
      "   90        185S 185S N   CA   1393 1394  179.62  151.20                  -35.00\n",
      "   91   4948 185S 186S C   N    1397 1399  -73.12  -72.40    2.59    0.13  -64.10  170.35   12.90\n",
      "   91        186S 186S N   CA   1399 1400  154.89  152.40                  -35.00\n",
      "   92   4949 186S 187A C   N    1403 1405 -105.20 -134.00   32.68    0.75  -62.50  177.66   27.35\n",
      "   92        187A 187A N   CA   1405 1406  131.56  147.00                  -40.90\n",
      "   93   4950 187A 188K C   N    1408 1410 -123.30 -118.00    5.64    0.25  -62.90 -172.06   21.90\n",
      "   93        188K 188K N   CA   1410 1411  137.16  139.10                  -40.80\n",
      "   94   4951 188K 189R C   N    1417 1419 -111.24 -125.20   30.90    1.13  -63.00  161.50   19.07\n",
      "   94        189R 189R N   CA   1419 1420  113.03  140.60                  -41.10\n",
      "   95   4952 189R 190A C   N    1428 1430 -178.92 -134.00   54.47    1.34  -62.50 -176.91   36.06\n",
      "   95        190A 190A N   CA   1430 1431  177.80  147.00                  -40.90\n",
      "   96   4953 190A 191G C   N    1433 1435  103.17   78.70   45.71    0.72   82.20  148.28    8.24\n",
      "   96        191G 191G N   CA   1435 1436  155.29 -166.10                    8.50\n",
      "   97   4954 191G 192K C   N    1437 1439 -115.67 -118.00    9.96    0.51  -62.90  178.39   26.30\n",
      "   97        192K 192K N   CA   1439 1440  148.79  139.10                  -40.80\n",
      "   98   4955 192K 193K C   N    1446 1448  -74.22  -70.20    6.07    0.36  -62.90  174.62   23.31\n",
      "   98        193K 193K N   CA   1448 1449  144.94  140.40                  -40.80\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0    5   17   63   72  111  128  138  165  166  191\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf     DOPE score    GA341 score\n",
      "----------------------------------------------------------------------\n",
      "humanH1.B99990001.pdb         1033.15295    -9054.66211        1.00000\n",
      "humanH1.B99990002.pdb          876.37494    -9417.93262        1.00000\n",
      "humanH1.B99990003.pdb          894.53113    -9577.04102        1.00000\n",
      "humanH1.B99990004.pdb          917.39508    -9488.55469        1.00000\n",
      "humanH1.B99990005.pdb         1068.43774    -9147.17480        1.00000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#Let's run MODELLER\n",
    "from modeller import *\n",
    "from modeller.automodel import *\n",
    "\n",
    "env = environ()\n",
    "a = automodel(env, alnfile='humanH1-1HSTA.ali',\n",
    "              knowns=('1HSTA'), sequence='humanH1',\n",
    "              assess_methods=(assess.DOPE,assess.GA341))\n",
    "a.starting_model = 1\n",
    "a.ending_model = 5\n",
    "a.make()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'humanH1.B99990001.pdb'"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mod_f_name_0=a.outputs[0]['name']\n",
    "mod_f_name_0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "openf___224_> Open           $(LIB)/restyp.lib\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/resgrp.lib\n",
      "rdresgr_266_> Number of residue groups:        2\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/sstruc.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     15537215   15173.062    14.817\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     15537743   15173.577    14.818\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/resdih.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     15586343   15221.038    14.864\n",
      "rdrdih__263_> Number of dihedral angle types         :        9\n",
      "              Maximal number of dihedral angle optima:        3\n",
      "              Dihedral angle names                   :  Alph Phi Psi Omeg chi1 chi2 chi3 chi4 chi5\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/radii.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     15599643   15234.026    14.877\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/radii14.lib\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/af_mnchdef.lib\n",
      "rdwilmo_274_> Mainchain residue conformation classes:  APBLE\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/mnch.lib\n",
      "rdclass_257_> Number of classes:        5\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/mnch1.lib\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/mnch2.lib\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/mnch3.lib\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/xs4.mat\n",
      "rdrrwgh_268_> Number of residue types:       21\n",
      "openf___224_> Open           $(LIB)/top_heav.lib\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "openf___224_> Open           $(LIB)/par.lib\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     15738895   15370.015    15.010\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     15740495   15371.577    15.011\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     15744475   15375.464    15.015\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     15746475   15377.417    15.017\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     15749475   15380.347    15.020\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     15753975   15384.741    15.024\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     15759547   15390.183    15.029\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     15762347   15392.917    15.032\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     15766547   15397.019    15.036\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     15772847   15403.171    15.042\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     15775619   15405.878    15.045\n",
      "\n",
      "Dynamically allocated memory at  amaxparameters [B,KiB,MiB]:     15777019   15407.245    15.046\n",
      "read_pa_232_> parameters    BONDS   ANGLS  DIHEDS IMPROPS    MODE\n",
      "                              227     561     661     112       0\n",
      "openf___224_> Open           humanH1.B99990001.pdb\n",
      "readlinef__W> File: humanH1.B99990001.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15787992   15417.961    15.057\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     15788068   15418.035    15.057\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15788239   15418.202    15.057\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15789089   15419.032    15.058\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15790347   15420.261    15.059\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15792251   15422.120    15.061\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     15792291   15422.159    15.061\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15792381   15422.247    15.061\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15795237   15425.036    15.064\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     15795297   15425.095    15.064\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15795432   15425.227    15.064\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15799716   15429.410    15.068\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     15799804   15429.496    15.068\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15800002   15429.689    15.068\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15806428   15435.965    15.074\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     15806560   15436.094    15.074\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15806857   15436.384    15.075\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15816479   15445.780    15.084\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     15816679   15445.976    15.084\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15817129   15446.415    15.084\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15831579   15460.526    15.098\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     15831879   15460.819    15.098\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15832554   15461.479    15.099\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     15854212   15482.629    15.120\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     16008772   15633.566    15.267\n",
      "read_mo_297_> Segments, residues, atoms:        1      194     1466\n",
      "read_mo_298_> Segment:        1     1   194     1466\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     16014950   15639.600    15.273\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     16125528   15747.586    15.379\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     16127450   15749.463    15.380\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     16238028   15857.449    15.486\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     16258060   15877.012    15.505\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues atoms bonds angles dihedrals impropers:\n",
      "                     1      194  1465  1486      0         0       509\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     16319512   15937.023    15.563\n",
      "\n",
      "patch_______> segment topology patched using RTF:     1  ;  MET  ;  NTER\n",
      "              segments residues atoms bonds angles dihedrals impropers:\n",
      "                     1      194  1465  1486   1970      2365       509\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     16383528   15999.539    15.625\n",
      "patch_______> segment topology patched using RTF:   194  ;  LYS  ;  CTER\n",
      "              segments residues atoms bonds angles dihedrals impropers:\n",
      "                     1      194  1466  1487   1972      2367       510\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues atoms bonds angles dihedrals impropers:\n",
      "                     1      194  1466  1487   1972      2367       510\n",
      "transfe_506_> MODEL is an average of all templates.\n",
      "transfe_511_> Number of templates for coordinate transfer:        1\n",
      "              After transferring coordinates of the equivalent template atoms,\n",
      "              there are defined, undefined atoms in MODEL:     1466        0\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LYS:OXT  LYS\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "Dynamically allocated memory at amaxhash_contac [B,KiB,MiB]:     13810166   13486.490    13.170\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     14137954   13806.596    13.483\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     14173138   13840.955    13.517\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     14243506   13909.674    13.584\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     14384242   14047.111    13.718\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     14665714   14321.986    13.986\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     15228658   14871.736    14.523\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      194\n",
      "Number of all, selected real atoms                :     1466    1466\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   125547\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :       -9055.1719\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 2 Bond angle potential               :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 3 Stereochemical cosine torsion poten:       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 4 Stereochemical improper torsion pot:       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 5 Soft-sphere overlap restraints     :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "14 Sidechain Chi_1 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       0.000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Non-bonded restraints              :  125515       0    311   6.447   6.447     -9055.2       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "do_prof_492W> Are you sure you want to calculate a normalized energy profile!\n",
      "\n",
      "\n",
      "\n",
      "openf___224_> Open           model.profile\n",
      "# Energy of each residue is written to: model.profile\n",
      "# The profile IS normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:   13\n",
      "# The sum of all numbers in the file:      -6.5890\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -9055.171875\n",
      "openf___224_> Open           1HST.pdb\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14946859   14596.542    14.254\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     14946935   14596.616    14.255\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14947106   14596.783    14.255\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14947956   14597.613    14.255\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14949214   14598.842    14.257\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14951118   14600.701    14.258\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     14951158   14600.740    14.259\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14951248   14600.828    14.259\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14954104   14603.617    14.261\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     14954164   14603.676    14.261\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14954299   14603.808    14.262\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14958583   14607.991    14.266\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     14958671   14608.077    14.266\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14958869   14608.271    14.266\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14965295   14614.546    14.272\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     14965427   14614.675    14.272\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14965724   14614.965    14.272\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     14965808   14615.047    14.273\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14965808   14615.047    14.273\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14975430   14624.443    14.282\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     14975630   14624.639    14.282\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14976080   14625.078    14.282\n",
      "\n",
      "Dynamically allocated memory at amaxcoordinates [B,KiB,MiB]:     14990530   14639.189    14.296\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     15093522   14739.768    14.394\n",
      "read_mo_297_> Segments, residues, atoms:        2      148     1128\n",
      "read_mo_298_> Segment:        1    24 A    97 A      564\n",
      "read_mo_298_> Segment:        2    24 B    97 B      564\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     15098366   14744.498    14.399\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     15183082   14827.229    14.480\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     15184590   14828.701    14.481\n",
      "\n",
      "Dynamically allocated memory at   amaxstructure [B,KiB,MiB]:     15269306   14911.432    14.562\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     15277474   14919.408    14.570\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues atoms bonds angles dihedrals impropers:\n",
      "                     1       74   564   571      0         0       225\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     15300742   14942.131    14.592\n",
      "\n",
      "patch_______> segment topology patched using RTF:     1  ;  SER  ;  NTER\n",
      "              segments residues atoms bonds angles dihedrals impropers:\n",
      "                     1       74   564   571    760       886       225\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     15326558   14967.342    14.617\n",
      "patch_______> segment topology patched using RTF:    74  ;  LYS  ;  CTER\n",
      "              segments residues atoms bonds angles dihedrals impropers:\n",
      "                     1       74   565   572    762       888       226\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues atoms bonds angles dihedrals impropers:\n",
      "                     1       74   565   572    762       888       226\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     15334750   14975.342    14.624\n",
      "\n",
      "mkilst______> segment topology constructed from sequence and RTF:\n",
      "              segments residues atoms bonds angles dihedrals impropers:\n",
      "                     2      148  1129  1143    762       888       451\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     15358046   14998.092    14.647\n",
      "\n",
      "patch_______> segment topology patched using RTF:    75  ;  SER  ;  NTER\n",
      "              segments residues atoms bonds angles dihedrals impropers:\n",
      "                     2      148  1129  1143   1522      1774       451\n",
      "\n",
      "\n",
      "Dynamically allocated memory at       amaxmodel [B,KiB,MiB]:     15385662   15025.061    14.673\n",
      "patch_______> segment topology patched using RTF:   148  ;  LYS  ;  CTER\n",
      "              segments residues atoms bonds angles dihedrals impropers:\n",
      "                     2      148  1130  1144   1524      1776       452\n",
      "\n",
      "genseg______> segment topology constructed from sequence and RTF:\n",
      "              segments residues atoms bonds angles dihedrals impropers:\n",
      "                     2      148  1130  1144   1524      1776       452\n",
      "transfe_506_> MODEL is an average of all templates.\n",
      "transfe_511_> Number of templates for coordinate transfer:        1\n",
      "              After transferring coordinates of the equivalent template atoms,\n",
      "              there are defined, undefined atoms in MODEL:     1128        2\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LYS:OXT  LYS\n",
      "iatmcls_286W> MODEL atom not classified:  LYS:OXT  LYS\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "Dynamically allocated memory at amaxhash_contac [B,KiB,MiB]:     13610137   13291.149    12.980\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     13929861   13603.380    13.285\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     13956981   13629.864    13.310\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     14011221   13682.833    13.362\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     14119701   13788.771    13.466\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     14336661   14000.646    13.673\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     14770581   14424.396    14.086\n",
      "\n",
      "Dynamically allocated memory at  amaxrestraints [B,KiB,MiB]:     15638421   15271.896    14.914\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      148\n",
      "Number of all, selected real atoms                :     1130    1130\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   177640\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -14587.3301\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 2 Bond angle potential               :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 3 Stereochemical cosine torsion poten:       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 4 Stereochemical improper torsion pot:       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 5 Soft-sphere overlap restraints     :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       0.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "14 Sidechain Chi_1 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "15 Sidechain Chi_2 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       0.000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 Non-bonded restraints              :  177314       0    129   6.684   6.684     -14587.       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       0.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       0.000\n",
      "do_prof_492W> Are you sure you want to calculate a normalized energy profile!\n",
      "\n",
      "\n",
      "\n",
      "openf___224_> Open           1HSTA.profile\n",
      "# Energy of each residue is written to: 1HSTA.profile\n",
      "# The profile IS normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:   13\n",
      "# The sum of all numbers in the file:      -5.8732\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -14587.330078\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "-14587.330078125"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Let's do some evaluations\n",
    "from modeller import *\n",
    "from modeller.scripts import complete_pdb\n",
    "\n",
    "log.verbose()    # request verbose output\n",
    "env = environ()\n",
    "env.libs.topology.read(file='$(LIB)/top_heav.lib') # read topology\n",
    "env.libs.parameters.read(file='$(LIB)/par.lib') # read parameters\n",
    "\n",
    "# read model file\n",
    "mdl = complete_pdb(env, mod_f_name_0)\n",
    "\n",
    "# Assess with DOPE:\n",
    "s = selection(mdl)   # all atom selection\n",
    "s.assess_dope(output='ENERGY_PROFILE NO_REPORT', file='model.profile',\n",
    "              normalize_profile=True, smoothing_window=15)\n",
    "\n",
    "# read model file\n",
    "mdl = complete_pdb(env, '1HST.pdb')\n",
    "\n",
    "# Assess with DOPE:\n",
    "s = selection(mdl)   # all atom selection\n",
    "s.assess_dope(output='ENERGY_PROFILE NO_REPORT', file='1HSTA.profile',\n",
    "              normalize_profile=True, smoothing_window=15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "openf___224_> Open           $(LIB)/restyp.lib\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/resgrp.lib\n",
      "rdresgr_266_> Number of residue groups:        2\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/sstruc.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     15859881   15488.165    15.125\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     15860409   15488.681    15.126\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/resdih.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     15909009   15536.142    15.172\n",
      "rdrdih__263_> Number of dihedral angle types         :        9\n",
      "              Maximal number of dihedral angle optima:        3\n",
      "              Dihedral angle names                   :  Alph Phi Psi Omeg chi1 chi2 chi3 chi4 chi5\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/radii.lib\n",
      "\n",
      "Dynamically allocated memory at   amaxlibraries [B,KiB,MiB]:     15922309   15549.130    15.185\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/radii14.lib\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/af_mnchdef.lib\n",
      "rdwilmo_274_> Mainchain residue conformation classes:  APBLE\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/mnch.lib\n",
      "rdclass_257_> Number of classes:        5\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/mnch1.lib\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/mnch2.lib\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/mnch3.lib\n",
      "openf___224_> Open           ${MODINSTALL9v21}/modlib/xs4.mat\n",
      "rdrrwgh_268_> Number of residue types:       21\n",
      "openf___224_> Open           humanH1-1HSTA.ali\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     15924311   15551.085    15.187\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     15925761   15552.501    15.188\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     15928661   15555.333    15.191\n",
      "\n",
      "Dynamically allocated memory at   amaxalignment [B,KiB,MiB]:     15989917   15615.153    15.249\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     15990209   15615.438    15.249\n",
      "\n",
      "Dynamically allocated memory at    amaxsequence [B,KiB,MiB]:     15990981   15616.192    15.250\n",
      "\n",
      "Read the alignment from file       : humanH1-1HSTA.ali\n",
      "Total number of alignment positions:   194\n",
      "\n",
      "  #  Code        #_Res #_Segm PDB_code    Name\n",
      "-------------------------------------------------------------------------------\n",
      "  1      1HSTA      74      1    1HST.pdb MOL_ID  1; MOLECULE  HISTONE H5; CHAIN  A, B; ENGINEERED  YES\n",
      "  2    humanH1     194      1 model_file_\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAnAAAAFzCAYAAAC+bzSQAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nOzdeXyU1fnw/8/JxhLCFpYAAQJJkDUkioiAK7t1b2trW5dqq221C+5W64p1rfap1aqPy0+futSv31qXArIIoiIIyiI7YU3YyQokIcuc3x9n7swkmSSz3TP3JNf79ZrX7DMnIcxc93Wdcx2ltUYIIYQQQsSOuGgPQAghhBBCBEYCOCGEEEKIGCMBnBBCCCFEjJEATgghhBAixkgAJ4QQQggRYySAE0IIIYSIMQnRHkAk9erVS2dkZER7GEIIIYQQrfrmm2+Oaq17+7qvXQVwGRkZrF69OtrDEEIIIYRolVJqT3P3SQlVCCGEECLGSAAnhBBCCBFjJIATQgghhIgx7WoOnC81NTUUFhZSVVUV7aHEtI4dO5Kenk5iYmK0hyKEEEK0ee0+gCssLCQlJYWMjAyUUtEeTkzSWlNUVERhYSFDhgyJ9nCEEEKINq/dl1CrqqpITU2V4C0ESilSU1MliymEEEJESLsP4AAJ3sJAfodCCCFE5EgA5wBKKa666qr667W1tfTu3ZsLL7wwoNfJyMjg6NGjIT9GCCGEEM4mAZwDJCcns2HDBiorKwFYuHAhAwYMiPKohBBCCOFUEsA5xKxZs/jvf/8LwNtvv82VV15Zf19xcTGXXnopOTk5TJgwgfXr1wNQVFTE9OnTycvL48Ybb0RrXf+cf/7zn4wfP57c3FxuvPFG6urqIvsDCSGEEMI2EsB5U8qekx9+/OMf884771BVVcX69es544wz6u+7//77ycvLY/369fz5z3/m6quvBuDBBx9k8uTJrFmzhosvvpi9e/cCsHnzZv71r3/x5ZdfsnbtWuLj43nzzTfD//sSQgghRFS0+zYiTpGTk8Pu3bt5++23ueCCCxrc98UXX/C///u/AJx//vkUFRVRVlbGsmXL+Pe//w3A9773PXr06AHA4sWL+eabbzj99NMBqKyspE+fPhH8aYQQQog2asUKeO89mDwZLr00asOQAM6bVwkyGi6++GJuu+02li5dSlFRUf3t2se4rFWfvlZ/aq255pprePTRR+0brBBCCNEerV4Nf/kLVFRENYCTEqqDXHfdddx3332MGTOmwe1nn312fQl06dKl9OrVi65duza4fd68eZSUlAAwZcoU3nvvPQ4fPgyYOXR79uyJ4E8ihBBCtFFlZea8W7eoDkMycA6Snp7O73//+ya3P/DAA/z85z8nJyeHzp078/rrrwNmbtyVV17JqaeeyjnnnMOgQYMAGDlyJHPmzGH69Om4XC4SExN57rnnGDx4cER/HiGEEKLNsQK47t2jOgzlqzzXVo0bN06vXr26wW2bN29mxIgRURpR2yK/SyGEEG3ejTfCSy/BP/4Bv/qVrW+llPpGaz3O131SQhVCCCGE8FdpqTmPcglVAjghhBBCCH85ZA6cBHBCCCGEEP5yyBw4CeCEEEIIIfwlJVQhhBBCiBgjJVQhhBBCiBgjJVQBZkP63NxccnNzSUtLY8CAAfXXq6urw/5+tbW1dG/lj664uJgXXngh7O8thBBCxLSaGrMDQ3w8JCdHdSjSyDfKUlNTWbt2LWAa9nbp0oXbbrstqmOyArhf2dzfRgghhIgpVvata1fwsZVlJEkGzsFef/11xo8fT25uLr/5zW9wuVz1GbTbb7+dU089lRkzZrBy5UrOOecchg4dyty5cwF4+eWXueyyy5gxYwannHIKc+bMafL65eXlnH/++Zx66qnk5OTw8ccfA3DXXXexdetWcnNzueuuuwB47LHHGD9+PDk5OTz00EOR+yUIIYQQTuGQ+W8gGbgG1IP2RNP6/sB3u9iwYQPvv/8+y5cvJyEhgRtuuIF33nmHK664grKyMqZPn86TTz7JRRddxAMPPMDixYtZt24dN954IxdccAEAX3/9NRs2bCApKYnTTz+dCy+8kNGjR9e/R6dOnfjggw9ISUnh8OHDTJo0iQsvvJDHHnuM/Pz8+szg3Llz2bt3LytXrkRrzQUXXMDy5cuZOHFieH5BQgghRCxwyPw3kADOsRYtWsSqVasYN87soFFZWcnAgQMBE3hNmzYNgDFjxtCtWzcSEhIYM2YMu3fvrn+NGTNm0KNHDwAuvfRSvvjiiwYBnNaaO++8ky+++IK4uDgKCgo4evRok7EsWLCAefPmkZeXB8Dx48fZtm2bBHBCCCHaF4e0EAEJ4BoIJlNmF6011113HQ8//HCD22tra0lKSqq/HhcXR4cOHeov19bW1t+nGtXnG19/4403KCsr49tvvyUhIYH09HSqqqp8juXee+/l+uuvD/nnEkIIIWKWg0qoMgfOoaZOncq7775bnxErKipi7969Ab3GggULKC0tpaKigg8++IBJkyY1uL+srIw+ffqQkJDAwoUL2bdvHwApKSkcO3as/nEzZszglVde4cSJEwAUFhb6zNQJIYQQbZqUUEVrxowZw/3338/UqVNxuVwkJibywgsv0L9/f79fY/LkyfzkJz9hx44dXHXVVeTm5jbI0F111VVcdNFFjBs3jlNPPZXs7GwA+vbty7hx4xgzZgzf+973eOyxx9iyZQsTJkwATID31ltv0atXr/D+0EIIIYSTOSgDp7R2TtnQbuPGjdOrV69ucNvmzZsZMWJElEZkn5dffpkNGzbw17/+NWLv2VZ/l0IIIQQADz4IDzwA994LjaY42UEp9Y3Wepyv+6SEKoQQQgjhDwdl4KSE2kb94he/iPYQhBBCiLbFQXPgJAMnhBBCCOEPB7URkQAO0yZDhEZ+h0IIIdo8B5VQ230A17FjR4qKiiQACYHWmqKiIjp27BjtoQghhBD2cVAJtd3PgUtPT6ewsJAjR45EeygxrWPHjqSnp0d7GEIIIYR9HFRCjWoAp5SaCfwfIB54WWv9WKP7lfv+C4AK4Fqt9bf+PNdfiYmJDBkyJPgfQgghhBDtg5RQQSkVDzwHzAJGAlcqpUY2etgsINt9ugH4RwDPFUIIIYQIHwngABgP5Gutd2qtq4F3gEsaPeYS4A1trAC6K6X6+flcIYQQQojwqKqC6mpISgIHzPmOZgA3ACjwul7ovs2fx/jzXACUUjcopVYrpVbLPDchhBBCBMV7/ptS0R0L0Q3gfP30jZeCNvcYf55rbtT6Ja31OK31uN69ewc4RCGEEEIIHFU+heguYigEBnpdTwf2+/mYJD+eK4QQQggRHg5qIQLRzcCtArKVUkOUUknAj4EPGz3mQ+BqZUwAyrTWB/x8rhBCCCFEeDiohQhEMQOnta5VSt0MfIJpBfKq1nqjUupX7vtfAOZiWojkY9qI/Lyl50bhxxBCCCFEeyAlVA+t9VxMkOZ92wtelzVwk7/PFUIIIYSwhcMCuHa/lZYQQgghRKtkDpwQQgghRIxx2Bw4CeCEEEIIIVojJVQhhBBCiBgjJVQhhBBCiBgjJVQhhBBCiBgjJVQhhBBCiBgjAZwQQgghRIyROXBCCCGEEDFG5sAJIYQQQsQQraG83FyWAE4IIYQQIgacOAF1ddC5MyQmRns0gARwQgghhBAtc1j5FCSAE0IIIYRomcNWoIIEcEIIIYQQLZMATgghhBAixjishQhIACeEEEII0TKZAyeEEEIIEWOkhCqEEEIIEWOkhCqEEEIIEWOkhCqEEEIIEWOkhCqEEEIIEWMkgBNCCCGEiDEyB04IIYQQIsbIHDghhBBCiBgjJVQhhBBCiBgjJVQhhBBCiBgjJVQhhBBCiBhSVwfHjpnLKSnRHYsXCeCEEEIIIZpjBW9du0J8fHTH4kUCOCGEEEKI5jhwAQNIACeEEEII0TwHzn8DCeCEEEIIIZonGTghhBBCiBjjwBYiIAGcEEIIIUTzpIQqhBBCCBFjpIQqhBBCCBFjpIQqhBBCCBFjpIQqhBBCCBFjpITqoZTqqZRaqJTa7j7v0czjZiqltiql8pVSd3nd/kOl1EallEspNS5yIxdCCCFEuyIBXAN3AYu11tnAYvf1BpRS8cBzwCxgJHClUmqk++4NwOXAssgMVwghhBDtkkPnwCVE6X0vAc51X34dWArc2egx44F8rfVOAKXUO+7nbdJab3bfFomxCiGE8+Tnw+LFoHXD27t2hR/8AJKSojMuIdoah86Bi1YA11drfQBAa31AKdXHx2MGAAVe1wuBMyIxOCGEcLSKCjjvPCgs9H3/8eNwww2RHZMQbZVDS6i2BXBKqUVAmo+77vH3JXzcpn3c1to4bgBuABg0aFCgTxdCCOd56ikTvGVlwZQpntu3bYMlS+CrrySAEyJc2lsJVWs9tbn7lFKHlFL93Nm3fsBhHw8rBAZ6XU8H9gcxjpeAlwDGjRsXcAAohBCOsm8fPP64ufzyy3DOOZ77li+HSZNg7drojE2ItsihJdRoLWL4ELjGffka4AMfj1kFZCulhiilkoAfu58nhBDt1z33mBLq5Zc3DN4AcnJAKdi4EaqrozM+IdqSmhqorIT4eEhOjvZoGohWAPcYME0ptR2Y5r6OUqq/UmougNa6FrgZ+ATYDLyrtd7oftxlSqlC4Ezgv0qpT6LwMwghRGStWQOvv24WKDzxRNP7u3SB7GzzpbNpU+THJ0RbY5VPu3Y1B0cOEpVFDFrrImCKj9v3Axd4XZ8LzPXxuPeB9+0coxBCOM5//mPOr78eMjN9PyY318yFW7vWXBZCBM+h899AdmIQQojYkZ9vzsePb/4xeXnmfM0a+8cjRFvn0PlvIAGcEELEDiuAy8pq/jFW1k0WMggROoe2EAEJ4IQQInZs327OWwrgrAzc2rXgctk/JiHaMimhCiGECElxMZSUmIUKffs2/7i+fSEtDcrLYffuiA1PiDZJSqhCCCFC4l0+bW01nHcWTggRPCmhCiGECIk/5VOLNQ9OFjIIERoJ4IQQQoTEysBlZ7f+WFnIIER4yBw4IYQQIfFnBapFWolEj8tlmi3v3RvtkYhwkDlwQgghQhJIAJeZCZ07m31TrS8gERn/+hdcey3Mnh3tkYhwkBKqEEKIkFhz4PwpocbFmZWoAEeO2Dcm0dRHH5nzlSujOw4RHsXF5jw1Nbrj8EECOCGEcLqSEigqMlk1KzBrTe/e5lwCuMipq4NP3Ftz79snv/u2wArgevaM7jh8aDWAU0r1VUq9opSa574+Uil1vf1DE0IIAcCOHebcnxYiFgngIm/1as8XPsC6ddEbiwiPWA7ggP8P+ATo776+DfiDXQMSQgjRSCAtRCy9eplzCeAiZ968htdlEUnsi/EArpfW+l3ABaC1rgXqbB2VEEIIj0BaiFgkAxd58+eb84svNufSxiW2VVVBRQUkJkJycrRH04Q/AdwJpVQqoAGUUhOAMltHJYQQwiOQFagWK4A7ejT84xFNHT0KX38NSUlwyy3mNsnAxTbv7Ju/UxciKMGPx9wCfAhkKqW+BHoDP7B1VEIIITyCKaFKBi6yFiwAreGss+CMMyA+HrZuNRmczp2jPToRDAeXT6GVDJxSKg7oCJwDTARuBEZprddHYGxCCCEguBKqk+fAaW0WZmgd7ZGEj1U+nTULOnaEESNMU98NG6I7LhE8B7cQgVYCOK21C/iL1rpWa71Ra71Ba10TobEJIYQoLTVBWKdO0K+f/89zcgbuhRdMNvHNN6M9kvBZuNCcz5xpzmU3jNgXyxk4twVKqe8r5cACsBBCtHXr3QWPkSNNg15/OXkO3D//ac6XLYvuOMKlpAQOHjSl0pEjzW2yH23sc3gA5+8cuGSgTilVCShAa6272joyIYQQngDAyuj4y6kl1JISWLHCXLZKw7HOe5GJleuwAjjJwMWuWA/gtNYpkRiIEEIIH6wAzgoI/JWSYlZEVlQ4ayL9woVmbhi0zQDOYv17rV9vdmiIj4/8uERoiorMuUMDOL/y8Uqpi5VST7lPF9o9KCGEEG5WBifQDJxSzpwHZ032BygogMrK6I0lXHztU9uzJwwaZH6+bduiMy4RGodn4PzZSusx4PfAJvfp9+7bhBBC2Km6GjZuNMHYmDGBP99p8+C09gRwHTua8507ozeecGmuT5+Vhfvuu8iOR4RHrAdwwAXANK31q1rrV4GZ7tuEEELYadMmqKkxgUFKELNZnDYPbv16OHAA+veHc881t7WFMmpzAdywYQ3vF7ElltuIeOnudbmbHQMRQgjRSLALGCxOK6Fae4XOnOkpN7aF4Ka5Pn1WQNcWfsb2yOEZOH9WoT4KrFFKLcGsQD0buNvWUQkhhPDMfwt0AYPFqQHcrFmwf7+5bM0fi1VlZc336bMCuFj/GdurWA/gtNZvK6WWAqdjArg7tdYH7R6YEEK0e+HKwDlhDlxRESxfblZjTp0KX31lbo/17JQ1/szMpn362lKWsT1yeADnzyKGy4AKrfWHWusPgCql1KX2D00IIdoxrYNvIWJx0hy4hx6C2lqYPh26d2875cWWtjlLT4cOHUyT3+PHIzsuEZrqavNvFh8f3PzTCPBnDtz9Wusy64rWuhS4374hCSGEYNcuKC+HtDRzCoZTSqhbt8Lzz5sM1WPuJgYZGebLce9eqKqK6vBCYpVHGy9gAPPzDh1qLu/YEbkxidB5Z98cuhGVPwGcr8f4M3dOCCFEsELNvoFzArjbbjPZt+uvh5wcc1tiognitDbBaqxqbgWqxcrMyTy42OLw8in4F8CtVko9rZTKVEoNVUo9A3xj98CEEKJdC3UBA3hKqNGcA7dwIXz8sSlDPfxww/vaQhm1pRIqtI2fsT1yeAsR8C+A+y1QDfwL+B+gCrjJzkEJIUS7t26dOY/1DJxVMr3nHujbt+F9bSG4aS0D1xZ+xvYoBjJw/qxCPQHcBaCUigeS3bcJIYSwi1VyGzEi+New5u+UlJiGwImJ4Rmbv1wu+Pprc/nnP296f6y32Sgvh0OHzK4SAwb4foyUUGNTDARw/qxCfUsp1VUplQxsBLYqpW63f2hCCNFO1dV5tpjKzAz+deLjPSUga2PuSNq506zk698f+vRpen+st9mwFib4aiFikQxcbGoLARwwUmtdDlwKzAUGAVfZOiohhGjPCgpMG4P+/SE5ObTXiuY8uNYWYsR6cNNa+RRg4ECT+dy/H05I8SpmWAc8MR7AJSqlEjEB3Ada6xpA2zssIYRox/wJDPwVzXlwrS3EyMgwmas9e0zAGmtaaiFiiY+XViKxqI1k4F4EdgPJwDKl1GCg3M5BCSFEu9ZWArjWdpLo0MFkqFwuE8TFGn/L3LFeKm6P2kIAp7X+m9Z6gNb6Aq21BvYC54XypkqpnkqphUqp7e7zHs08bqZSaqtSKl8pdZfX7U8qpbYopdYrpd5XSnUPZTxCCOEoVmanudYUgYjmbgz+tEIZNMicFxTYP55ws8Zs/QzNifVScXvURtqINKCN2hDf9y5gsdY6G1jsvt6Ae8Xrc8AsYCRwpVJqpPvuhcBorXUOsA24O8TxCCGEc9iRgYv0HLhDh+DAAdP/zSoh+jJwoDmP5QDO+hmaE+urbdujtpCBs8klwOvuy69j5tc1Nh7I11rv1FpXA++4n4fWeoFXELkCSLd5vKKNq3PVYRLMQjhAlEqoWmvqXHWhvyd4yqdjxza/QhNiN4DT2v8ATkqosUcCuGb11VofAHCf+1hfzgDA+390ofu2xq4D5oV9hKLd+Gb/N3R7rBu3fHJLtIcihJkP5t2eIlQBBHA/+fdP6PtUX3YUh2Gyvb9bgcVqAFdWZlqkJCdD91Zm8UgJNfa0hQBOKdVZKfUnpdT/dV/PVkpd6MfzFimlNvg4XeLn2HztHtsgRaKUugeoBd5sYRw3KKVWK6VWH4n2foDCcbTW/HbebzlRc4L/s/L/sObAmmgPSbR3hYVw8qTZwD4lJfTXs/qvHTrU4sPm58/nnQ3vUFRZxO0Lw9Dqs60HcN7Zt9Y2Ox80CBISzL9tRYX9YxOhqakxTZrj4qBr12iPpln+ZOBeA04CZ7qvFwJzWnuS1nqq1nq0j9MHwCGlVD8A9/lhHy9RCHjnpdOB/dYVpdQ1wIXAT3ULtS+t9Uta63Fa63G9rSNRIdz+tfFffFX4FQAazexPZkspVURXOMunAP36mfODB5t9SK2rllsX3Fp//f0t77N099LQ3tdawNDcClRLWwjgWpOQ4JkHaK1cFc5VUmLOe/RoufwfZf6MLFNr/QRQA6C1rsR3diwQHwLXuC9fA3zg4zGrgGyl1BClVBLwY/fzUErNBO4ELtZay+GMCEplTSV3LroTgCenPUlqp1Q+2/MZ/9nynyiPTLRr4Q7g0tLMeQsB3EvfvMSmI5sY2mMo95x1DwC3fHJL8PPhTpyAbdtM4DJyZMuPbQ8BHEgZNZbEQPkU/AvgqpVSnXCXL5VSmZiMXCgeA6YppbYD09zXUUr1V0rNBXAvUrgZ+ATYDLyrtd7ofv7fgRRgoVJqrVLqhRDHI9qhZ1Y8w96yvYztO5bZE2bzwLkPAHD7wts5WRvqn7gQQQpnCxEwWYTERDNnq7Kyyd2lVaXct+Q+AJ6Y+gR/POuPpHdNZ83BNbyx7o3g3nP9ejPJf8QIs09oS1JTzWPKyuDYseDeLxqCDeBkJarzxUALEfAvgLsfmA8MVEq9iWn7cUcob6q1LtJaT9FaZ7vPi92379daX+D1uLla62Fa60yt9SNet2dprQdqrXPdp1+FMh7R/tS6anlq+VMAPD3jaeLj4rnxtBsZ0WsEO0p28N6m96I8QtFuhTsDFxcHffuayz7mwb265lWKKos4a9BZXD7icjonduaxKY8B8OgXjwb3nuvXm/PW5r+BmT8Wi1m4QAM4WYkaO6yWO7GegdNaLwQuB64F3gbGaa2X2jssIey1onAFJVUlDEsdxvlDzgcgMT6RX5z6CwDm5cvCZhEl4Q7gwFNGPXCgyV1zt88F4Nfjfo1yT8b/0egf0b1jd7YXbye/OIiAw8oyDR/u3+PbQwAnJdTYUVhozgf4anzhHP6sQj0bGAUcw2yhNdJ9mxAxa952E6DNyprV4Hbr+ic7PsGlXREfl2jnXC57ArhmFjIcrz7O53s/J07FMT1zev3tCXEJ9dfn588P/P0C/RnaUwAnJVTnC/TfNkr8KaHe7nX6E/AR8ICNYxLCdvN3mC+lxgHc8F7DGdxtMEcrjvLN/m+iMTTRnu3fD1VVpvVHONsXNLOQ4dNdn1JdV834AeNJ7dxwvo/1fyOobLQ/m7x7i7UALpAmvpaMDLOoo6DA51xE4SBtJYDTWl/kdZoGjAZabigkhIMdPH6Qbw98S8eEjpw9uGEyWSkV2heXEKGwI/sGzQZwVnZtZubMJk+ZkTkDgCW7llBVW+X/e3k3Im6rAdyRI6ZXX/fu0KWLf89JSDBBHMCuXbYNTYRBWwngfCjEBHFCxKRP8j8B4LyM8+iU2KnJ/TOzzJdZUKUjIUKxYoU5HzUqvK/rI4DTWtcfpMzKntXkKf1S+pGblktlbSXL9izz/7327TPBTSBZxFgL4IL9gpd5cLGhrQRwSqlnlVJ/c5/+DnwOrLN/aELYo7nyqeX8IeeTGJfIyn0rKa4sjuTQRHs3333QMH16y48LlDUHzmsRw9airewu3U1qp1RO63eaz6fVZ6O3B5CNDrR8Cp4vyr17/X9ONAX7BW+tRJV5cM7lcnkWMaQ7e5t1fzJwq4Fv3KevgDu11j+zdVRC2KTOVceCHQsAT6atsZQOKZw1+Cxc2sXCHQsjOTzRnpWXw5dfQnw8TJ0a3tf2kYGzMswzsmYQHxfv82n12egdAWSjrexSIH3svDNwsbATimTg2q5Dh8xWWqmp0LlztEfTIn/mwL3udXpTa/1lJAYmhB1W7V9FcWUxmT0yyU71+oI5fBhqa+uvWnOCZB6ciJhFi8zf4Jlntr45eqB8BHD15dNmMtEAZ6afSdcOXdlydAu7S3f7917BzOPr1s3s+1pZ6Wmi6mQSwLVdMVI+hRYCOKXUd0qp9c2dIjlIIcLFKgU1yL4tWWL6/Vx7bf1N1pygT3Z8Inujisiwyqezmg+ogmY18j14ELSmqraKz3Z/BtCgfUhjifGJTBs6zQzP3zmhwS7EiKV5cFJCbbvaQgCH2Sj+IswuDPOBn7pPcwFpUy9iUpP5b3V18Pvfm8zHm2/CMjNZe1TvUfTo2IODxw+y79i+aA1XtBdawzx3tnem79J+SDp3NgsKamqguJgNhzdwsu4kw3sNp09ynxafah3s+J2NDmYOHLSPAG7wYFMi37vXLPQQztMWAjit9R6t9R5gktb6Dq31d+7TXcCMyA1RiPA4cuIIq/atokN8B87NONfc+Mor8N13ZrshgNmzweVCKUVumtkGaO3BtdEZsGg/Nm40E6f79vVv+6lgeDXztf6m89LyWn2aFcAt3rm49T2Cg2khYmkPAVxSkgnitJZWIk7VFgI4L8lKqcnWFaXURCDZviEJYY8FOxag0Zw9+GySk5LN5tn33mvufPVVU0b99lt4w2zgbX25SQAnbGdl32bM8BxMhJvXPLhAArj0rumM6TOGEzUn+LKglSnQBw6YeWy9egU+j2/QIHPu9ACurs60SoHgVinKjgzOZv39WX+PDpbgx2OuB15VSnVzXy8FrrNvSDHs88/hqafg7bcdv3rFL1qbDNWWLU3vy86GG24wG1HHiCbl0z//2TTknDwZrr7afHFefTX88Y/wgx/UZ+DWHFwTrSGLtqygAF54wZTSPvrI3GbH/DeLVwC35qT5m7b+xlszM2sm3x3+jnnb59XvHexTsOVTiJ0M3MGDJojr0wc6dgz8+dnZsGCBLGRwqhjKwLUawGmtvwHGKqW6AkprXWb/sGKQ1mYu1Zo1Joi7775ojyh0//0v/PKXzd/fty9cemnkxhMCl3bVN/CdmTXTfGk+95y58+mnTSD605/C3/4Gq1fDE0+Qd/MVgGTghE0efNAcIFkSE2HaNPvezx3AuQ7sZ121aeXpbwA3K2sWTy5/knn583hy+pPNPzCYFiIWKyw2nqwAACAASURBVOOxZ0/gz42kULJv4Pnd+DowFtHXFgI4pdTPtNb/VErd0uh2ALTWT9s8ttiilAkEzjsPHn8crr/elORiVU0N3HqruXzNNTDaa/ONzZtNyfG220zGoEOH6IwxAN8e+JYjFUcY3G0ww3sNh8WL4cQJyMmB0083D4qLg2eegbPOgief5JTrrqFDfAd2luykrKqMbh27tfwmQgRi9WpzPns29O8P48aZ3lN2cQdw+Yc2c6LLCQakDKB3cm+/njpp0CS6JHVh45GNFJQVMLBbM19uoWwFlplpzq05dE51+LA579Py4o9m5eSY83XSD99xamvNNAClYuL7u6XJFtY8t5RmTqKxc8+Fyy6Digq4555ojyY0zz8P27aZo8WXXjLBmnV64QUYPtx80P7979EeqV+s9iGzsmaZgxCrZUPjFX+TJ8MVV0BVFYn33MfoPiZwXXdIPmxFGJ08aRYuKAUPPWT+X517rr3v6V7EsPaYKXPm9Wt9/pslKT6JKUOmAK20EwmlhJqebib5HzgAx48H/vxIOXrUnPf2L/htYuxYc75+vSnFCufYv98sxElLMxlxh2tpFeqL7vMHfZ0iN8QY88QT5h/+9dfhm2+iPZrgFBeb8g7AX/5iPlS9JSaa2wEeftjMI3M4qwVCff83a9K4rzlHjz9usopvvUVevDkKkzKqCKtNm8zRfna2/5uhh8qdgVtbY0pEuX0DW+1qzR1tcVeGUEqo8fEwdKi57OQsnPV516tXcM/v2dOsRK2sNAfJwjliqHwK/u2F+oRSqqtSKlEptVgpdVQpJVtpNScrC373O3N59uzY2BamsT//GUpKYMoUuPBC34+ZNcvs11hWBnPmRHZ8ASqpLGHlvpUkxiWaCdh795ov0JQUmDix6RMyMsy/HZC7aAMgCxlEmK11HxDk+Z8FC5k7gFuTWAT4P//NYh38LNyxkJq6mqYPqK0NrYQKnsDPyRP8rQAu2AwceFrFrA3hwLCmBr7//ZbnKYvAtLUADpiutS7HNPYtBIYBt9s6qlh3773m6Ozzz+F//zfaowncArNXKPff3/wqU6v0A/Dxx5EZV5BW7luJS7s4I/0MUjqkeMqnU6Y0zS5a7r4bOnQg98udgGTgRJitcR8Q2NXzzRcrA9f1BBBYCRVgcPfBZHTP4Fj1MXaU+MiQvfiimT4ybBj06BHcGGNhq6lwBnBrQjgwfPFF+Pe/4eWXnb/wI1bs3WvO21AAZxWCLwDe1lrHwEZ1Uda9uye4ueMOqKqK7ngCobXnw3PMmJYfe+qpJgDaudNk4hzKCr5OTTvV3NBS+dTStSuMGUPOIVAoNh7eSHVdtc0jFe2GlXmJZADXqxcHu8ZxMFnTtUNXMrpnBPwSw1KHAZBf3CjAKikxB3wAjz4a/BhjoUdaqHPgwJN5DTYD5/37Bs9BqQhNG8zAfaSU2gKMAxYrpXoDMRSRRMkvfwmjRplu23/7W7RH479AGnEmJnpWp6537va4VgCXm5YL1dVm03Bofcui3FxSqiErrhc1rho2Hdlk80hFu+ByRaeEGh/P2uFmJXVujxHEqcAbBmf1MAFWkwBuzhwoKoJzzjELuYIVSyXUYOfAQcMSajDTbB5+2MxVTnavNZzn5zZnomVtLYBzb511JjBOa10DVACX2D2wmJeQ4JnoP2cOfPihOUr67rvojqs1ga4is76AQikF2Myav5ablgvLl5sVbiNHtt5p2/2z5ZabpsxSRhVhsWsXHDtmSprWJvMRsibDNJ7N7ZgR1POzU02Atb3IK0O2fTs8+6ynlVIozb3bSwl10CBTZj5yxKx89Me335rvkHff9fy+33nH3Ld4sTk4FaFpawGcUqozcBPwD/dN/THZONGaGTNMme7YMbjkEnM51x1EOFWgq8jCMRnXRserj7O9aDuJcYmM6jMKFi40d/izYbj7Z8vbY/Z/lABOhEU0sm/WW/c12Z68uuB6mGX1dGfgSrwCrEcfNRPqr73WTKsIxcCBJrO/b5+ZT+dE4QjglArss/OLL+C008x3yI9+ZBaM/PznZpHZ6NHmoPTLVrY5E61rg3PgXgOqAWu5XiHg7GWHTvL883D55SaYy8sz5ZM//MGcO1Ggq8jCMRnXRusPrUejGdl7JEnxSZ7WLpMnt/xEMA03lWLERtO4c1uRLPkXYRCN+W/WW3erNG9dElzz7foAziqhulyeRUy33Rby+EhIcHYrkZMnzQF5fHzge702FkgA9+9/m/Phw813yY9/DI89Zm6b2ag1kgjOgQNmfmPXrqaxdgzwJ4DL1Fo/AdQAaK0rgdjZADPaMjLMStT582HZMtNMc9UqeOutaI/Mt0BLqFZTyo0bHZnCbzD/TevAVv916QLZ2WQdNcF2k3k/QgQjGitQcWejE8pJrIORO48F9RpDug8hTsWxu3S3WdTz7bcmIzVoEIwYEZ6BOrmMai1g6NUr9H2gAzn4tYKzF1803yVvv+3JAFqLsSSAC433gVWM7PHtTwBXrZTqBGgApVQmcNLWUbVVXbrAI4+Yy3ff7cwSQaAl1JQU84FbU2N6qznMmgPmwzEvLc9sQn34MHTrZgJrf+TmMrTEXNxVuotaV609AxXtR5RKqFY2etRhSMrfFdRrdEjowKBug3BpF7tLdzdc0R2uLz0nr0QNR/nU4u9K1N27zb6pXbvCmWc2vX/yZLOYYcMGKCwMfVztVRSnNgTLnwDufmA+MFAp9SawGLjD1lG1ZddcY/5ACgtN6vsnP4Ff/MKzQXI0ebcQCaQRZ6hL4m209pBXBs478+Hvl01eHp1rIL2uC7WuWvaUSr8lEYIjR8z/9eRkz96fEWIdzOQeJKTsVoMyanNb0oUiFjJw4Qjghg83bZh27IDy8uYfZ/2Op071vb1TUpLpaen9WBG4KGXGQ9FiAKfMzvVbgMuBa4G3MatRl9o+srYqLg7++ldz+fPPTSr8lVfgppuiOy6AQ4fMBu89ewbWiNOhCxlqXbV8d8is+h2bNja4Iyz3z5ZVHg9IGVWE6KuvzHlenvksiCBrOkHeQUxWJ8gpD9k93StRC9bBihUmqLACiHBwciuRcGbgEhM9U1Csvwtf/OlbOW1a668jWtbWMnBaaw38R2tdpLX+r9b6Y6310QiNre06+2yzqujNN82eqcnJ8MEHsGRJdMdllSwC3cfQoa1Ethzdwsm6kwzpPoTuHbsHN3nc/bNlFZhytwRwIiTWl/H06RF/6/pstO5rFh/s3h3U69Rn4DZ8Zl5n8mQzlSJc2ksJFTyBV3OZs5MnTYsQaDnLaWVzrVWUIjDHjpm/t8TE8M3ljAB/DgFXKKVOt30k7c2kSaZ8evXVZj4cmP036+qiN6Zg9zEMtSmlTRosYIDgUuR9+0JaGtmHzN6P24sd+KUiYoPWngAunCVHP9TU1Xiy0d2HmxuDzHDVB3D7zD7BYf9ZBg82q1ELC01TcScJRxNfb62tIP3yS1MVGT0a0tObfx2r7YXVx0wExmpEP2pU89srOpA/Adx5wFdKqR1KqfVKqe+UUs5tux+LbrnFrOJatw5eey164wg2gEtLgz59zDyOXcFNjrZDgwUMx46Zny8pKfAjrNxcstwbyEkGTgRt61azZ2Xv3qanVyTfumgrJ+tOMrTHULoNdf/9BxnA1ZdQaw6aG1oq7QUjIQGGDDGXd+4M72uHKpxz4MAsSujWzfxt+Prs9Kd8Cg0DOAcdRMeMGCyfgn8B3CwgEzgfuAizqf1Fdg6q3enUCR5/3Fy+916TNo+GYEuoSjmyjNpgAcO6debGYI6w8vIkgBOhs76MZ8yI+Py3BgczIS4SGNJjCArF7pQ6atL7e7bTCyenllHDXUJNSGi5jOrvIpFu3UwZu6LC7JMqAhODCxjAv6209lgnYIbXZRFOP/qRCS4OHTL94qIh2AwceD7Et24N33hCtOXoFgCzA0MoR1i5uWS6A7idJTullYgITpTKp9BoOkGIwVHHhI4MTEilLg72TB5tT88sKwMX5Dw924Q7gIPmy6gFBaY1SJcu/jUelzJq8NpwBs7br2wZhTAfghe5E5vRWAoebAsRi8OW/p+oPsH+Y/tJjEtkULdBoXW/z8sjuQb6n4ijxlVDQZl8QIoAnTgBn31m/p/PmBHxt2+wH3AYVnlmYVap5/cLbkeHVjk1GAn3HDjwBHCfftqw+mJ9D0yZ4l/VwKm/M6erqfHsUW6tCo4RgQZwsdGeOFZFc0uUw4fNPLHu3U0bkUBZXwoOKXnsKDHb8AztMZSEuARPijyYI6zMTOjShawjsiODCNLSpaZtx+mnh/fL3w9aa08LkbQ8s1WVUia7VVMT1GtmVyUDsL2HTVsCOjUYCfccOIABA8y2fSdOmO4ElkB77Dn1d+Z0W7aY/5uZmaZZcgxprQ9cnFLqCq+bZO6bnSZONH9Amzebyc6R5L0DQzAlEYdl4KwgK6une5eIDe4Vczk5gb9YXByMHUu2zIMTwYpi+bSgvICSqhJ6de5F/5T+0LGjWdFYWxv050xWeQIA+Z2rwjlUDycGIy4XFBWZy6mp4X3txgfvNTWwaFHD+1rjxN9ZLIji3sShaq0PnAu42eu67NNhp8RE020bIl9G3ebeqD2Y8imYL4SkJLNd1fHj4RtXkBoEcJs3h36E5bUSVVqJiICcOGH2Q4bwr9j0w7qDZgFPblouyjo4C7GMmnXEtDvKjy8LeXw+OTEYKS42QVyPHr53RAiF9Xfx7rumdcry5WZV//Dh/m/758TfWSywFri1tQDObaFS6jal1EClVE/rZPvI2qtolVGtP+IxY4J7fny8p5mkA7Jw24tMkJXdMzs8E1SllYgI1hNPmAObceNg/PiIv/22InNwNjx1uOfGEDPm2YWmsfX22sMhja1ZAwaYSsCBAyZT6AR2zH+znHWW+ewtKDA79VgH8IEE/BLABcdKXgwf3vLjHMifAO464CZgGfCN+7Q6lDd1B4ELlVLb3ec+921SSs1USm1VSuUrpe7yuv1hd0+6tUqpBUqp/qGMx1GsAG7x4qC3uglKKHPELA4qo+aXeGXgwpEil1YiIhgFBfDkk+by009HvH0IeP5es1O92gOF+H916A7TqmJXxX57VmUnJZn+ki4X7N8f/tcPhh3z3yzx8fDMM+byn/8M//M/5rIEcPbznj4UY/xpIzLEx2loiO97F7BYa50NLHZfb0ApFQ88h+lDNxK4Uik10n33k1rrHK11LvAxcF+I43GOgQNNO5Hjx00X7kjQOjxBjpMCOO8SajiC01GjyCo1/112lOygzhXFHTNE7Lj7blMS++EPTZYlChoczFhCWXRUV0eng0dJL4NaXcveMpu2b3JaQGJHCxFvU6aYTgTHj5sN7jt3Duxvxvp9FRaawFe0rq7O/K7BU0GKIa0GcEqpzkqpe5VSL7mvZyulLgzxfS8BXndffh241MdjxgP5WuudWutq4B3389Bal3s9LhloW62nraOuX/4Szj0XLrvMlGDssmuXmW/h3jYqaA5ZiVpRU0FheSEJcQkMDrWFiKVjR7pkjyLtGFTXVVNYLtNBRSu+/trsd5yU5GnUHQUNDmYsoRxsHTkCLhfZx8w8MGu6Qti1twAO4KmnTHNfgPPOMwtO/NW5s+kgUF3tGato2b595veVlmb67cUYf/L5rwHVwET39UJgTojv21drfQDAfd7Hx2MGAN7/cwvdtwGglHpEKVUA/JQWMnBKqRuUUquVUquPxMof9fe/b8537DC9o/7zH7jjDvveL1xNDB2SgdtZYrbfGdJ9CAkF+6C01Hzo9usX2gvLPDjhL63N3sZgzq3GtBF2svYke8v2Eq/iyeie4bkjM9NMxM/PN5PzA+E+mMyqMV94tv1fcGoAZ2cbmGHDPH83V1zR8mN9sX5nsqm9f4Ldfcgh/AngMrXWTwA1AFrrSvzoB6eUWqSU2uDjdImfY/P1HvWZNq31PVrrgcCbeK2UbfIErV/SWo/TWo/rbeeRUzhNmAAbN8KSJfDBB+YI/v/9P1i1yp73C9cyaocEcA0yDt7Baagd4/Py6luJyEpU0aJ33zUrCfv0gT/+MWrD2FW6C5d2Mbj7YJLivZrBdupkuvu7XLBwYWAvagVwygQy7SaAs3MOnLfHH4dNm+CqqwJ/rtN+Z04XSvN6B/AngKtWSnXCHTwppTKBVjfr1FpP1VqP9nH6ADiklOrnfr1+gK+lTIXAQK/r6YCv2axvAd/34+eILSNHmvLpxRfDH/5gbps9256NisMxRwxg0CBzVL9/v2mdECU+V6CGY4m4ZOCEPyorPRnzOXOi2hzUZ/nUYk3VCLRl0YED5jU7mrVjth3MOC0YiUQJFcyB5ogRwR1wOu135nTtIIC7H5gPDFRKvYlZdBBqPe9D4Br35WuAD3w8ZhWQrZQaopRKAn7sfh5KKe9858XAlhDH42z33GM+NL78Et57L/yvH64gJz7edHkHz8TQKAj7AgbL2LGeAM6ueT8i9j3zjClh5eTAdddFdSgNDmYas1a8z58f2KR3dwYuu5v5v95uMnCRCuBCMWiQOXfK78zp2noJVWu9ELgcuBZ4GxintV4a4vs+BkxTSm0Hprmvo5Tqr5Sa637fWkxp9BNgM/Cu1nqj9Xx3OXY9MB34fYjjcbauXeHhh83lO+6AqjB2Pz961KxaSk4Oz1GIA8qoYW8hYunZk+wkM48u/+Cm0F9PtD0HD8Kjj5rLTz9tDmqiqMUM3OjRpt/awYOwfr3/L+oO4Ib2GQaYOae2rMp2WgBnlVAjvBVaQJz2O3O6dpCBAzgHmAKcB4S8Fl5rXaS1nqK1znafF7tv36+1vsDrcXO11sO01pla60e8bv++uxybo7W+SGu9L9QxOd7115sP3N27TaPHcLECnLFjw9OjKgwbZYeqPusQ18tkQjp3DtsRVuaA0QDsOLYXl5al+qKRe+81bSAuusi0hYgyny1ELEp5yqiBNA53B3DJ/QbTP6U/Na4aCsptCBjS0syKzMOHG27yHi3WNloSwLUNLldMtxAB/9qIPA/8CvgO2ADcqJR6zu6BiUYSEswRPZhGj+FqK2KVGMO1jYh1JBOlViKVNZUUlBcQr+IZvMs0GyUnJ2yZkK5DR9DnOFTpavaVt/3jBhGAtWvh1VfN/9Wnnor2aIBWSqjQsIzqL/ccOPr1q39dW1qJxMdDf3eP9kIHtO2xVuv2dPBGRBLA+W/fPlPN6tMn5jaxt/iTcjkHmKG1fk1r/RpwAXCuraMSvk2bBt/7Hhw7Bn/6U3heM1wtRCxRLqHuKt0FQEb3DBLXuzewD+ced1lZsqm9aMpqG6I13HyzaQcRZdV11ewp20OcimvYQsTb1Kkm4PzySyjzc19T6+AxLa0+s9fm58FVV5vMakICpKREdywtsbYg27/fOVuQOVUM78Bg8SeA2woM8ro+EAhgwoQIK6vR4yuvePYvDUU454hB1AM4KxMQ9gUMluxs2dReNPXBB7B0qcnO3OeMjWF2l+7GpV0M6jaIDgkdfD+oWzeYONF0pF+0yL8XdkAAt6N4B2999xbajlX5vnhn30JtR2SnpCTTkN3l8mRKhW8xPv8N/AvgUoHNSqmlSqmlwCagt1LqQ6XUh7aOTjQ1fDj8+tfmSP+5ECvZpaWwZYsJCEeNCs/4Bg82r1dYCBUV4XnNAGwt2gq4S0Yb3Bm4nJzwvUFWlrQSEU1Z+53efz/08Lm1c8S1Wj61eO+/3JoTJ0wFoEMH6NbNU0KNYCuRk7UnmfHPGfz03z9lwY4F9rxvY7FQPrVYK1H37InuOJyunQRw92H2I73ffboAeBj4i/skIu2HPzTn334b2ussWmSO1CZONI09wyEhwdN1fufO8LxmANYdMlnJnL45ng+woaFu3etl8GCyS8wReP7RreF7XRG7iothxQrTA/HnP4/2aOq1uALV2xlnmHMrY92SQ4fMeVoaKBWVDNyzXz/LjhIz+fzjbR/b876NxVIA54BWTjEhxluIACS09gCt9WeRGIgIwNix5vy776CmxnxxBMOauGwdgYdLVpb5z5Gfb1bORtDag6YknNvtFCgpMZmCcPZtSkwkq9MAoJD8Q227/aDw08KF5kDonHMcNT/K7wDO+jxZv96UUlta8OO1gAEgs6dZvbejZAd1rjri48LcNqVRAHfkxBEeXvZw/d3zdwTYhDhYsRTAOaATQExoJxk44TRdu5plz9XVpgQaDK09AZzVSiBcovQBUlFTwZajW4hX8Yyucq8qSk8P+5yVrN6nAJBfvltaiQhPC45w/z8KkVXWbDWAS001gVJFReurx73mvwF0SepCvy79qK6rprDchpWijQK4+5feT/nJcqZnTqdHxx7kF+dHZipDLAVwDujF6XhaSwAnoshadOBP2cOXDRvMMuq0NM8ReLhEqZXIhsMbcGkXw3sNp9MBd9f0gQNbflIQug0ZQe8TUKmrOXBMJgq3ay6XfZnsEFmBTatz4MCz0Mda1NScRgEceAJEW+bBDRzIry6EU87bwCl/P4UXv3mReBXPMzOeYXrmdADmbQ+gh12wYjGAi1Irp5iwf7/Z8q53b7OQJ0Y1G8AppZptjKKUGtTcfSJC/P3AbY6VNZg5M/yrqqJ0BGiVT/P65XnmzNgQwHmvRJWFDO3cunVmXtiAARGfLtCSmroadpfuRqEY0mNI60/w94DQRwB3SqrJSG8+sjmYobYoP76MF8fBth4uthVtw6Vd/O6M3zGy90hmZpmAOSJlVKuJbywEcN4VkEit0o01K1aY83At3ouSljJwS60LSqnGy5P+Y8tohP+sD9xQAzg7yj5RKqGuOWC+fHL75tobwHmtRJVWIu2c9/8jB7WX2F26mzpdx8BuA+mY0LH1JwSagXPPgQPITTOfRWsOBlkNaMG8fBOcXbwFtsz8L/m/zeep6aZJshXALdm1hKraMG4v6IuVgUtNtfd9wqFXLzPNpqzME3iKhqz/t9OnR3ccIWopgPP+NGp82OGcT6r2yvuIOdCjrGPH4IsvzNZZ06aFf2yDB5uJ0AUF4d23tRVrD0UoAyetRISlLZRPwf/PE2sRg1cGzgrgrAx4OFnZtcs3wyllCWT2zCROma+ttC5p5KXlUVlbybI9y8L+3g3EUglVKSmjtsTO+d8R1lIAp5u57Ou6iLT+/U39vrTU7PcZiMWLTZfuCRPC1rOquLKYJbuWcPjEYbMqNiPD/EeJUCuROlcd6w+Z/tJj+461N4DLyPBqJbIt/K/fSPnJclYWrmRl4UpW7Vtlf7bBSSor4euvYeVKczp2LNoj8igtheXLzcHK1KnRHk0Dfq9AtQweDN27w5EjLTeA9VFCzembg0Kx4fAGquuqgx1yE1W1VSzZtQSAGTvwuRuDlYWzfR5cLAVw0GQai9aaXSW7Itf42AkOHvR8bqxZY1ZYg73zvyOspQCuj1LqFqXUrV6Xreth7MsggqJU8AsZbDj6+MWHv+D8N85n4Y6F5oYIl1G3F2+noqaCgV0Hkto51fNhP8iG6ZpJSWR1MCWk7Qc3hf/1vdTU1TDp1UlMeGUCE16ZwPiXx3P1+1fb+p6OcsUVpk/ZhAnmdO65ng/iaFu0yIxl4kTHTYT2ewWqxfvzpKUy6v795twrgEvpkEJWzyxqXDVhnQe3bM8yKmsrydN9STuOzwBuVpb5DHt/y/thDR6biLUArtHn7yOfP8LQvw3l9XWvR3FQEbRvn2l6b31unHoq3HOPuc/O+d8R1lIA93+BFKCL12Xr+sv2D020KpiFDFo3/AMOkybzYCKcwm+wgEFrT1bSjgwckNXL3Urk2G5bj2r/sfofbDi8gZ6dejJ+wHgAPtr2EZU1lba9p2OUl5uDjbg4GD/eZIi+/dZsGO8EDi7DBFxChdYDuOJik53r1Mks2vB+qg3z4Kys2qwu7s85HwHcmQPP5JTUU9hTtofnVz0ftvduItYCOK/P371le3nk80cAeHvD21EcVAT98Y9mDuDAgXD66SZQe/pp833k4P+3gWo2gNNaP6i1fhB4zrrsfYrgGEVzglnIsGWLCW769DFHJWGSl2Y+ZOvnwUR4JWqDBQylpWbLny5dbMuM9BgygtQKqHCd5ODxg7a8R3FlMQ8sfQCA1y55jZW/WMlp/U6jqraKz/a0g/7aVql/4kRTBnnhBXP7vfea4C6aHD6PJuASKrSe0bf2Xs7JadLst8n//zCYl28CuJkDzzM3+AjgEuIS+Mt0syHQg589SFGFTZP2Y2kVKjT4/L1r0V310y4+2/0ZFTWR3+IwolavhjfeMPvCLl1qpmBcc41pen/zzZ753w6b9hCMltqIXKiUOgKsV0oVKqUmRnBcwh9WBi6QEqqVfZsxw/wRh4n3EbjWOuIl1GYXMNiVIo9AK5GHPnuIkqoSzh9yPhcNuwjwlIwi0vsq2hqvlL7iChPMHT4Mjz4avXGBo+fR1Lpq2VW6C4ChPQLYRq61jL71OWM9zku4FzLsKtnF1qKtdOvQjTOHuxda+QjgAC7IvoBpQ6dRWlVaf8ATVjU1Zu5lXJxZ3RkL3AHciuNbeHvD23SI70Bmj0xO1p1k6e6l0R2bnbSG2bPN5d//3rOt2COPQHIyLFhg/j0nTIidYLwFLX2D/xk4S2vdH/g+EOVPTNFEdrYpZ+zd60nxt8aG8ilAetd0UjulUlxZbDqyRzADp7X2ZODSbG4hYrG5lcjWo1t5btVzxKk4npnxDEopqKlh5irzplZ2os3yVepXCp55xlx++mnYtSs6YwNHz6PZU7qHWlctA7sOpFNiAHscDx9ushb5+b4Xi1iBnZWp8+IdwIVjSsF8d/uQaZnTSBjs7mNXUOBzhaxSiqdnPE2ciuMfq//BpiNhnpdaUmLOe/QI60GvXT7f8zk3f/swN12SyM/PN/+Ot555K1eOvhLw/G4j6qOP4N//tv993nvPZNh69/bMeQOz6O/OOz3XHZg1D0ZLf421WustCD/XUwAAIABJREFUAFrrlZj5b8JJ4uM9ZdC33mr98SdOwLJl5gsnzP1vlFINj8IzMsyH3d69cPJkWN+rsSMVRzhScYRuHboxuNvgiAVw2e6qih0ZuNsW3katq5br864np2+OufHZZznjj8/T/aRie/F2dhS34c2qN22CwkLo27dhwDB+PPzsZ2YbOe8P5Ehra+VTMMHbmDHmstXo1FsLGbh+Kf3om9yXspNl7C7dHdj7+mC1D5mZOdNMg+jSxXx+lZb6fPzoPqO5Pu966nQdr615LeT3byCGesCVVpVy2b8u47lVz/N8Xg1bekNah1TumnwXs7Ld2ftIH/yVlsLll8MPf2iaXtvpb38z5w8+2HT6zK23eha1XXihveOIEH9Wod6ilLrFx3XhBLfdZs7vv99zpNicJUvMF9/pp5tmj2FmzYNZc3CN+TIYPNhsNWRzpmR7kcmADUsdZjJVkQjghgwhy6ZWIot2LuLjbR+TkpTCw+e5N+4+ehQeeogEF0zfbrIQUTmSjhTvDFfjrMejj5rM8//8jznajjTvPooOnEcTdAAHnp/nk08a3l5VBZs3m5+5mR0nwrWQ4WTtSRbvNL3jZ2a5M5yN9kT1xZpmEPaGwjG0gGHOsjkUVRYxrv84/r4/l7//F5b0vZOUDimMHzA+svvHWhYtMnNZXS5TwrRLSYlp65OQAD/9adP7O3eGzz4zc2t9ZJFjkT+rUK1T4+vCCS65xLRWKC6Ghx9u+bE2Zw2azIOJ0Dy4Jl9YkQjgOnQgK8m0UghnK5FaVy2zPzFzOP541h/p26WvueOBB8yqqlGjmOn+dc77LgIliWhpqdSfng63324uz55tvhgiafFiR8+jsUr6Aa1AtVifDfMaZWk2bjQtU4YPN1+EPoRrIcMXe7/gRM0JxvQZw4Cu7tWufgRwTebhhkuMBHD5xfn8beXfUChe+N4L3JQ6i5tWwfC9ZtFCQlwC0zLNfMKIHvx5/y01/rsKp0WLzGfBpEnNz1XMyIDzz7dvDBHW6irU5k6RHKRogTUvSCl49lnY1kw2yHtOkc0BXKRbiTTpeRWJAA7I7jUMgPzy8LUSeeXbV9hweAMZ3TP4w4Q/mBs3bTIrMOPi4J13mDnmMgA+3ftZ22zqe+wYfP55yzuF3HGHmdeyejX885+RHZ+Dy6cQYgZu4kRISTF/c94Nwq3yaQuZi3Bl4KzgwlqwA/gVwDWZhxsuMRLA3b7wdmpcNVyTew2n9T/N5+dv/SKoSJVRvVdrg8ns2tXH0ebvNydqcUamUmqWUmqZUuqoUuqIUuozpdQFkRqc8FNuLlx3nUlTW5kJMP95rrnGfCCnpJhdEVJTYdw4W4ZxSq9T6JjQkd2luymtKo3YQoYmPa8iFMD1HDKSHpVw3FVpdqAIUUVNBX9a8icAnpj6hGcPy9tuMx96N9wAo0fTb85fGXtIURlXx6en9jD/tt/7XtvZuHrhQpPhGj+++XlHycmelah33237PMt6XgdCa84cQvrT6aQ8mtLkdONHN0ZmPD6EFMAlJnrKqN5fvC0sYLCEayWqFVxYc7YAvwK4JvNwwyUGWoh8ufdL/rPlPyQnJvPI+abnm68AbkbmDMDsH3u8+nj4B3L0qPkbedCd4/nuO9P8uV8/GDLEBMOrV4f/fb0DRYdta2enltqI/BJ4GHgAGApkAg8CDyilbojI6IT/5swxE30//BA+/dTc9t//mn44x4+bCcAA11/fpIdTuCTEJTCmj5kEvfbg2uiUULU2k9/B9gAu3CtRP931KUcqjpCblssPRv7A3Dh/vgkYunaFhx4ytw0axA8HmEUoD0yownXiOMydaz4sY11dnefD/wc/aPmxP/uZKent3x+5uXDuPoq6T29u3v08+47t43j18Sanl9e8bF9PshbUumrZWWK2r8vsmRnci1hfgN7lrhYWMFiyemaRnJhMYXkhRyuOBvXWBWUFbDyykS5JXZg40KtzlR8BHNjTUDgWMnDvbHgHgJtOv4n+Kf3NjRkZ5rzQk43sl9KPyYMmU1lbyeNfPB7+gbz3nukX+NBDsH59w6kQzZXnw2H9etNkul8/06ewnWgpAzcbmK61/lRrXe4+fQrMct8nnCQtzWQiwMwLqqoyq24AnnzSND49fhwet+E/rZcG82AiUELVWjcM4I4cMdmYHj1MlsZOYV6JavV2u+SUS8xijNpaz7/hn/5klsa7/f729+iXnMaqAfDWrya5X6ANtBZ59VXzYTxoEPzmNy0/Ni4OLr7YXI7Uz+5+n3cvH87ywuX0Se7DgVsPUH5Xef3p3IxzcWkXC3cujMyYvBSUFVDjqmFAygA6J/qeq9Yq64t28WKz6Mnl8jTxbSEDFx8XX79iOtgMmFU+nTp0KknxSZ47/Azg7GgoHAsBnJW1vHT4pZ4b+7rnzx461GCe6ONTzXfAU189xd6yAPfRbnUg7v+HLhfcckvDsqb1dzXfhvl33tk3h7X1sVNLAZzSWjdpLqa1jvxhpfDP7Nnmi2/9etMmZNs2GDYMfvc7U2azO6ChURllyBDzJbtnj/kisMHRiqOUnSyjW4du9OrcK2LlUyCszXy11p7SkTX356WXzFykzEz47W8bPL5LUhf+PNWUEO8asJkTicR+AFdebnZZAHOg0cmPHmZ2fin4Mm8elQlwR7pZuDLnvDmkdUkjpUNK/emCLDPLJBq9+kIqn1oGDoRRo8xcxOXLYccOk8FPT2919XqoJcz63RcyG5XBJAPXrO1F29lRsqPBdnsAdOhgxlxXZ0qbbhMHTuRHo35EVW0Vdy++O3wDqa42QT+YisHixWbVZ3y8mct63nmmO8HXXzcYT1i0w/lv0HIAV66UatJi3H2bjy6PIuo6dYInnjCXP//cnD/1lPlPEyENPkA7dDABpcsFu3fb8n7eX1gRayFiGTKELHfnllBbiWwv3s6u0l2kdkplXP9xpnfSffeZO594wvwuG7l67NWc2u9U9tUW89QkBV9+Gf0tpgJVXm7K/u+/D3/4g9ll4cwz4Uc/8u/5EyeaqQMbN7b65R4ydx/FZ86EvbVF5PTN4bq865o8zJq79Un+J7h0ZFfIBryJfXOsMuqLL8Jr7r5qfrReaNBKKEA1dTUs2rnIvH1WMwFcYWGLcz2bzMMNB4f3gbOyltMzpxMf12h6TJpZKc/Bhtv9PT71cTrEd+Ct795iRaGPnn8B+mDLByz5+Fmqq06YXoLWdA8wq7W7dzcJhLPPNv9+4WwnUl5uPvusQLEdaSmAuxX4UCn1gFLqIvfWWg8CHwDSB86prO2GAKZMiXjDwpy+OSgUm45s4mTtSdsXMjTJONi8iX0DHTuSlWDKFNsPhdZKxCqf1n8I/+MfZvL0OefAZZf5fI61SwPAk5MVFarWcwQcC7Q2f6+XXGIafVqBwl//6n8ZJCnJM+ne7gzkkiUcTKrm0XPMx+bT059u+oUJjOo9ivSu6Rw6cSi8pTw/bCsyBxIhB3BWJuOddzyLRVqY/2YJJQO3vGA5x6qPMaLXCAZ3H9zwzuRkMy3i5EkzTaIZ3vNw1x1cF/AYfHJ4Bq7ZrCU0G8AN7j6YW8800zNmfzI75FX0ty28jfO/u41v+mH+dn7zG1P9gYZZMTsy5kuXmukmVqDYjrTURuQL4Az3Y64FrnNfnuC+TziRUvD66/DLX8LLL0d8PkByUjLDUodR66pl45GNts+Da9Lzaod7dwJrDzyb1bcSKdsV0oeg1Xm+vnz60UfmfPbsFv8Nzx58Nqf1O40TCS4+G0zkSonhMHeuaSuQkgKXXmpOzz5rVp8GIlJl1Pnzufd8OJ7g4qJhFzFl6BSfD1NK1X+ZRrrZ8vpD6wHqg5ignXeeWdFu/btcdZVZBd2K0X1GE6/i2XJ0S8CbpjeZQtBYtMqoDl6FWllTWb+36YysGU0f0K+fOW8UwAHcNfku0rqksaJwRf0iiGBYjYF7VMczfh8me5uYCO++C7/+dcO5rFZmd/788PVv/PZbcz5pUnheL4a02EZEa31Qa30f8CvgRq31n7TWTf8ShLNkZZn5U9YqpAjL6+c1kdjmlahNMnDW+2QH0cQ0CKkZI+hWBeV1FUGvvPP+EJ6eOd0c8a9caT4E/Wg6Wd/bKRuThYqFdiI1NZ4FGg8+aEqo778PN98c+GtZXwqLFtk21xKtWbviP7yaBwkqnqemP9Xiw6OxbZHWuj7zZQUxQYuLM6V769/ljTfMHLhWdErsxPBew3FpFxsObwjoLev7v2WHFsCFfSGDgzNwy/Yso7K2kry0PNK6pDV9QDMZOICUDinMOW8OAHcuupPKmsqgxlBfwt1WR3xyF08gNXYsPP+8yZxaRoww02qOHPEEXqHyo0dhW9VSGxHlLp8eAbYAW9294O6L3PBELMrt61VGiXQJ1cr0ZYVYQvKTysoOuZXI0t1Lqaqt4rR+p5mdFxYuNEenZ51lslOtsL7w5p8Sb77cNoV5M287vPACbN1qAu2bbgrttQYNgpEjzaT7r74Kz/ga0du2MXv0PrSCm0+/iWGpw1p8/JQhU0iIS+Crgq/CNxerFXvL9lJSVULvzr09rSSiIJgy6v5j+1l3aB2dEztz1qCzfD8oGhm42lqzA4pSTffWdIBWs5ZWAHfggM+7r829lty0XArKC3j6q6dDG8N2zLSdluZcK9UwCxcOVo9CP0r8bU1LGbg/AJOA8VrrVK11T0xJdZJSStqIiGY1+AC1sYSqtW44abu21rPvaoRKqGRnh9xKxDqCrZ+43dI2Uj6MHzCe7h27s717HTt64PwyakmJ2RoMTIubcCyy8dW7LEQVZUf5zZ2juWx2f2Y+O56lQ6BnXRL3nftAq8/t1rEbEwdOpE7X1U/Mt5t39u3/b+/Mw6Oszj58P0lIAiQhIZCwhX0TFIIigoKWXbpoXWq19atVq63WtiK4IFpti0VBQe3mZzfXtmq1rfUTWRVBBATEALKEfQthh4Ql6/n+OPNOJmSSzD6T5Lmva66ZedeTNzNnfu+zShRLKbgTGQp8F1Bzt9req6O6jSIpoWbCDuCzgKsRhxsMx13iOz09bPUzg6Feq2UdFjiwpV9mjbPCbfrS6ewv2u/X+c+Wn+XDHR8CMH4bvmWBhrIe3JEjNu65efOIeV1iiboE3PeAm4wx7k7kxpjtwM2udYriFUfAfXHgCyq7dbV3XTt3WrdZCDl65ijHzx4nNTGVrJZZ9otcXm5dPbX0aww5PXvS2yXgNh7aGNAhFu9aDLjcp5WVfrdqSohLsPvicqMuXhzQOCLGW29Zt9Tll1fVcQuWb9hG5vz5z1U/ukHy5Kxr+UOLDfw7vYB5bW127y873kxG84x69rS4Xdv5kXGjOhYnR0BFC7cFrtB3C1ydgfgOnpmodeAZh/vloSCt0THsPt1xbAebj2ymVVIrhnYa6n2jOmLgHEZ2G8k3+36TU2WneGTRI36Nwe3CLYyjXTG+3XSOHm0bzi9fbm/mgsGpTzhgQEwK7HBTl4BrZoypEdRjjDkENAvfkJSGTnZKNu1T2lNUWsT2M/vtxFtRYevBhZAaJUQcN22E3KcAdO/OwEL7cq0fFgeH0opSvjz0JYJwYfsL7YRUWAgdO9paXD7iDprvSZVLIVZx7ry/+93QJdlccYUVhIcPwxNPBH24PRtX8HSFLcXzu5Rv806XB1hwwdPcfccffT6GY1H9YNsHoW2uXgshi38LEuf8eYV5VFTW3/eyvLLcXfS4VksS+CRGzh1D0G7UGC4h4ojesT3GkhCX4H2jeixwDjPHzqRZXDNeWvsSawp8j01zbk6u3FJp49u6dKlnD2xYyPDh9mZ1fpDFrpuw+xTqFnB1RQOHKVJYaSxUS2QIkxu1Rs2rCMe/AdC8ObnYSXLtfv+Dcr889CVllWX0yuxFSmJK9YKUfogbRyws6gZnC/ZUZc7FGp7FPkPZs1AEZs+2z889F3TM5ZQ/3ciZZnDDiRzunvQPrvn+U4y+dhISV2feVzUGZg+kfUp79hftZ93B8Lc5ixUBl9kik5y0HE6XnfYpLnTF3hUcP3uc3pm96Z5RR+iDj2IEQpjIEMMZqO7Qi7qslj5es56te/KTIT/BYLhv7n0+33C4s+fz8e/7HCo3ahNOYIC6BdxAETnp5VEEBJmjrjR2qiUyhCkTtUYT+whnoDp0a9eXtLNw4OwhDhT7l6TtxAm5f3T9dJ86tE9tz8DsgZxpBks6E7tWuGXLbLJBv342+SCUXHgh3HKLddXff789T1GR31m5K+b+mdfTdpJUDk/d9reAhyMibmEdbjfq0TNH2XViF80TmtebYBEJ3BYwH6zSPrlPwS8BF3ILXIwJuJLyEhbtsD2vaxQ99iQjw2azHz8OZ+rOMn30ikfJbJ7J4l2LeWPDGxSVFFV7lFVUD4HZeXwnmw5vIq0sjqF78W/O8iz9E4x1Wi1w3jHGxBtj0rw8Uo0x6kJV6sRrIkOIBZxTpqBGCZFIWuCwmai5rt8Uf+/4ne0HtRtkJ9lly2x8yGjvNcbqwom5imk3arhb3jzxhC36+u9/23Y+aWkweLDP5UVMZSUT59wLwH0Mo+v5w4MajqcbNZw4RWsHZA/wWlw40jgWsE/31p0VbIzhP5v/A9TjPgUrohISbNxUSd3JCdXicIPphhGjAm7J7iWcKjvFgOwBdEzrWPuGcXHVe6LWQXpyOr/4yi8AuOntm0h7Mq3ao+3Mtmw6vMm9/b83/RuAsfmVNEtuYbPmfeX8822YyIEDVXFs/nLmDGzcaP/G888P7BgNHN/9AYriB15dqCEUcOsK1/GvTf8iXuIZ0cU1cUTDhQpw3nluAedP5h1UWQhy2+XaOmYVFbaTRgAlC77S9SsArOxIlWsh1vBsOh0OOnSw9cvS022LrYQEW2/qt7/1afc3/nQvn2YUk306jin3/jPo4YztPpY4iWPp7qWcLAlfm7NqNwIxgCPG/rTmT+w9WXvSwd/X/531B9fTLqUdV3S5ou6DeoqReqxwnnG4O47tqHPbOnF6dsZYDJxP7lMHP2IHfzj4h4zrMY6UxJRqj6T4JE6UnOC+ubYJ08mSk0xfajt03LQOW/g5Odn3P8CznEigbtQNG+x82bdv5JLWYoyoCDgRaS0i80Uk3/XsNa1LRK4Ukc0islVEHvKyfrKIGBGpu8OyEnG6Z3QnNTGV/UX7OdjJ9e8NUQycMYZJ8yZRaSq5a/Bd1gJXUQHbt9sNevQIyXl8JjeXQY4Fzo/Mu0pTWf2HN0hx4xbN7aBybQwKuH37IC/PTrb+3K37y913WytNUZG1xIHtzVhPA+0zJ4/yYP7vAZjW4WZSM4OvpZbRPINhnYZRXlnOwu3ha3NW7UYgBhjaaSjXnXcdZ8rP8PDCh71uc7rsNA8tsNP6E6OeoHmz5vUfONKJDE7brrZtAz9GGHDXXqvPagl+uZ4T4hKYe/NciqYUVXvsmbiHtKQ05mydwwdbP2D6kukcPHWQS09ncu1GArOoB1sPronHv0H0LHAPAQuNMb2Aha731RCReOB3wASgH3CTiPTzWJ8DjAV2R2TEil/ESRwD2w0EYG0Ll+Vhxw5b5iNI3s9/n/nb55OenM7jTk2uPXts7FOHDtaFFklyc6tcqAW+C7idx3dSVFpEu5R2ZLfMCjj+zSGrZRYdUtpTnATbCzfVG/MScebaWl+MGgVJtdT6CjVf/aptcH3iRFXtuVqY9dy32Z1SwcDjydx6t+/ZpvXhdqOGsa2W+0agfWxY4ABmjJ1BYnwir+a9ymf7Pquxftans9hzcg+57XK5ZeAtvh000okMjuiPIQG3+8Ruvjz0JSmJKVyac2n9O/hxzWqjbcu2PDLClhi55/17mL3c9mCe/Z8SBAK76Rwzxpb++OQT+/30lyYe/wZQS+5x2Lka+Irr9cvAR8CD52wzBNjqqj2HiPzDtZ9T2Gc28ADwnzCPVQmQ3Oxclu5eytpjGxnXqZOt37R7d1BFdssqypg0z7ZgeuyKx8hs4XJtRMt9CpCRQb/mnWlWsZv8o/kUlxbbjNJ6cNytg9oNgnXrrIWqXTvbgiZActsPYn9+AZ9nVdJz/Xq4+OKAjxVyQhT/VlRSxBsb3vC5SOvFj9zKkIULbfeHrl1t0U8X80o3kl9xkEpTyfSzCyARZo2YRnyzEBQXdjGh5wQe/fBR5mydgzEm5EV2z5afZePhjcRJHOdnxU4sUPeM7vzskp8xc9lMJs6dyJJbl7j/9v1F+3ly6ZMAzB4/2/e4vUgnMjgWuDax4+RxbgTGdB9DYrwPn9N6ujH4yk8v+SkvrH6Bbcdsv+nvthvHkM3zbNJYIF6P9HQbLrJkic1Mv/Za//Z3BFwTtsBFS8BlG2MKAIwxBSKS5WWbjoBnye292E4QiMhVwD5jzBfRrDiu1I1jDVixb4X9ku/da9snBSHg/rXpX2w+splerXtx98UeTZKjlMDgkDjwQvof3M3a9oa8wjyf7oyrlX3wdJ8G8Zke1G4Q7+e/z9p28K3PP48dAWdMyMqH3P3+3byW95rP2ycnJLPpRzfR5fev2+xUFx93gfG3emyYCFcfb8eoaycFNb5zGdR+EFkts9hzcg8bD2+kX9t+9e/kB2sPrKW8spzz2pxHi2axFQs0dcRUXlr7Ep/s+YS3vnyLG/rfYJcvmsqpslNc0/cad+ymTwQg4IKywMWgC3XhDvs98in+DfxyO9dFUkISM8fO5Lo3r6N5QnOmf+5K7Ajm+3zllVbAzZnjn4Dbtw9Wr7avVcCFHhFZAHjprstUXw/hZZkRkRauY4zzcRx3AncCdA512QKlTpyJeeH2hZRf8D8kfPihzTgKwgLzfv77ANx50Z3V7z6jVELETW4uuZ//m7XtrWXNFwFXLW5pzh/swiCtU26rQ3tiKxN11y4bl5aVBd26BXyYlftW8lreayTGJ3L7oNsRr9NEFasKVrFy30oeGn6Wv6c+aOPigEoMEzP+CRxiXEkOPStakRyfxOR7/hTw2GojTuIY32M8r+a9ypz8OSEXcE4bqpFdR4b0uKGgVXIrfjXyV/zo/37Egwse5Ko+V/HloS95ee3LNItrxoyxM/w7oCNGfLAm9Wjdg5TEFBuHe+qg7dbiLzEo4JxCu8Nyhvm2QwhcqA7X9L2GF772At2OGXJ+/mObJBRML+MJE2DqVCvgjPH95vXhh20m8nXXxZR1NNKETcAZY8bUtk5ECkWkvcv61h446GWzvUCOx/tOwH6gB9ANcKxvnYA1IjLEGFPjE2qMeRF4EWDw4MHhL4euuOme0Z3emb3ZcmQLy/umMByCyo6sNJU1+4Y6RNkCx6BBDJoDL+H7Hb87bim1FyxdarPsxtT6tfEJt9WhHbAshgScZ8BxgBZGYwwT59o2zBOHTuTJMU/Wu8+u47vo+7u+/GPL2/z0x8vcP3qvrH2JNf/5A53SOvGvezaF3XI1oecEK+C2zmHSpaG18PkV0B4Fbr/wdn772W9Zf3A9zy5/1rqSMfz0kp9WlQDyFT/ESJzEMTB7IJ/s+YS1B9a62835TEVFzHViOFlykq1Ht5IYn0jfNn192ymEAk5E+OFFd9q40spKuOce6NMn8AMOHGgzi/fts1mlvpQDWbUKXnnF9lCe4ecNQCMjWkkM7wJO1OoteI9j+wzoJSLdRCQRuBF41xizzhiTZYzpaozpihV6F3oTb0r0cbd4SncFAwdhFVp7YC2FpwrplNaJ/m3PaTMVzRg4qJbI4EvMzaFTh9hXtI+UxBR6rHYldwwdGnS9qe4Z3UltlkJBKhTmr7U/QrFACOJV3tzwJsv2LCOrZRYPj/Ce2XguXdK7MGmYFUwT506k0lRSXFrszoycPnp6RNyOY3uMRRCW7F5CcWlxyI575PQRVu5bSWJ8Ykxa4MBmNjoN0x/98FE+3vUxbVq04ZHL/eu7CfgtRoJKZDh2zIqU9HRbDDcGyCvMA6B/2/6+xb9BSAUcAO+9Z8MhMjLgsceCO1ZcnH/lRIyBifYmjp/9LKhwnMZAtATck8BYEcnHZpI+CSAiHUTkfQBjTDlwDzAX2Ai8aYzZEKXxKgHiWAXmnFxtJ8H8fCgO7AfMs/ZRtdjHigrYZgNroybgcnIYeDYdsAWGSyvqLhzriLyB2QOJ+8CVnRmC4rZxEsfA9i4rXKszIW9fFjBBZoydKTvDAwseAGDayGmkJaX5vO9Dwx+iXUo7VuxbwaD/HcRFL15EQXEBQzoO4TsXfCeg8fhLmxZtGNJxCKUVpXy448OQHXfetnkYDJd3uZyWiRHOvvaDsT3G8vXeX6e80mah//IrvyQ9Od3/A/kpRoJKZIhB92lA9f48r1mwPXnLy2HyZPv68cdDU+C4vrZaZ87ADTfAJZfYbitLl9r/yVRfo7EaL1ERcMaYI8aY0caYXq7no67l+40xX/XY7n1jTG9jTA9jjNcO1S5LXN0FnpSocUWXK0hOSGbNgc85MKiXnUDy8gI6Vq2uon37bKX9du1s8dZoIEKr/hdyQSGUVJTwpzV1x1L9ZuVvAGys3NKldmGQ7lMHt9WhHVXHjjZB1myavXw2u0/sZkD2AG4bdJtf+6YkpjBjjHW15BXmseXIFuIlnmfHP0ucRG4KdHfKCGE5EZ/bUMUAT499mpbNWjKo3SDuuOiOwA7imVHpgxgJKpEhBgVcjdZ7vtC8uS0MXlpqrYrB8MUXsGUL5OTAXXcFdyyHsWOtJW7pUneMajXeew/eegtWrqy6Efz1rwMqdt7Y0E4MSlhp3qy5O5lh3sWuu7UA3KjHzx7n0z2fkhCXwOhu57SZirb71CE3l8c/si9//uHPOX72uNfNFmxfwHtb3iM1MZVJg+6GTZtsPaQgyodUG4ZnIkOwzaJDwZEjtk5fixYBJZkcKD7grvo+a9ysgFpF/c/A/2HD3RtYfvtylt++nPyf5PseBB4i3H1RXeVEgqXSVDJ3m7Xexmr8myd92vRh20+38fGtH5MQF2D4dYuO+SJiAAAgAElEQVQWtj1aaaltPVcP/bP6kxCXwObDmzlVesq/c8WggHMKhftd7y9UblRn7h4+PHRu5datrXWtrAwWLaq53pnD7r0Xli+H9evhBz8IzbkbOCrglLDjWB7mdHBNoAEkMizYvoAKU8GlOZfSKvmcO69oZ6A6DBrENRvhiuI2HDlzhGkfT6uxSUVlhbsdzcMjHiZ7xyEbZ9O3b7X6ZMFQLZFhwQI7MUYTZ9IfMMAKVT95ZNEjFJcWc1Wfqxjd3f8esQ792vbjkk6XcEmnS+iWEXgmbKAM7jCYzOaZ7Di+gy1HtgR9vM8LPufgqYN0btWZ89qcF4IRhp/slGyfaiTWiR9iJDkhmfPanIfBsO7gOv/OE2NFfMsqytz9nwdkD/Bv51ALuFCX7qitK4MxVctuvdUKvf7nxD83YVTAKWHHsTzMM9uoEAKywM3Jd7lPe3qxNEQ7A9UhNxcBZn+UhCA8v+J5luxawubDm92P2ctns+7gOrqmd+XeofeGpR1M/7bW6rAlE4rPnoRP624oHnaCiH/7vOBz/vL5X0iIS2Dm2JkhHlhkiY+Ld2dChsKN6uk+bVL1MP1NZGgfYCJDjBXx3Xh4I6UVpfRs3dOvGFAgdAIuXO2rPOPgPK3TeXnWXd6hA1xwQWjP2QhQAaeEnV6te9E9oztHy0/yUVds1wE/W2o5rqIa5UMgdlyofftCUhKDVu3j+71voKyyjMtfupy+v+vrftw/3xaSfWrMUyQnJIelHUxSQhL92vbDCKzLIvpu1CDu2h/76DEMhnsuvofemb1DPLDI49yA/HPjP4N2ozoisCG4T0OKv4kM2a6QggI/Lf8x5kINKP7Nwamft39/4AOorLQxcBB6AXfRRVYo79plQ0ocnLkryALnjRUVcErYERF3r8OHvpZEZWlJ9S9pPRwoPsC+on2kJaV5dx3EigUuIQG+8Q0Api+AEZ1H0Duzd43H7YNu51v9vmX3CZNLoloiQ6DNokNFgHftxhiW7F4CwMRhE0M9qqjwjT7fID05naW7l/J/+f8X8HGOnTnGp3ttTOiobqNCOMIGgJ+todwhBYUBWuBiRMAFlIHq4MyNX35Z93Z1sW2brSDQoYMtyB1K4uJg/Hj72nO+8uxQo9RABZwSESYNm0SH1A6salPCawPwy43qeedZI2uwsjL6JUQ8mT4dmjUj+69v8vGA2Wy+Z3ONx5+u+pN1eVVUhO2O1p3I0CneXusg+yAGzJkzVqzHxfntAtl1YhfHzx6nTYs25KTl1L9DAyA9OZ3HrrC1sybNm1RvuZnamL99PpWmkuGdh/vvTmvo+Nkayvku5BXmucuY+ESMxcA5AjQgC5wzvwRRSD3szePPLSdy8qRtdB8fbzNVlRqogFMiQsvElkwfbTMJp4yGU5+v9HnfOu889++Hs2ftHWFaDPyQ9expC0w6BSfrcpNt3QqnT9uU/BBXendb4Hq6AsajZYVbv94K1QCSNDz/740pxuvui++mV+tebDmyhT989oeAjuF2n3qLCW3s+OlCzWieQZdWXThbfta/5JEYioEzxlTvnewvAwZYF+SGDTaDNxDCFf/mMG6cHePixXDqlE3AKi+HYcNsMWWlBirglIhx84CbGdyiJ/vTYOax//q8X7WeoecSK/FvnjzyiJ30lyyBd96pfbswTogD29mSJOtSTlMeR/QEXJAJDBDgD1YMkxifyNPjngbgF4t/wZHTR/za3xhTe0u5pkAAAfkBJTLEkAvVsUZntcyifUp7/w+QmmrnyLIy2LgxsEGE2wLXti0MHmwF5t13w9P2O6Lu09pRAadEjDiJY9ZXrBVuRoed7D2516f96rzzjJUSIp60agW//KV9/cADtumyN8I4IaYnp9M1vStnKWNzJjBvnt+JI0FjjC3ACbaCup+4a14FEvMT43yj9zcY1W0Ux84e46W1L/m1b15hHgXFBXRI7cAFWU0wMy8AAed3IoMxMSXgnBZaA7MHBm6NduaZQN2o4Soh4snXv26fX3mlKnv+a18L3/kaOCrglIgyYvB1XL8pnjPNYMqc+pt6F5UUkX80n2ZxzejXtl/NDWIlgeFc7rjD1ivavh2ef977NmGeEB3h83luti16utJ3t3VImDMH5s+37o/vfc/v3YNyGcU4IsIPBtlipE45EF9psuVDHJwYOD/iOv1OZCgutpagFi3sI8rkH7Gehj6ZQTSOd+aZQPpRFxba652aCt3CWEPxvvvg2Wdh2jT7eOON8ArGBo4KOCWyiDBjZy8Sy+G1TW+ycl/dosLdvDmrlubNsehCBZuR+swz9vWvfgUHD1Zfb0zVnXCYXBLuH60LO9gFkSwnUlZmJ2OAn//c7ziiI6ePsPvEbponNG8U5UO8Ma7HuIAa3NfaUq6p0KaNTYo5fNjnItWeLlSfyrfEkPUNYOtRe6PaKzMIT0MwiQyeN5txYZQNKSk2hnjqVPu44YbwnasRoAJOiTjdOvRj4nL7+r6599U5odabOh+LLlSH8eNtZlVRkRUxnhw4YEVdejp06RKW07sTGbJd1zeSAu6FF2DzZvt/+fGP/d79i0KbnTsge0BArbMaApktMv1ucH/i7AmW7VlGvMQzpntoeuc2OOLjq8pYnHtjVAs5aTlkJGdw+PRh9hXtq3+HGEpgANh6zM5zPVsHcaPq3CiuXet/U/twJzAoAaECTok8PXvy8BLIoiWf7PmEdzbWHuhfZwKDMVUCrkePcIw0eJ55xv7gvPiizTZ1Hs5EmJsbtgKV7lIipbsxSYmwerXPP3gB8dhj0Lmz/fsmT7bLZs6ERC+W03porAkM5+JuM+ejG3XhjoWUV5YzLGcY6clNODPPzzg4EXFb4XyKg4tRC1xQAq5dO8jOtuU5duzwb99wJzAoAaECTok8vXqRVgIPH7Yxba+te63WTeu0wBUU2DpjbdrEbpr5eedZV6IxsHdv1cMRUk7QbhjolNaJzOaZHD17lL3jhtqFc+eG52SrV9vEjT177N9XWmqtj1ddFdDhGnMCgyeOG9TXBvd1tpRrSgSQyDCkwxAAFu9aXP/GMVQDrqS8hN0ndhMv8XRN7xrcwTytcL5SWWlLe4DNElViBhVwSuRxxatd82UlAAu3L/Ra0LSsoszdgNprBwYn/i0W3aeePPWUrVe3e3f1x4EDMKn+RI5AEZEqK9zlrjiycJQTcWreAfzkJ/Zv27MH3nsvYOtiY05g8OSi9heR2TyTncd31lujzBjDB9uacP03TwJIZBjf01b696kPbQxZ4HYc30GlqaRLehfvccD+EEgcXF6enas6doTzzw/u/EpIUQGnRB6XgOu8bjf92vajqLSIZXuW1dhs0+FNlFaU0j2jO62SW9U8TqxmoJ6LiP3B8XSh5uRYd0aYcScydHMV0Z071xbWDSXvvGNr3rVpY61wOTnQqVPAwc5nys6w8dBG4iSOC7Ibd5mM+Lh4t7Coz4264dAG9p7cS3bLbHedvyZLABa4S3MuJTUxlQ2HNrDnxJ66N46hGLiQuE8dAslE1X6kMYsKOCXydOhgq/IfOsSEHNvH0XENeeLEv9WbwBDrAi6KuEuJlO2x6f9HjsCqVaE7QUkJ3H+/ff3LX4bElb3h0AYqTAV9MvvQoln0SziEmyt72EKl9VmGnO/IlT2vrNlSrqkRgIBLjE9kdPfRgA8xhzFkgXNKiPRqHQJPQyC14LQfaczSxGcBJSrExbmTDiYk9gdwu4Y8Wb1/NVCHG22Ly+WkAq5WnGu3umA15kovzaKD5eWXbUB0//629l0IWLRjEdD43acOjgXuo50fcbrsdK3bOd+RJtl94VwCEHBQ5Xqu140aQzFwIbXA9ewJLVvCvn2+XbsTJ6r6kY5polnPMYwKOCU6uETX8GOptGzWkrzCPPadrJ7eP3/7fABGdB7h/Rh5tkacxmXUTt82fclsnsmek3vYMtIVRxjKciL/dbVEmzjR1r4LkqNnjvLk0icBuOn8m4I+XkMgq2UWF7W/iJKKEj7a+ZHXbYpKiliyawlxEse4HuMiO8BYJEAB54jfBdsXeI27dRNDFriQlBBxiIuDyy6zrxcsqH/7BQtsyMWll8ZuolgTRgWcEh1ciQdJ23Yyqpt1o3reFe86vouNhzeSmpjKpTmX1tz/5EnrQk1MtJmeileqxVhlnbTXa+VK60oNlrNnYZG1ljEhNEH1v/joFxw7e4xR3Ubx9d7hy9CNNeqzDH2480PKKsu4pOMltG7eOpJDi00CSGIA6Nyqc51xt25iSMA5LtSQCDio+q76ciOn7tOYRgWcEh0ct+fWrVU/Xh5uVOeHbEz3MTSLb1Zzf0/rWzMv6xU3TozVnD2LYMQImzU6b17wB16yBE6fhoEDbVxjkGw+vJnfr/o9cRLH7PGzm1SbKM9yIt7wjH9TqG6B87MorU9u1BhJYiitKGXXiV3ESRzd0kPUwsoRY/UlNBlTJfJCdIOmhBYVcEp08BBwzo/S/G3zKauwrXHc7YJqK5cQicbKjQTHArd452JOX2mDuEPiRg3x3fnk+ZMpryzn9kG3ey8b04gZ0nEI6cnpbD261R3z5GCMqf/70NRISbE9Sk+ftn1L/cCZb2pNZCgpsd1TEhKi7jbceXwnlaaSzq06k5SQFJqD9ukDXbtaK/zq1bVvt2GDjZXLzrY3aUrMoQJOiQ5O7bb8fLpldOO8NudxouQEr+a9SmlFKQt3LATqsDiEuY9oYyKrZRaDOwy2MVaDMuzCuXNtgc5gCOHd+fxt83lvy3ukJqbyq5G/Cvp4DY2EuAR3bNu5lqHNRzaz68Qu2rRow0UdLorG8GIPkYDj4EZ0HkFyQjJ5hXmcLDlZcwNP61uUrcAhTWBwEPHNjepZPiSc/U+VgNH/ihIdOnaEpCQoLISiIqaOmArA1EVTmbt1LsWlxfRv25+cVjne91cLnF+43agl622dtoMHA2tq7bBrF2zcCKmpNsA5CMory7lvnm18//CIh8lOCX99vFjE/T86xzLkuE/H9xiv5UM8CTAOLikhyS2Ith3dVnMDRxA6AjGKhLSEiCeO1dwXAafu05hFZwMlOniUEmHbNm664CYu6XgJB4oPcNu7twF1uIvKymD9evt6QNNytQWKE2P1wba5/gUx14bjPh0zJugYxD+v+TPrD66na3pX7h16b1DHasg41uYPd3zI2fKz7uXqPq2FAC1wUGXRyj+aX3OlczxHIEaRsFjgAEaNqjuhqagIli618/TYsaE9txIyVMAp0cOJg8vPdweuAxw+bWswOaKjBhs32l6bPXtCWlokRtrgGdJxCBnJGTbGapTLahlMPbgQ3Z2fOHuCRz98FIAZY2aQnJAc1PEaMu1T25PbLpcz5Wf4eNfHAJwqPcXiXYsRRMuHnEswAi7Dzj3nxhtWO14MWOBCWkLEk5SUuhOaFi2yN8qXXAKtNes5VlEBp0QPjzg4gGE5w7jx/BsBaNmsJZflXOZ9P8f1p+5Tn0mIS2BsD3sn/UH7UzZA+9NPrSvVX06dgoU2RjHYBIYnljzBodOHGN55ONf3uz6oYzUGHDfqu5vfBWxx39KKUgZ3GEzbltEvaRFThMACF+sCLmwuVKj67nq7kXOWqfs0plEBp0SPvn3t84YN7kVPjXmKnq17ctfgu2rPunLi3zSBwS9GdbX19lYc+cJOzJWV8Itf+H+gGTNs5t/QoTaeLkC2H9vOcyueA2DWuFlNqmxIbdzQ/wYA/rTmT+w8vtOd0KDlQ7wQhIDrlWkFkVcXqhNTF2UBt79oP9uObaNFsxb0aN0j9CcYZecDVqyovtyzfIjWf4tpVMAp0cOxoHkE03du1Zn8n+Qzc9zM2vdTC1xADGrv6ota8DlMn27b4/zv/1YT0PWydy/MdP1vZtbxP/KBB+Y/QGlFKd8b+D0u7nhxUMdqLAxqP4jvXPAdSipKeGjBQxr/VhcBJjFAw7DAOeJ9VLdRJMYnhv4E/ftbS/yWLdVLsWzaZJOU2raFizTrOZZRAadEj/PPtyJi82Zbz8kXjNEM1AC5IOsC4iSOTYc3caZ3d7jzTlvIc9Ik3w8yZQqcOQPf+hYMHx7wWD7e9TFvb3ybFs1a8OtRvw74OI2R6aOnk5yQzBsb3mDbsW1kJGcwpOOQaA8r9gjCAtcprRNJ8UkcKD5Acek5deRiJIkh7OI9KcmKOGNg3bqq5Y77dPx4LR8S4+h/R4keycm2DVZlZVVWaX3s2mUbLGdlRX2CbWg0b9acvm36UmEq2HBog3Wftmpla8L9+tfw+uvVH198Uf0AK1fCa6/Z7LWnngp4HJWmkolzJwLwwKUP0DGtYzB/VqOjc6vOTB422f1+XI9xxMfFR3FEMUoQAi5O4txuyRpWuBiwwJVXljN/m+0FHVb3uXMT7NwUg7pPGxAq4JTo4sSx+VqT7LPP7HNubtSLbDZEctvZCfvzgs+ti+RRmwHK1Klw883VH5dcAttcdbKMsQ3rAe67D7oF3tbnlS9eYU3BGjqmdmTypZPr36EJ8uDwB2mfYm9Q1H1aC1lZ9vngwbpbQtWCVzeqMTEh4JbvXc6JkhP0zuxN94zu4TvRuWEsp07B4sV2bh2nWc+xTkK0B6A0cXJz4dVXq98B1sXcufb5iivCN6ZGzKB2g/jbur+x9oDrev/kJ/YHcO/e6htu2gRr1sADD8Dbb8Obb8KyZfZHc8qUgM9fXFrMwwsfBuDJMU/SMrFlwMdqzKQkpvDOt9/h3c3vctMFN0V7OLFJs2a2W8Lhw/aR7V8BaK+lRIqLbThHixa21EaUcIo3h128OzfQzvz70Ue2RNPFF9sbPCWmUQGnRBdvJvzaMEbT24PEbYE74Lrjrs0dun+/LfPyzjtWND/4oF0+bVpQtfdmfDKDguICLu5wMd+54DsBH6cpMLTTUIZ2GhrtYcQ27dtb8VZQ4L+Ac4r5HvHIRPXMQI2ihf+DbXaeC7uAc3qcrlsH5eXafaGBoS5UJbo4Ai4vr343yPr1trlyu3aawBAgjoDLK8yjorKO692hAzz0kH39zW/a2MMBA+C22wI+954Te3h62dMAzB4/W9tCKcETglIiTrHcaseJYnztgeIDrClYQ3JCMpd3uTy8J0tPt+EQZ8/aZDK9QW5Q6AyqRJfWraFzZ+u2yPdSk8kT5+5w/HiNfwuQNi3a0CmtE6fKTnkvoeDJpEnQqZOd3AFmz7ZZwx7sOr6L69+8nvGvjWf8a+P54X9/SEl5idfDTVk4hTPlZ7ih/w1c1rmWIs2K4g+hLuYbA/Fvc7faMJGRXUfSvFnz8J/QuRl+6y0b89q6tXWhKjGPCjgl+viayKB3hyFhUDt7vd1xcLXRogU8bS1mXHttVeFPD55f8Txvb3ybedvmMW/bPF5c86K7OK8nK/au4PV1r5MUn8RTYwLPYFWUagQh4HLSckiMT2R/0X5OlZ6qfpwoCbiKygpmLZ8FwDd6fyMyJ3Xm3+eft8/jxtW4UVNik6gIOBFpLSLzRSTf9ZxRy3ZXishmEdkqIg95LH9cRPaJyFrX46uRG70ScnyJg9PmyiHDcaPWK+AAvv1t695+/XWvq51YneeufI7fTPgNANM+nsbBU1Utuowx7rIh9w27j67pXYMYvaJ44Lg6AxBw8XHx7gzPbce2VT9OlATcXz7/C3mFeXRp1YVbB90amZM68++xY/ZZb5AbDNGywD0ELDTG9AIWut5XQ0Tigd8BE4B+wE0i0s9jk9nGmFzX4/1IDFoJE+dmQnlDmyuHjBqJDPVxwQW2Zt857D6xmy8PfUlqYip3Db6Le4bcw4SeEygqLeLnH/7cvd0bG97g072fkt0ymynDA89gVZQaOEIrgG4M4MWN6hwnCjFwJ0tO8siHjwC2pWByQs3vXFg4N554/PjInFcJmmgJuKuBl12vXwa+6WWbIcBWY8x2Y0wp8A/Xfkpjw7MWkTHet9HsqJDhswu1HpxSB2O6j6FZfDMAnhn3DPESzx/X/JFle5ax9+ReHlxgM1injZpGalJqUOdUlGoE4UIFL6VEomiBm75kOgdPHeTSnEvdPXEjQqdOkJlpX194od/ZvEr0iJaAyzbGFAC4nrO8bNMR2OPxfq9rmcM9IpInIn+pzQWrNBA6d4aMDDh0yGaZnotnc2UVcEHTNb0rrZJaUXiqkO3Htgd8HMd96lkp/ry253HX4LuoNJVc9pfLyJmdw+4TuxmYPZBbcyPkElKaDiGywH1R6Oo6EiUBt+PYDnfs27Pjn0UimaQlUnUTrfNrgyJsAk5EFojIei8PX61o3j7BjnnmD0APIBcoAJ6pYxx3isgqEVl16NAhv/4GJUKIwLBh9vWiRTXXb9oEu3fbwpIXXhjZsTVCRISv9f4aQDVXpz+UVpSyYPsCoGarn8e/8jgXd7iYrJZZZLXMolt6N/7wtT9oOygl9HTpYuNit2+vypb2g9HdRxMncbyx/g02HtoYNQH34IIHKa0o5eYBN3NxxyhkgP74xzBkCPzgB5E/txIwYRNwxpgxxpjzvTz+AxSKSHsA1/NBL4fYC+R4vO8E7Hcdu9AYU2GMqQT+iHW31jaOF40xg40xg9tqZenYxem751jaPPEsH6LNlUPCtJHTSIxP5PV1r7Ni7wq/91+2ZxnFpcX0a9uPzq06V1uX2SKTlXespHByIYWTC9n+s+0MyxkWqqErShUtWkCfPraGpK/9lD3o26YvPxj0AypMBZPnTbJdSUSq2nRFgKW7l/LWl2/RPKE500dPj9h5q3HNNbBiBXTtGp3zKwERrV/Dd4FbXK9vAf7jZZvPgF4i0k1EEoEbXfs5os/hGsD/b64SWzgCbt68mgV9tblyyOmW0Y2JQ21m6MS5EzG1xR7WQsRa/ShKffiSBFUHvxz5S1ITU3l/6xzmdau07bmaNQvhAGun0lS6M7Tvv/R+OqV1ish5lcZBtATck8BYEckHxrreIyIdROR9AGNMOXAPMBfYCLxpjNng2n+GiKwTkTxgJDAx0n+AEmJ69YIePeDo0aqG9WCbK3/8sb0r1uyokPLwiIfJapnFp3s/5Y0Nb9S7/c7jO3lm2TPM/GQmb335FqACTokB/GnH54XslGweudxmf943HsrbBxbEv2zPMmZ+MpOZn8xk1qez2H1id737vJb3Gqv2r6JDagceuOyBgM6rNF2i0gvVGHMEGO1l+X7gqx7v3wdqlAgxxvxPWAeoRIcJE+C3v7UWt6GuHpAffmibKw8ZYu+MlZCRlpTGtJHTuPO9O3lwwYNc3efqWiu/V1RWcPU/riavMM+9LCUxheGdh0dquIriHc8s9gD52SU/44Uls9mQdYA/DiznLj/333NiD2NeGcOZ8jPuZa+ve53P7vis1pZxp0pPMWWhLaszffR0Wia2DHT4ShNFA4qU2MFxkTodFzxfq/s0LNw26DYGZA9g94ndzF4+u9bt/vz5n8krzKNTWicmD5vM5GGTeeeGd0hKSIrgaBXFC46A++ILqKwM6BBJCUnMSL0WgJ933cHxs8f92t9pEze001AmD5tM+5T2rClYw6tfvFrrPjOXzWR/0X4GdxjMzQNuDmjcStNGBZwSO4wcCUlJ1oV66JCWD4kA8XHxzBpnyxdMXzqdgqKa5RhOlpzk0Q8fBeDpsU8zc9xMZo6bydge2hFDiQHatoWOHW24xdZ6+vvWwXUnOzFiFxyOL+GJj5/weT/PNnF/v+7vzBw3090ubsrCKRSXFtfYZ+/Jvcz4ZAYAs8bNqtVKpyh1oZ8aJXZo0QKuuMIKt3nzbHP77du1uXKYGd19NFf1uYri0mIeWfRIjfW/XvLr6BQYVRRfCTKRAUAKC5n9AQjCcyueq97kvhY828RNHDrR3SbuuwO+y+AOgykoLnALNU8ci931/a5nRJcRAY9ZadpEJQZOUWrlyiuteLv3XmjuisfS5sphZ+bYmbyf/z5/XftXVu5fWW3d5sObgSgUGFUUX8nNhffeswLuhgBvMgoKuKgAvpc2gpdPfszwvwynbcu6S0+VVZSx+chmslpmMWVEVZu4OInj2fHPMvyvw3l62dPcceEd5LSyVbFW7lvJa3mvkRif6LbUKUogqIBTYotrroEpU+Dw4aplN94YvfE0EXpn9ua+ofcxY9kM1h+sWZXnttzbolNgVFF8wbHABZHIwDbb0P7XAybyn1V5FJ4qpPBUoU+7zhgzg7SktGrLLut8GTf0v4E3N7zJlIVTeO3a12pY7LpndA98vEqTR/yt/9SQGTx4sFm1alW0h6HUx8GDUOiaONPSbLV1JexUmkryj+RTWlFabXl8XDx9MvtoJwUldtm+3ZYhatcusLZaZWWQkmIz3k+c4Gizcvad9NLWzwstE1vWKsR2Ht9J39/2paSihOW3L2fn8Z3c+PaNZLXMIv8n+TVEn6Kci4isNsYM9rZOLXBK7JGVFdFK6IolTuLo06ZPtIehKP7Ttau92TtwwD78bYW1aZMVbz16QFoarYHWzVsHP6z0rtw37D6mL53Ozz74GQeKbauuX438lYo3JWg0iUFRFEVp2MTFBVfQ19nHOUYImTJ8Ctkts1mxbwW7TuzigqwLuH3Q7SE/j9L0UAGnKIqiNHwc8bV4sf/7OrFzTixdCElNSmXaqGnu97PGz9JwBCUkqIBTFEVRGj7XX2+fn38e9u/3b98wWuAAbs29lTsuvIOpI6YypvuYsJxDaXpoEoOiKIrSOLj2WvjXv+D734e//tW3fYyxtSaPH4e9e21RYEWJEepKYlALnKIoitI4mDEDmjWDl1+G1at922f3bive2raFDh3COz5FCSEq4BRFUZTGQc+e8NOfWqvaxIn2uT6c+LfcXNBC1UoDQgWcoiiK0nh45BHIzIQlS2DNmvq3d+LfwpDAoCjhRAWcoiiK0nhIT7excAAffFD/9mFOYFCUcKECTlEURWlcXHmlfZ4zp/5tw1hCRFHCiQo4RVEUpXExZgwkJMCnn8KxY7Vvd/SoTeYF0eoAAA0nSURBVGJo3hx69Yrc+BQlBKiAUxRFURoXaWlw2WVQWQkLFtS+nZOpOmAAxGtxXaVhoQJOURRFaXzU50Y1Bp580r4eOTIyY1KUEKICTlEURWl8TJhgnz/4wHs5kf/+FxYtgowMuP/+yI5NUUKACjhFURSl8TFgALRvDwUFkJdXfV1pKUyaZF8//rjtxKAoDQwVcIqiKErjQ6R2N+rvfgdbt0KfPnDXXZEfm6KEABVwiqIoSuPE043qUF4OTzxhXz/9tG29pSgNEBVwiqIoSuNk7FibXfrJJ3DypF326adw5Ii1vn3ta9Edn6IEgQo4RVEUpXGSng5Dh1qr28KFdpnjTp0wQXufKg0aFXCKoihK48VxozrCzVPAKUoDRgWcoiiK0njxFHAFBbb3afPmcPnl0R2XogSJCjhFURSl8ZKbC1lZsHcvzJpll40cCcnJ0R2XogSJCjhFURSl8RIXV1VO5Lnn7LO6T5VGgAo4RVEUpXHjCLaysurvFaUBowJOURRFadyMHWstcQA9e0KPHtEdj6KEABVwiqIoSuMmMxOGDLGv1fqmNBJUwCmKoiiNn4kToXt3uPPOaI9EUUKCCjhFURSl8XPDDbBtG5x/frRHoighQQWcoiiKoihKA0MFnKIoiqIoSgNDBZyiKIqiKEoDQwWcoiiKoihKAyMqAk5EWovIfBHJdz1n1LLdlSKyWUS2ishD56z7iWvdBhGZEZmRK4qiKIqiRJ9oWeAeAhYaY3oBC13vqyEi8cDvgAlAP+AmEennWjcSuBoYYIzpDzwdqYEriqIoiqJEm2gJuKuBl12vXwa+6WWbIcBWY8x2Y0wp8A/XfgB3AU8aY0oAjDEHwzxeRVEURVGUmCFaAi7bGFMA4HrO8rJNR2CPx/u9rmUAvYERIrJCRBaLyMW1nUhE7hSRVSKy6tChQyEavqIoiqIoSvRICNeBRWQB0M7Lqqm+HsLLMuN6TgAygKHAxcCbItLdGGNq7GDMi8CLAIMHD66xXlEURVEUpaERNgFnjBlT2zoRKRSR9saYAhFpD3hzge4FcjzedwL2e6x7xyXYVopIJdAGUBOboiiKoiiNnmi5UN8FbnG9vgX4j5dtPgN6iUg3EUkEbnTtB/BvYBSAiPQGEoHDYR2xoiiKoihKjBAtAfckMFZE8oGxrveISAcReR/AGFMO3APMBTYCbxpjNrj2/wvQXUTWY5MbbvHmPlUURVEURWmMSFPSPYMHDzarVq2K9jAURVEURVHqRURWG2MGe13XlASciBwCdoXp8G1QN65eA4teB4teB4teB70GDnodLHodfL8GXYwxbb2taFICLpyIyKraVHJTQa+BRa+DRa+DRa+DXgMHvQ4WvQ6huQbaC1VRFEVRFKWBoQJOURRFURSlgaECLnS8GO0BxAB6DSx6HSx6HSx6HfQaOOh1sOh1CME10Bg4RVEURVGUBoZa4BRFURRFURoYKuCCRESuFJHNIrJVRB6K9ngihYjkiMiHIrJRRDaIyM9cyx8XkX0istb1+Gq0xxpuRGSniKxz/b2rXMtai8h8Ecl3PWdEe5zhQkT6ePy/14rISRG5tyl8FkTkLyJy0FVU3FlW6/9eRKa45orNIjI+OqMOPbVch5kisklE8kTkXyKS7lreVUTOeHwuXojeyENLLdeh1u9BY/w81HIN3vD4+3eKyFrX8sb8WajtNzJk84O6UINAROKBLdhuEnux7b9uMsZ8GdWBRQBXD9v2xpg1IpIKrAa+CdwAFBtjno7qACOIiOwEBhtjDnssmwEcNcY86RL2GcaYB6M1xkjh+k7sAy4BbqWRfxZE5HKgGHjFGHO+a5nX/72I9AP+DgwBOgALgN7GmIooDT9k1HIdxgGLjDHlIvIUgOs6dAXec7ZrTNRyHR7Hy/egsX4evF2Dc9Y/A5wwxvyykX8WavuN/D4hmh/UAhccQ4CtxpjtxphSbFuvq6M8pohgjCkwxqxxvS7CtjvrGN1RxRRXAy+7Xr+M/eI2BUYD24wx4SqYHVMYYz4Gjp6zuLb//dXAP4wxJcaYHcBW7BzS4PF2HYwx81wtEQGWA50iPrAIU8vnoTYa5eehrmsgIoK9yf97RAcVBer4jQzZ/KACLjg6Ans83u+lCYoY113UIGCFa9E9LrfJXxqz69ADA8wTkdUicqdrWbYxpgDsFxnIitroIsuNVJ+cm9pnAWr/3zfl+eI2YI7H+24i8rmILBaREdEaVATx9j1oip+HEUChMSbfY1mj/yyc8xsZsvlBBVxwiJdlTconLSIpwNvAvcaYk8AfgB5ALlAAPBPF4UWKy4wxFwITgB+7XAhNDhFJBK4C3nItaoqfhbpokvOFiEwFyoHXXYsKgM7GmEHAfcDfRCQtWuOLALV9D5ri5+Emqt/gNfrPgpffyFo39bKszs+DCrjg2AvkeLzvBOyP0lgijog0w34wXzfGvANgjCk0xlQYYyqBP9IIXAL1YYzZ73o+CPwL+zcXumIgnFiIg9EbYcSYAKwxxhRC0/wsuKjtf9/k5gsRuQX4OvBd4wq4drmIjrherwa2Ab2jN8rwUsf3oEl9HkQkAbgWeMNZ1tg/C95+Iwnh/KACLjg+A3qJSDeX9eFG4N0ojykiuGIZ/gxsNMbM8lje3mOza4D15+7bmBCRlq4AVUSkJTAO+ze/C9zi2uwW4D/RGWFEqXZ33dQ+Cx7U9r9/F7hRRJJEpBvQC1gZhfFFBBG5EngQuMoYc9pjeVtXsgsi0h17HbZHZ5Thp47vQZP6PABjgE3GmL3Ogsb8WajtN5JQzg/GGH0E8QC+is1E3QZMjfZ4Ivh3D8ead/OAta7HV4FXgXWu5e9is3CiPt4wXofuwBeuxwbnMwBkAguBfNdz62iPNczXoQVwBGjlsazRfxawgrUAKMPeQd9e1/8emOqaKzYDE6I9/jBfh63YmB5nfnjBte11ru/KF8Aa4BvRHn+Yr0Ot34PG+Hnwdg1cy18CfnTOto35s1Dbb2TI5gctI6IoiqIoitLAUBeqoiiKoihKA0MFnKIoiqIoSgNDBZyiKIqiKEoDQwWcoiiKoihKA0MFnKIoiqIoSgNDBZyiKDGFiFwjIkZE+nos6yoi612vB4vI89EboXdE5JuuhtThPMcy13NXEfmOx/KYvCaKooQPFXCKosQaNwFLsYWxa2CMWWWM+Wlkh+QT3wTCKuCMMZe6XnYFvuOxPFaviaIoYUIFnKIoMYOrb+Bl2AKoXgWciHxFRN5zvW4rIvNFZI2I/K+I7BKRNi4L1UYR+aOIbBCReSLS3LXPRyIyW0Q+dm1zsYi8IyL5IjLN4zw3i8hKEVnrOrZTMb5YRJ4QkS9EZLmIZIvIpdg+sDNd2/c4Z8wvicgLIrJERLaIyNddy5NF5K8iss7V0Huka3l/j3PniUgv59yuQz4JjHCtn3jONWktIv927bdcRAa4lj8utpn6RyKyXURU8ClKA0YFnKIoscQ3gQ+MMVuAoyJyYT3bPwYsMsZciO1D29ljXS/gd8aY/sBxbNV3h1JjzOXAC9hWNj8Gzge+LyKZInIe8G3gMmNMLlABfNe1b0tguTFmIPAxcIcxZhm2yv79xphcY8w2L2PtClwBfA14QUSSXefFGHMB1vL4smv5j4DnXOcejK1o78lDwBLXuWafs+4XwOfGmAHAw8ArHuv6AuOx/Tgfc/VqVBSlAZIQ7QEoiqJ4cBPwrOv1P1zv19Sx/XBsf0mMMR+IyDGPdTuMMWtdr1djBZSD07N4HbDBGFMAICLbsQ2lhwMXAZ/ZloY0p6rpdCnwnsdxx/r4t71pbEPzfNd5+rrO8xvX+DeJyC5sM+9Pgaki0gl4xxiT7+M5cB3zOtcxF7kEaSvXuv8zxpQAJSJyEMimpjhUFKUBoAJOUZSYQEQygVHA+SJigHjAiMgDde1Wx7oSj9cVWBF27rrKc7arxM6LArxsjJni5bhlpqoHYQW+z6Pn9i001DJ+Y8zfRGQF1lo3V0R+YIxZ5ON5vB3TOfe510R/AxSlgaIuVEVRYoXrgVeMMV2MMV2NMTnADqxFqTaWAjcAiMg4ICNEY1kIXC8iWa5jtxaRLvXsUwSk1rH+WyIS54qP645tWP0xLtesiPTGuoA3i0h3YLsx5nmstXCAH+fyPOZXgMPGmJP1jF1RlAaGCjhFUWKFm7BxbJ68jUe2pRd+AYwTkTXABKAAK26CwhjzJfAIME9E8oD5QPt6dvsHcL8rGaGHl/WbgcXAHOBHxpizwO+BeBFZB7wBfN/l4vw2sF5E1mJdra+cc6w8oNyVSDHxnHWPA4Nd434SuMWnP1pRlAaFVHkCFEVRGhYikgRUGGPKRWQY8AdX4H9MISIvAe8ZY/4Z7bEoitI40PgHRVEaMp2BN0UkDptccEeUx6MoihIR1AKnKIqiKIrSwNAYOEVRFEVRlAaGCjhFURRFUZQGhgo4RVEURVGUBoYKOEVRFEVRlAaGCjhFURRFUZQGhgo4RVEURVGUBsb/A9h6sdA+rf6+AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 720x432 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "import pylab\n",
    "import modeller\n",
    "\n",
    "def r_enumerate(seq):\n",
    "    \"\"\"Enumerate a sequence in reverse order\"\"\"\n",
    "    # Note that we don't use reversed() since Python 2.3 doesn't have it\n",
    "    num = len(seq) - 1\n",
    "    while num >= 0:\n",
    "        yield num, seq[num]\n",
    "        num -= 1\n",
    "\n",
    "def get_profile(profile_file, seq):\n",
    "    \"\"\"Read `profile_file` into a Python array, and add gaps corresponding to\n",
    "       the alignment sequence `seq`.\"\"\"\n",
    "    # Read all non-comment and non-blank lines from the file:\n",
    "    f = open(profile_file)\n",
    "    vals = []\n",
    "    for line in f:\n",
    "        if not line.startswith('#') and len(line) > 10:\n",
    "            spl = line.split()\n",
    "            vals.append(float(spl[-1]))\n",
    "    # Insert gaps into the profile corresponding to those in seq:\n",
    "    for n, res in r_enumerate(seq.residues):\n",
    "        for gap in range(res.get_leading_gaps()):\n",
    "            vals.insert(n, None)\n",
    "    # Add a gap at position '0', so that we effectively count from 1:\n",
    "    vals.insert(0, None)\n",
    "    return vals\n",
    "\n",
    "e = modeller.environ()\n",
    "a = modeller.alignment(e, file='humanH1-1HSTA.ali')\n",
    "\n",
    "template = get_profile('1HSTA.profile', a['1HSTA'])\n",
    "model = get_profile('model.profile', a['humanH1'])\n",
    "\n",
    "# Plot the template and model profiles in the same plot for comparison:\n",
    "pylab.figure(1, figsize=(10,6))\n",
    "pylab.xlabel('Alignment position')\n",
    "pylab.ylabel('DOPE per-residue score')\n",
    "pylab.plot(model, color='red', linewidth=2, label='Model')\n",
    "pylab.plot(template, color='green', linewidth=2, label='Template')\n",
    "pylab.legend()\n",
    "pylab.show()\n",
    "# pylab.savefig('dope_profile.png', dpi=65)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:moldyn]",
   "language": "python",
   "name": "conda-env-moldyn-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
